## Abstract {.page_break_before}

Correlation coefficients are used across different research areas to identifying intriguing relationships or answer questions.
Knowing that genes have correlated expression can suggest shared function or influence on shared traits.
Gene regulatory networks are playing an increasingly important role in precision medicine since the introduction of more advanced models of the genetic architecture of complex traits.
These strategies are usually deployed with linear or rank-based correlation coefficients, which are not well suited to discovering non-linear or non-monotonic patterns.
We provide a computationally efficient, easy-to-use implementation of Clustermatch, a not-only-linear correlation coefficient based on machine learning models.
Clustermatch derives a machine learning based similarity value between features, and it has a single parameter.
Applying Clustermatch to human gene expression data reveals both linear and biologically meaningful nonlinear gene-gene relationships.
Clustermatch is most similar to the previously described Maximal Information Coefficient, although Clustermatch runs in a fraction of time and can be practically applied to genome-scale data.
We anticipate that Clustermatch will dramatically improve the detection of crucial molecular patterns completely missed by standard coefficients.
