## Abstract {.page_break_before}

Correlation coefficients are used across different research areas to identify intriguing relationships or answer critical questions.
Knowing that genes have correlated expression can suggest that they share functions or are part of networks that influence different traits.
Indeed, gene regulatory networks have been playing an increasingly important role in precision medicine since the introduction of more advanced models of the genetic architecture of complex traits.
However, the strategies to detect these gene-gene connections are usually deployed with linear correlation coefficients, which are not well suited to discovering more complex, nonlinear patterns.
Here we introduce the Clustermatch Correlation Coefficient (CCC), an efficient, easy-to-use and not-only-linear coefficient based on machine learning.
Clustermatch derives a similarity value between numerical and categorical features by applying clustering algorithms on objects.
Applying Clustermatch to human gene expression data reveals both linear and biologically meaningful nonlinear gene-gene relationships.
We show that Clustermatch significantly improves the detection of different types of relationships, it is robust to outliers, and its single parameter can easily balance between pattern complexity and computation time.
Clustermatch is most similar to the previously described Maximal Information Coefficient, although our method runs in a fraction of time and can be practically applied to genome-scale data.
We anticipate that Clustermatch will dramatically improve the detection of crucial molecular patterns completely missed by standard coefficients.
