## Introduction

Data collection technology has vastly improved, providing a wealth of information across disciplines.
This data offers the potential to answer scientific questions, but only if we have the right tools to detect patterns.
Correlation analysis is an important statistical method for understanding relationships between variables [@pmid:21310971].
Correlation coefficients are used for exploratory data mining, such as clustering or community detection, to measure similarity between objects like genes [@pmid:27479844] or lifestyle factors [@doi:10.1073/pnas.1217269109].
They are also used in supervised tasks, like feature selection, to improve prediction accuracy [@pmid:27006077; @pmid:33729976].
The Pearson correlation coefficient is widely used across many areas, so even minor improvements could have a major impact on research and industry.


The analysis of gene-gene relationships has become increasingly important in the study of human diseases.
To uncover the molecular mechanisms underlying complex traits, researchers have used correlation analysis to suggest gene function [@pmid:21241896], discover common and cell lineage-specific regulatory networks [@pmid:25915600], and capture important interactions in a living organism [@pmid:21606319; @pmid:16968540].
Advances in RNA-seq datasets [@pmid:32913098; @pmid:34844637] have enabled researchers to explore the transcriptional mechanisms of disease [@pmid:27479844; @pmid:31121115; @pmid:30668570; @pmid:32424349; @pmid:34475573].
With the introduction of the omnigenic model of complex traits [@pmid:28622505; @pmid:31051098], gene-gene relationships are now being used to prioritize "core" genes that directly affect diseases [@doi:10.1186/s13040-020-00216-9; @doi:10.1101/2021.07.05.450786; @doi:10.1101/2021.10.21.21265342].
These core genes are not identified by standard statistical methods but are believed to be part of highly-interconnected, disease-relevant regulatory networks.
An efficient not-only-linear correlation coefficient based on machine learning could be used to prioritize candidate drug targets in the precision medicine field and find wide applications across many areas of biology.


Pearson and Spearman correlation coefficients are widely used to capture linear or monotonic relationships in data.
However, they may miss complex yet critical nonlinear patterns.
Novel coefficients such as Maximal Information Coefficient (MIC) [@pmid:22174245] and Distance Correlation (DC) [@doi:10.1214/009053607000000505] have been proposed to capture these relationships, and have been successfully applied across several domains [@pmid:33972855; @pmid:33001806; @pmid:27006077].
However, their computational complexity makes them impractical for even moderately sized datasets [@pmid:33972855; @pmid:27333001].
We previously developed a clustering method that outperformed Pearson, Spearman, DC and MIC in detecting clusters of simulated linear and nonlinear relationships with varying noise levels [@doi:10.1093/bioinformatics/bty899].
Here, we introduce the Clustermatch Correlation Coefficient (CCC), an efficient not-only-linear coefficient that works across quantitative and qualitative variables.
CCC has a single parameter that limits the maximum complexity of relationships found (from linear to more general patterns) and computation time.
We assessed its performance on gene expression data from the Genotype-Tissue Expression v8 (GTEx) project across different tissues [@doi:10.1126/science.aaz1776], and found that CCC captured both strong linear relationships and novel nonlinear patterns, which were entirely missed by standard coefficients.
For example, some of these nonlinear patterns were associated with sex differences in gene expression.
We also found that the CCC behaves similarly to MIC in several cases, but is much faster to compute.
Furthermore, CCC's ability to efficiently handle diverse data types (including numerical and categorical features) reduces preprocessing steps and makes it appealing to analyze large and heterogeneous repositories.
