## Introduction


New technologies have vastly improved data collection and thus generated a deluge of information across different disciplines.
This large amount of data provides new opportunities to address unanswered scientific questions, provided we have both efficient and complex enough tools for the analysis.
For this purpose, correlation analysis is an essential statistical technique to discover relationships between variables [@pmid:21310971].
Correlation coefficients are used in fundamental exploratory data mining techniques, such as clustering or community detection algorithms, to compute a similarity value between a pair of objects of scientific interest such as genes [@pmid:27479844] or morpho-agronomic traits in crop plans [@doi:10.1093/bioinformatics/bty899].
Correlation methods are also successfully used in supervised tasks for feature selection to improve prediction accuracy [@pmid:27006077; @pmid:33729976].
Coefficients such as Pearson or Spearman are ubiquitous across different application domains and scientific areas.
Even minor and significant improvements in these techniques could have enormous consequences in industry and research.


In transcriptomics, almost every single analysis starts with the correlation between a pair of genes.
More sophisticated approaches built on correlation analysis can suggest gene function [@pmid:21241896], aid in discovering common and cell lineage-specific regulatory networks [@pmid:25915600], and capture important interactions in a living organism that can illuminate gene function in other species [@pmid:21606319; @pmid:16968540].
The analysis of large RNA-seq datasets [@pmid:32913098; @pmid:34844637] can also reveal complex transcriptional mechanisms underlying human diseases [@pmid:27479844; @pmid:31121115; @pmid:30668570; @pmid:32424349; @pmid:34475573].
Since the introduction of the omnigenic model of complex traits [@pmid:28622505; @pmid:31051098], gene-gene relationships are playing an increasingly important role in genetic studies of human diseases [@pmid:34845454; @doi:10.1101/2021.07.05.450786; @doi:10.1101/2021.10.21.21265342].
In this context, very recent approaches combine disease-associated genes from genome-wide association studies (GWAS) with gene co-expression networks to prioritize "core" genes directly driving diseases.
These core genes are not captured by standard statistical methods but are part of disease-relevant and highly-interconnected regulatory networks.
Therefore, more sophisticated correlation coefficients could dramatically improve the identification of more attractive candidate drug targets in the precision medicine field.


Standard correlation methods such as Pearson and Spearman are the most popular approaches since they use simple statistics that can be computed quickly.
However, they can only capture linear or monotonic patterns, which might certainly not be enough to detect complex relationships in data across different domains.
Novel approaches, such as Maximal Information Coefficient (MIC) [@pmid:22174245] or Distance Correlation (DC) [@doi:10.1214/009053607000000505] can capture nonlinear patterns.
However, they are impractical not only for big data but also for even moderately sized datasets.
We previously showed that Clustermatch, a method for cluster analysis on highly diverse datasets, significantly outperformed Pearson, Spearman, MIC and DC in detecting simulated linear and nonlinear relationships with varying levels of noise [@doi:10.1093/bioinformatics/bty899].
Here we present the Clustermatch's correlation coefficient, an efficient not-only-linear method that can handle quantitative and qualitative variables.
We applied Clustermatch to gene expression data from the Genotype-Tissue Expression (GTEx) project, where it captured both known linear relationships and novel nonlinear and biologically meaningful gene-gene patterns completely missed by standard coefficients.
<!-- TODO: Add something about GIANT results? -->
Clustermatch scores distribute very similarly to MIC, although it is much faster to compute and results are easy to interpret.
It also has a single parameter that controls the complexity of relationships of interest and computation time.
Our results show that Clutermatch is an attractive replacement for standard linear-only coefficients that can be readily used across different research areas.
Its ability to efficiently handle diverse data types (including numerical and categorical features) reduces preprocessing steps and makes it appealing to analyze large and complex repositories.
