## Introduction

The sheer volume of data available today has led to the need for more efficient correlation coefficients capable of uncovering nonlinear relationships.
Machine learning techniques have been used to develop novel correlation coefficients that can identify nonlinear relationships [@pmid:29098265].
In this paper, we present an efficient not-only-linear correlation coefficient based on machine learning.
Our approach is designed to identify nonlinear relationships between variables, providing a more accurate representation of the data than the Pearson correlation coefficient.
We evaluate the performance of our proposed coefficient on a variety of datasets and compare it with other correlation methods.


In transcriptomics, many analyses start with estimating the correlation between genes.
This can suggest gene function [@pmid:21241896], aid in discovering common and cell lineage-specific regulatory networks [@pmid:25915600], and capture important interactions in a living organism that can uncover molecular mechanisms in other species [@pmid:21606319; @pmid:16968540].
The analysis of large RNA-seq datasets [@pmid:32913098; @pmid:34844637] can also reveal complex transcriptional mechanisms underlying human diseases [@pmid:27479844; @pmid:31121115; @pmid:30668570; @pmid:32424349; @pmid:34475573].
Since the introduction of the omnigenic model of complex traits [@pmid:28622505; @pmid:31051098], gene-gene relationships have become increasingly important in genetic studies of human diseases [@pmid:34845454; @doi:10.1101/2021.07.05.450786; @doi:10.1101/2021.10.21.21265342; @doi:10.1038/s41588-021-00913-z], even in specific fields such as polygenic risk scores [@doi:10.1016/j.ajhg.2021.07.003].
In this context, recent approaches combine disease-associated genes from genome-wide association studies (GWAS) with gene co-expression networks to prioritize "core" genes that directly affect diseases [@doi:10.1186/s13040-020-00216-9; @doi:10.1101/2021.07.05.450786; @doi:10.1101/2021.10.21.21265342].
These core genes are not captured by standard statistical methods, but they are believed to be part of highly-interconnected, disease-relevant regulatory networks.
Therefore, advanced correlation coefficients could immediately find wide applications across many areas of biology, including the prioritization of candidate drug targets in precision medicine.


The Pearson and Spearman correlation coefficients are widely used because they capture linear or monotonic relationships quickly and intuitively.
However, they may miss complex yet critical nonlinear patterns.
To capture such patterns, novel coefficients have been proposed such as the Maximal Information Coefficient (MIC) [@pmid:22174245] and the Distance Correlation (DC) [@doi:10.1214/009053607000000505], which have been successfully applied across several domains [@pmid:33972855; @pmid:33001806; @pmid:27006077].
However, they are computationally expensive and impractical for even moderately sized datasets [@pmid:33972855; @pmid:27333001].
We previously developed a clustering method that outperformed Pearson, Spearman, DC and MIC in detecting clusters of simulated linear and nonlinear relationships with varying noise levels [@doi:10.1093/bioinformatics/bty899].
Here, we introduce the Clustermatch Correlation Coefficient (CCC), an efficient not-only-linear coefficient that works across quantitative and qualitative variables.
CCC has a single parameter that limits the maximum complexity of relationships found (from linear to more general patterns) and computation time.
We provide an efficient CCC implementation that is highly parallelizable, allowing for faster computation across variable pairs with millions of objects or conditions.
To assess its performance, we applied CCC to gene expression data from the Genotype-Tissue Expression v8 (GTEx) project across different tissues [@doi:10.1126/science.aaz1776].
We found that CCC captured both strong linear relationships and novel nonlinear patterns that were missed by standard coefficients.
For example, some of these nonlinear patterns were associated with sex differences in gene expression.
In addition, CCC behaved similarly to MIC in several cases, although it was much faster to compute.
Moreover, its ability to handle diverse data types (including numerical and categorical features) reduces preprocessing steps and makes it appealing to analyze large and heterogeneous repositories.
