## Introduction

Data collection technologies have greatly advanced, generating a huge amount of data across various disciplines.
This data provides new possibilities to answer unresolved scientific questions, but only if we have efficient tools that can identify different types of patterns.
Correlation analysis is an important statistical tool for finding relationships between variables [@pmid:21310971].
Correlation coefficients are used to measure the similarity between objects such as genes [@pmid:27479844] or lifestyle factors related to diseases [@doi:10.1073/pnas.1217269109] in exploratory data mining techniques like clustering or community detection algorithms.
They are also used in supervised tasks, for instance, to improve prediction accuracy with feature selection [@pmid:27006077; @pmid:33729976].
The Pearson correlation coefficient is widely used in many application domains and scientific areas.
Therefore, even minor improvements in this technique could have major impacts in industry and research.


Estimating the correlation between genes is a starting point for many transcriptomics analyses.
This approach can suggest gene function and help discover regulatory networks, as well as uncover molecular mechanisms in other species [@pmid:21241896; @pmid:25915600; @pmid:21606319; @pmid:16968540].
RNA-seq datasets can also reveal complex transcriptional mechanisms underlying human diseases [@pmid:32913098; @pmid:34844637; @pmid:27479844; @pmid:31121115; @pmid:30668570; @pmid:32424349; @pmid:34475573].
The introduction of the omnigenic model of complex traits [@pmid:28622505; @pmid:31051098] has made gene-gene relationships increasingly important in genetic studies of human diseases [@pmid:34845454; @doi:10.1101/2021.07.05.450786; @doi:10.1101/2021.10.21.21265342; @doi:10.1038/s41588-021-00913-z], including polygenic risk scores [@doi:10.1016/j.ajhg.2021.07.003].
Combining disease-associated genes from GWAS with gene co-expression networks has enabled the prioritization of "core" genes that directly affect diseases [@doi:10.1186/s13040-020-00216-9; @doi:10.1101/2021.07.05.450786; @doi:10.1101/2021.10.21.21265342].
Advanced correlation coefficients can be used to identify these core genes, which are not captured by standard statistical methods, and could have wide applications in biology, such as prioritizing candidate drug targets in precision medicine.


Pearson and Spearman correlation coefficients are widely used because they are able to reveal intuitive linear or monotonic relationships quickly.
However, they are not designed to detect complex, yet critical, nonlinear patterns.
Novel metrics, such as the Maximal Information Coefficient (MIC) [@pmid:22174245] and the Distance Correlation (DC) [@doi:10.1214/009053607000000505], have been proposed in order to capture more complex relationships.
These metrics have been successfully applied across several domains [@pmid:33972855; @pmid:33001806; @pmid:27006077], but their computational complexity makes them impractical for even moderately sized datasets [@pmid:33972855; @pmid:27333001].
For example, it takes several seconds to compute MIC on a single variable pair across a few thousand objects or conditions [@pmid:33972855].
We recently developed a clustering method that outperformed approaches based on Pearson, Spearman, DC and MIC in detecting clusters of simulated linear and nonlinear relationships with varying noise levels [@doi:10.1093/bioinformatics/bty899].
Here, we introduce the Clustermatch Correlation Coefficient (CCC), an efficient not-only-linear coefficient that captures quantitative and qualitative variables.
CCC has a single parameter that limits the maximum complexity of relationships found (from linear to more general patterns) and computation time.
Additionally, it provides a high level of flexibility to detect specific types of patterns important to the user, while providing safe defaults to capture general relationships.
We also provide an efficient, highly parallelizable CCC implementation, which allows for faster computation across variable pairs with millions of objects or conditions.
To assess its performance, we applied our method to gene expression data from the Genotype-Tissue Expression v8 (GTEx) project across different tissues [@doi:10.1126/science.aaz1776].
CCC was able to detect both strong linear relationships and novel nonlinear patterns, which were entirely missed by standard coefficients.
For example, some of these nonlinear patterns were associated with sex differences in gene expression, suggesting that CCC can capture strong relationships present only in a subset of samples.
We also found that CCC behaves similarly to MIC in several cases, although it is much faster to compute.
Moreover, gene pairs detected in expression data by CCC had higher interaction probabilities in tissue-specific gene networks from the Genome-wide Analysis of gene Networks in Tissues (GIANT) [@doi:10.1038/ng.3259].
Furthermore, CCC's ability to efficiently handle diverse data types (including numerical and categorical features) reduces preprocessing steps and makes it appealing to analyze large and heterogeneous repositories.
