## Introduction

Data collection technologies have advanced significantly, creating a wealth of information in various disciplines.
To make use of this data, we need efficient tools to identify different types of patterns.
Correlation analysis is a statistical technique that helps uncover relationships between variables [@pmid:21310971].
Correlation coefficients are used in exploratory data mining, such as clustering and community detection algorithms, to determine the similarity between two objects, such as genes [@pmid:27479844] or lifestyle factors [@doi:10.1073/pnas.1217269109].
They are also used in supervised tasks, such as feature selection, to improve prediction accuracy [@pmid:27006077; @pmid:33729976].
The Pearson correlation coefficient is widely used in various fields, so even small improvements to this technique could have a major impact on research and industry.


Many studies in transcriptomics begin by estimating the correlation between genes.
This approach can suggest gene function [@pmid:21241896], uncover common and cell lineage-specific regulatory networks [@pmid:25915600], and reveal interactions in living organisms that can inform molecular mechanisms in other species [@pmid:21606319; @pmid:16968540].
The analysis of large RNA-seq datasets [@pmid:32913098; @pmid:34844637] can also uncover complex transcriptional mechanisms underlying human diseases [@pmid:27479844; @pmid:31121115; @pmid:30668570; @pmid:32424349; @pmid:34475573].
The introduction of the omnigenic model of complex traits [@pmid:28622505; @pmid:31051098] has made gene-gene relationships increasingly important in genetic studies of human diseases [@pmid:34845454; @doi:10.1101/2021.07.05.450786; @doi:10.1101/2021.10.21.21265342; @doi:10.1038/s41588-021-00913-z], even for specific fields such as polygenic risk scores [@doi:10.1016/j.ajhg.2021.07.003].
Recent approaches combine disease-associated genes from GWAS with gene co-expression networks to identify core genes that directly affect diseases [@doi:10.1186/s13040-020-00216-9; @doi:10.1101/2021.07.05.450786; @doi:10.1101/2021.10.21.21265342].
These core genes are not captured by standard statistical methods, but they are believed to be part of highly-interconnected, disease-relevant regulatory networks.
Thus, advanced correlation coefficients could be useful for a range of biological applications, including the prioritization of candidate drug targets in precision medicine.


The Pearson and Spearman correlation coefficients are widely used because they reveal intuitive linear or monotonic relationships quickly.
However, they may miss more complex yet critical patterns.
To address this, other coefficients have been proposed such as the Maximal Information Coefficient (MIC) [@pmid:22174245] and the Distance Correlation (DC) [@doi:10.1214/009053607000000505], which have been used successfully across several domains [@pmid:33972855; @pmid:33001806; @pmid:27006077].
However, their computational complexity makes them impractical for even moderately sized datasets [@pmid:33972855; @pmid:27333001].
We have previously developed a clustering method that significantly outperformed Pearson, Spearman, DC, and MIC in detecting clusters of linear and nonlinear relationships with varying noise levels [@doi:10.1093/bioinformatics/bty899].
Here we introduce the Clustermatch Correlation Coefficient (CCC), an efficient not-only-linear coefficient that works across quantitative and qualitative variables.
CCC has a single parameter that limits the maximum complexity of relationships found (from linear to more general patterns) and computation time.
It also provides a high level of flexibility to detect specific types of patterns that are more important to the user, while providing safe defaults to capture general relationships.
We provide an efficient CCC implementation that is highly parallelizable, allowing it to speed up computation across variable pairs with millions of objects or conditions.
We applied our method to gene expression data from the Genotype-Tissue Expression v8 (GTEx) project across different tissues [@doi:10.1126/science.aaz1776], and found that CCC captured both strong linear relationships and novel nonlinear patterns that were missed by standard coefficients.
For example, some of these nonlinear patterns were associated with sex differences in gene expression.
We also found that CCC behaves similarly to MIC in several cases, although it is much faster to compute.
Furthermore, gene pairs detected in expression data by CCC had higher interaction probabilities in tissue-specific gene networks from the Genome-wide Analysis of gene Networks in Tissues (GIANT) [@doi:10.1038/ng.3259].
CCC's ability to efficiently handle diverse data types (including numerical and categorical features) reduces preprocessing steps and makes it appealing to analyze large and heterogeneous repositories.
