## Introduction

Data collection has advanced significantly, generating more information than ever before in a variety of disciplines.
This data provides opportunities to answer scientific questions, but only if we have efficient tools to identify the patterns present.
Correlation analysis is a statistical tool used to uncover relationships between variables [@pmid:21310971].
It is used in exploratory data mining techniques such as clustering or community detection algorithms to measure similarity between objects such as genes [@pmid:27479844] or lifestyle factors [@doi:10.1073/pnas.1217269109].
It is also used in supervised tasks such as feature selection to increase prediction accuracy [@pmid:27006077; @pmid:33729976].
The Pearson correlation coefficient is widely used across various scientific fields.
Even small improvements to this technique could have a significant impact on industry and research.


Estimating the correlation between genes is a common starting point for transcriptomics analyses, which can suggest gene function [@pmid:21241896], aid in discovering common and cell lineage-specific regulatory networks [@pmid:25915600], and uncover important interactions in a living organism [@pmid:21606319; @pmid:16968540].
Large RNA-seq datasets [@pmid:32913098; @pmid:34844637] can reveal complex transcriptional mechanisms underlying human diseases [@pmid:27479844; @pmid:31121115; @pmid:30668570; @pmid:32424349; @pmid:34475573].
Since the introduction of the omnigenic model of complex traits [@pmid:28622505; @pmid:31051098], gene-gene relationships play an increasingly important role in genetic studies of human diseases [@pmid:34845454; @doi:10.1101/2021.07.05.450786; @doi:10.1101/2021.10.21.21265342; @doi:10.1038/s41588-021-00913-z], such as polygenic risk scores [@doi:10.1016/j.ajhg.2021.07.003].
Combining disease-associated genes from genome-wide association studies (GWAS) with gene co-expression networks can prioritize "core" genes directly affecting diseases [@doi:10.1186/s13040-020-00216-9; @doi:10.1101/2021.07.05.450786; @doi:10.1101/2021.10.21.21265342], which are not captured by standard statistical methods but are believed to be part of highly-interconnected, disease-relevant regulatory networks.
Advanced correlation coefficients could then be applied across many areas of biology, including the prioritization of candidate drug targets in the precision medicine field.


Pearson and Spearman correlation coefficients are widely used for their ability to quickly reveal linear or monotonic relationships.
However, they may miss more complex patterns, which can be critical.
To capture nonlinear patterns, the Maximal Information Coefficient (MIC) and the Distance Correlation (DC) have been proposed [@pmid:22174245; @doi:10.1214/009053607000000505].
MIC, in particular, has had successful applications in various domains [@pmid:33972855; @pmid:33001806; @pmid:27006077], but its computational complexity makes it impractical for larger datasets [@pmid:33972855; @pmid:27333001].
We previously developed a clustering method that outperformed Pearson, Spearman, DC, and MIC in detecting clusters of linear and nonlinear relationships with varying noise levels [@doi:10.1093/bioinformatics/bty899].
Here, we introduce the Clustermatch Correlation Coefficient (CCC), an efficient not-only-linear coefficient for quantitative and qualitative variables.
CCC has a single parameter that can limit the complexity of the relationships and computation time.
It also provides safe defaults to capture general relationships.
Furthermore, our efficient implementation is highly parallelizable, allowing for faster computation on variable pairs with millions of objects or conditions.
To assess its performance, we applied CCC to gene expression data from the Genotype-Tissue Expression v8 (GTEx) project across different tissues [@doi:10.1126/science.aaz1776].
CCC detected both linear and nonlinear patterns, including those associated with sex differences in gene expression.
Additionally, it behaved similarly to MIC in several cases while being much faster to compute.
Gene pairs detected in expression data by CCC had higher interaction probabilities in tissue-specific gene networks from the Genome-wide Analysis of gene Networks in Tissues (GIANT) [@doi:10.1038/ng.3259].
Lastly, its ability to handle diverse data types (including numerical and categorical features) reduces preprocessing steps and makes it suitable for analyzing large and heterogeneous repositories.
