## Introduction

New technologies have enabled us to collect vast amounts of data in various disciplines.
This data offers opportunities to answer previously unanswered scientific questions, provided we have the right tools to identify different types of patterns.
Correlation analysis is a statistical technique that is used to determine relationships between variables [@pmid:21310971].
This method is often employed in data mining, for instance, to calculate the similarity between genes [@pmid:27479844] or lifestyle factors relevant to diseases [@doi:10.1073/pnas.1217269109].
It is also used in supervised tasks, such as feature selection, to improve prediction accuracy [@pmid:27006077; @pmid:33729976].
The Pearson correlation coefficient is widely used in a variety of application domains and scientific fields.
Therefore, even small improvements in this technique could have a significant impact on research and industry.


In the field of transcriptomics, correlation analysis is used to estimate the relationship between genes.
This approach has been used to suggest gene function [@pmid:21241896], identify common and cell-specific regulatory networks [@pmid:25915600], and uncover molecular mechanisms in different species [@pmid:21606319; @pmid:16968540].
The analysis of large RNA-seq datasets [@pmid:32913098; @pmid:34844637] has also revealed complex transcriptional mechanisms underlying human diseases [@pmid:27479844; @pmid:31121115; @pmid:30668570; @pmid:32424349; @pmid:34475573].
The introduction of the omnigenic model of complex traits [@pmid:28622505; @pmid:31051098] has emphasized the importance of gene-gene relationships in genetic studies of human diseases [@pmid:34845454; @doi:10.1101/2021.07.05.450786; @doi:10.1101/2021.10.21.21265342; @doi:10.1038/s41588-021-00913-z], including those related to polygenic risk scores [@doi:10.1016/j.ajhg.2021.07.003].
Recent approaches have combined disease-associated genes from genome-wide association studies (GWAS) with gene co-expression networks to prioritize "core" genes that directly affect diseases [@doi:10.1186/s13040-020-00216-9; @doi:10.1101/2021.07.05.450786; @doi:10.1101/2021.10.21.21265342].
These core genes cannot be identified using standard statistical methods, but are believed to be part of highly-interconnected, disease-relevant regulatory networks.
Therefore, advanced correlation coefficients could have a wide range of applications in biology, including the prioritization of candidate drug targets in precision medicine.


The Pearson and Spearman correlation coefficients are widely used because they can quickly reveal intuitive linear or monotonic relationships.
However, these coefficients may miss more complex relationships.
Several novel metrics have been proposed to capture nonlinear patterns, such as the Maximal Information Coefficient (MIC) [@pmid:22174245] and the Distance Correlation (DC) [@doi:10.1214/009053607000000505].
MIC has been successfully applied across a range of domains [@pmid:33972855; @pmid:33001806; @pmid:27006077], but its computational complexity makes it impractical for moderately sized datasets [@pmid:33972855; @pmid:27333001].
For example, computing MIC on a single variable pair across a few thousand objects or conditions can take several seconds [@pmid:33972855].
We previously developed a clustering method that outperformed Pearson, Spearman, DC and MIC in detecting clusters of simulated linear and nonlinear relationships with varying noise levels [@doi:10.1093/bioinformatics/bty899].
Here, we introduce the Clustermatch Correlation Coefficient (CCC), an efficient not-only-linear coefficient that works across quantitative and qualitative variables.
CCC has a single parameter that limits the maximum complexity of relationships found (from linear to more general patterns) and computation time.
Additionally, it is highly parallelizable, allowing for faster computation across variable pairs with millions of objects or conditions.
We assessed its performance by applying it to gene expression data from the Genotype-Tissue Expression v8 (GTEx) project across different tissues [@doi:10.1126/science.aaz1776].
CCC captured both strong linear relationships and novel nonlinear patterns, which were missed by standard coefficients.
For example, some of these nonlinear patterns were associated with sex differences in gene expression.
Additionally, CCC behaved similarly to MIC in several cases, although it is much faster to compute.
Gene pairs detected in expression data by CCC had higher interaction probabilities in tissue-specific gene networks from the Genome-wide Analysis of gene Networks in Tissues (GIANT) [@doi:10.1038/ng.3259].
Moreover, its ability to efficiently handle numerical and categorical features reduces preprocessing steps and makes it suitable for analyzing large and heterogeneous repositories.
