## Introduction


New technologies have vastly improved data collection, generating a deluge of information across different disciplines.
This large amount of data provides new opportunities to address unanswered scientific questions, provided we have efficient tools that implement models that are sufficiently complex to discover underlying patterns.
Correlation analysis is an essential statistical technique to discover relationships between variables [@pmid:21310971].
Correlation coefficients are often used in exploratory data mining techniques, such as clustering or community detection algorithms, to compute a similarity value between a pair of objects of scientific interest such as genes [@pmid:27479844] or morpho-agronomic traits in crop plans [@doi:10.1093/bioinformatics/bty899].
Correlation methods are sometimes used in supervised tasks for feature selection to improve prediction accuracy [@pmid:27006077; @pmid:33729976].
The Pearson correlation coefficient is ubiquitously deployed across application domains and diverse scientific areas.


In transcriptomics, many analyses start with estimating the correlation between genes.
More sophisticated approaches built on correlation analysis can suggest gene function [@pmid:21241896], aid in discovering common and cell lineage-specific regulatory networks [@pmid:25915600], and capture important interactions in a living organism that can uncover molecular mechanisms in other species [@pmid:21606319; @pmid:16968540].
The analysis of large RNA-seq datasets [@pmid:32913098; @pmid:34844637] can also reveal complex transcriptional mechanisms underlying human diseases [@pmid:27479844; @pmid:31121115; @pmid:30668570; @pmid:32424349; @pmid:34475573].
Since the introduction of the omnigenic model of complex traits [@pmid:28622505; @pmid:31051098], gene-gene relationships are playing an increasingly important role in genetic studies of human diseases [@pmid:34845454; @doi:10.1101/2021.07.05.450786; @doi:10.1101/2021.10.21.21265342], even in specific fields such as polygenic risk scores [@doi:10.1016/j.ajhg.2021.07.003].
In this context, recent approaches combine disease-associated genes from genome-wide association studies (GWAS) with gene co-expression networks to prioritize "core" genes directly driving diseases. <!--Reference?-->
These core genes are not captured by standard statistical methods but are part of disease-relevant and highly-interconnected regulatory networks.
Therefore, more sophisticated correlation coefficients could dramatically improve the identification of more attractive candidate drug targets in the precision medicine field.


The Pearson and Spearman correlation coefficients are widely used because they reveal intuitive types of relationships and can be computed quickly.
However, they can only capture linear or monotonic patterns, which can miss complex yet essential relationships.
The Maximal Information Coefficient (MIC) [@pmid:22174245] or Distance Correlation (DC) [@doi:10.1214/009053607000000505] were proposed as metrics that capture nonlinear patterns.
However, they are impractical not only for big data but also for even moderately sized datasets.
We previously developed Clustermatch, a method for cluster analysis on highly diverse datasets, significantly outperformed Pearson, Spearman, MIC and DC in detecting simulated linear and nonlinear relationships with varying levels of noise [@doi:10.1093/bioinformatics/bty899].
Here we introduce the Clustermatch correlation coefficient (CCC)<!--maybe? felt like everything else had an acronym-->, an efficient not-only-linear coefficient that works across quantitative and qualitative variables.
To assess its performance in RNA-seq data, we applied our method to gene expression data from the Genotype-Tissue Expression (GTEx) project across different tissues.
Clustermatch captured both known linear relationships and novel nonlinear and biologically meaningful gene-gene patterns completely missed by standard coefficients.
<!-- TODO: Add something about GIANT results? -->
The Clustermatch coefficient is most similar to MIC, although it is much faster to compute.
It also has a single parameter that balances the complexity of relationships found and computation time.
We find that Clustermatch can complement linear-only coefficients.
Its ability to efficiently handle diverse data types (including numerical and categorical features) reduces preprocessing steps and makes it appealing to analyze large and complex repositories.
