## Introduction


The development of new technologies has vastly improved data collection and thus generated a deluge of information across different diciplines.
This provides new opportunities to address unanswered scientific questions [CITATION], provided we have both efficient and complex enough tools for the analysis.
For this purpose, correlation analysis is an essential statistical technique to discover relationships between variables [@pmid:21310971].
Correlation coefficients are used in many fundamental exporatory data mining techniques, such as a clustering or community detection, to compute a similarity value between a pair of objects of scientific interest such as genes [@doi:10.1038/nn.4353] or morpho-agronomic traits in crop plans [@doi:10.1093/bioinformatics/bty899].
Correlation methods were also successfuly used in supervised tasks for feature selection to improve prediction accuracy [@pmid:27006077; @doi:10.1109/TCYB.2021.3061152].
Coefficients such as Pearson or Spearman are obiquitous across different application domains and scientific areas, such as engineering or medicine, and even minor and significant improvements in these techniques could have enormous concequences in industry and research.


In transcriptomics, almost every single analysis starts with the correlation between a pair of genes, which can suggest their function [@doi:10.1016/j.cell.2011.01.004], aid in discovering common and cell lineage-specific regulatory networks [@doi:10.1038/ng.3259], and capture important important interactions in a living organism that can iluminate their function in other species [@pmid:21606319; @pmid:16968540].
The analysis of large RNA-seq datasets [@doi:10.1126/science.aaz1776; @doi:10.1186/s13059-021-02533-6] can also reveal complex transcriptional mechanisms underlying human diseases [
    @doi:@doi:10.1038/nn.4353; <!-- Genome-wide prediction and functional characterization of the genetic basis of autism spectrum disorder -->
    @doi:10.1016/j.cels.2019.04.003; <!-- MultiPLIER -->
    @doi:10.1371/journal.pgen.1007889; <!-- MultiXcan -->
    @doi:10.1038/s41588-020-0625-2; <!-- Quantifying genetic effects on disease mediated by assayed gene expression levels -->
    @doi:10.1038/s41588-021-00913-z; <!-- Large-scale cis- and trans-eQTL analyses identify thousands of genetic loci and polygenic scores that regulate blood gene expression -->
].
Since the introduction of the omnigenic model of complex traits [@doi:10.1016/j.cell.2017.05.038; @doi:10.1016/j.cell.2019.04.014], gene-gene relationships are playing an increasingly important role in genetic studies of diseases [
    @doi:10.1101/2021.03.19.436212; <!-- Identifying disease-critical cell types and cellular processes across the human body by integration of single-cell profiles and human genetics -->
    @doi:10.1101/2021.07.05.450786; <!-- PhenoPLIER -->
    @doi:10.1101/2021.10.21.21265342; <!-- Linking common and rare disease genetics through gene regulatory networks -->
].
These very recent approaches combine disease-associated genes (via genome-wide association studies or GWAS) with gene co-expression networks to prioritize key genes not captured by standard methods, but potentially part of disease-relevant and highly-interconnected regulatory networks.


Standard correlation methods such as Pearson and Spearman are by far the most popular approaches, since they use easy-to-compute statistics and can therefore be quickly computed.
However, they can only capture linear or monotonic patterns, which might certeinly not be enough to detect complex relationships in data across different domains.
Novel approaches, such as Maximal Information Coefficient (MIC) [REF] or Distance Correlation (DC) [REF] can capture nonlinear patterns, but they are impractical not only for big data but also for even just moderately sized datasets.
We previously shown that Clustermatch, a method for cluster analysis on highly diverse datasets, significantly outperformed Pearson, Spearman, MIC and DC in detecting linear and nonlinear relationships with varying levels of noise [@doi:10.1093/bioinformatics/bty899].
Here we present the Clustermatch's correlation method, an efficient not-only-linear coefficient that can handle both quantitative and qualitative variables.
We applied Clustermatch to gene expression data from the Genotype-Tissue Expression (GTEx) project and showed that it can detect known linear and also novel nonlinear and biologically meaningful gene-gene patterns completely missed by standard coefficients.
<!-- TODO: Add something about GIANT results? -->
Clustermatch scores distribute very similarly to MIC, although it is much faster to compute and results are easy to interpret.
It also has a single parameter that controls the complexity of relationships of interest alongside computation time.
Our results show that Clutermatch is an attrative replacement for standard linear-only coefficients that can be readily used across different research areas.
Its ability to efficiently handle diverse data types (including numerical and categorical features) reduces preprocessing steps and makes it appealing for the analysis of large and complex repositories.





The input to build these graph-based structures are pairwise correlation matrices, where Pearson and Spearman are by far the most common coefficients, mainly because of their fast computation and straightforward interpretation.
However, these coefficients can only capture linear or monotonic relationships between a gene pair, and are suceptible
MIC, DC, 

- there exist methods that can reduce noise or apply different normalization techniques, but this is manual work and time consuming
- another important aspect is that

- the gene co-expression networks can capture these gene
- methods such as random matrix theory (RTM) can detect a threshold under
- noise is a problem; and techniques to deal with it are complex and require several manual steps and be careful ["Addressing noise... paper below"]


AFTER TALKIN ABOUT GCN AND CORRELATION, say that "here we focused on this input, correlation matrices"


- take a look at the other papers related to GIANT (Casey), which are cited in HumanBase, and might have words interesting to put here

"Addressing noise in co-expression network construction"
https://academic.oup.com/bib/article/23/1/bbab495/6446269
