## Introduction

Data collection technology has drastically improved, resulting in an abundance of information in various fields.
To make use of this data, we need efficient tools to recognize patterns.
Correlation analysis is a fundamental statistical technique for finding relationships between variables [@pmid:21310971].
Correlation coefficients are often used in data mining, such as clustering or community detection, to measure similarity between objects, like genes [@pmid:27479844] or lifestyle factors associated with diseases [@doi:10.1073/pnas.1217269109].
They can also be used in supervised tasks, such as feature selection to improve prediction accuracy [@pmid:27006077; @pmid:33729976].
The Pearson correlation coefficient is widely used in many areas of research, so even small improvements can have a huge impact.


In transcriptomics, correlation analysis is a key tool for estimating gene function, discovering regulatory networks, and uncovering molecular mechanisms in living organisms [@pmid:21241896; @pmid:25915600; @pmid:21606319; @pmid:16968540].
Large RNA-seq datasets have revealed complex transcriptional mechanisms underlying human diseases [@pmid:32913098; @pmid:34844637; @pmid:27479844; @pmid:31121115; @pmid:30668570; @pmid:32424349; @pmid:34475573].
The introduction of the omnigenic model of complex traits [@pmid:28622505; @pmid:31051098] has made gene-gene relationships increasingly important in understanding human diseases [@pmid:34845454; @doi:10.1101/2021.07.05.450786; @doi:10.1101/2021.10.21.21265342; @doi:10.1038/s41588-021-00913-z], especially in the field of polygenic risk scores [@doi:10.1016/j.ajhg.2021.07.003].
Combining disease-associated genes from genome-wide association studies (GWAS) with gene co-expression networks can reveal "core" genes directly affecting diseases [@doi:10.1186/s13040-020-00216-9; @doi:10.1101/2021.07.05.450786; @doi:10.1101/2021.10.21.21265342], which are not detectable with standard statistical methods.
Advanced correlation coefficients could thus have a wide range of applications in biology, including the identification of candidate drug targets in precision medicine.


Pearson and Spearman correlation coefficients are widely used to capture linear or monotonic relationships in data.
However, they may miss complex yet critical nonlinear patterns.
To address this, several metrics have been proposed, such as the Maximal Information Coefficient (MIC) and the Distance Correlation (DC).
MIC has been successfully applied to various domains, but its computational complexity makes it impractical for moderately sized datasets.
We previously developed a clustering method that outperformed Pearson, Spearman, DC and MIC in detecting clusters of simulated linear and nonlinear relationships with varying noise levels.
Here we introduce the Clustermatch Correlation Coefficient (CCC), an efficient not-only-linear coefficient that works across quantitative and qualitative variables.
CCC has a single parameter that limits the maximum complexity of relationships found, and is highly parallelizable, allowing for faster computation across variable pairs with millions of objects or conditions.
We assessed its performance on gene expression data from the Genotype-Tissue Expression v8 (GTEx) project across different tissues.
CCC captured both strong linear relationships and novel nonlinear patterns, which were entirely missed by standard coefficients.
Furthermore, its ability to efficiently handle diverse data types reduces preprocessing steps and makes it appealing to analyze large and heterogeneous repositories.
