### Clustermatch detects linear and nonlinear patterns in human transcriptomic data

![
**Distribution of coefficient values on gene expression (GTEx v8, whole blood).**
**a)** Histogram of coefficient values.
**b)** Corresponding cumulative histogram. The dotted line maps the coefficient value that accumulates 70% of gene pairs.
**c)** 2D histogram plot with hexagonal bins between all coefficients, where a logarithmic scale was used to color each hexagon.
](images/coefs_comp/gtex_whole_blood/dist-main.svg "Distribution of coefficient values"){#fig:dist_coefs width="100%"}


We used gene expression data from GTEx v8 across different tissues.
We selected the top 5,000 genes with the largest variance for our initial analyses on whole blood and then computed the pairwise correlation matrix using Pearson, Spearman and Clustermatch.
In Figure @fig:dist_coefs a, we show how the pairwise correlation values distribute, where Clustermatch (mean=0.14, median=0.08, sd=0.15) has a much more skewed distribution than Pearson (mean=0.31, median=0.24, sd=0.24) and especially Spearman (mean=0.39, median=0.37, sd=0.26).
Coefficients reach 70% of gene pairs at $c=0.18$, $p=0.44$ and $s=0.56$ (Figure @fig:dist_coefs b).
Clustermatch and Spearman tend to agree more than any of these with Pearson, although many gene pairs seem to have a relatively higher value for Clustermatch (Figures @fig:dist_coefs c).


![
**Intersection of gene pairs with high and low coefficient values (GTEx v8, whole blood).**
**a)** UpSet plot with six categories (rows) grouping the 30% of the highest (green triangle) and lowest (red triangle) correlation values for each method.
Columns show different intersections of categories grouped by agreements and disagreements.
**b)** Hexagonal binning plots with examples of gene pairs where Clustermatch ($c$) disagrees with Pearson ($p$) and Spearman ($s$).
For each method, green and red triangles indicate if the gene pair obtained a correlation among the top (green) or bottom (red) 30% of correlation values.
No triangle means that the correlation value for the gene pair is between the 30th and 70th percentiles (neither low nor high).
A logarithmic scale was used to color each hexagon.
](images/coefs_comp/gtex_whole_blood/upsetplot-main.svg "Intersection of gene pairs"){#fig:upsetplot_coefs width="100%"}


A closer inspection of gene pairs detected and missed by these coefficients revealed the ability of Clustermatch to capture more complex yet biologically meaningful patterns.
For this, we analyzed the agreements and disagreements by obtaining for each coefficient the top 30% of gene pairs with the largest correlation values ("high" set) and the bottom 30% ("low" set), resulting in six potentially overlapping categories.
An UpSet plot [@doi:10.1109/TVCG.2014.2346248] is shown in Figure @fig:upsetplot_coefs a, where the intersections of these six categories allowed to precisely identify the gene expression patterns captured and missed by each coefficient.
For most cases, the three coefficients agree on whether there is a strong linear correlation (42.1%) and whether there is no relationship (34.3%).
This is crucial because it implies that the user will not miss important linear patterns in expression data when using Clustermatch.
The figure also confirms that Clustermatch and Spearman agree more on highly correlated pairs (4.0% in "high", and 7.0% in "low") than any of these with Pearson (all between 0.3%-3.5% for "high", and 2.8%-5.5% for "low").
Regarding disagreements, more than 20 thousand gene pairs (20,987) with a high Clustermatch value are not highly ranked by any other coefficients.
There are also gene pairs with a high Pearson value with either low Clustermatch (1,075) or low Clustermatch and low Spearman values (531).
However, these cases mostly seem to be driven by outliers (Figure @fig:upsetplot_coefs b).
Clustermatch does not miss gene pairs highly ranked by Spearman.


Figure @fig:upsetplot_coefs b shows individual gene pairs where Clustermatch disagrees with either Pearson, Spearman or both.
The first three gene pairs at the top (*IFNG* / *SDS*, *JUN* / *APOC1*, and *ZDHHC12* / *CCL18*), with a high Clustermatch and low Pearson coefficient, seem to follow a non-coexistence relationship: in samples where one of the genes is highly (slightly) expressed, the other is slightly (highly) activated.

ME QUEDE ACA:
These two genes are strongly expressed in white blood cells, both in myeloid and lymphoid lineages, and using tissue-specific gene networks from GIANT [@pmcid:PMC4828725] we found strong evidence of interactions in these cell types (Supplementary Figure ADD).


![
**Predicted network interactions between different gene pairs.**
Nodes represent genes and edges are the probability that the gene pair is part of the same biological process in a tissue or cell type (indicated at the top of each subfigure).
`SAY SOMETHING ABOUT WEIGHTS`{.red}
This analysis can be performed online using HumanBase. For example, for *SDS* and *IFNG* and *SDS* [@url:https://hb.flatironinstitute.org/gene/10993+3458].
<!-- 
IFNG vs SDS: https://hb.flatironinstitute.org/gene/10993+3458
JUN vs APOC1: https://hb.flatironinstitute.org/gene/3725+341
ZDHHC12 vs CCL18: https://hb.flatironinstitute.org/gene/6362+84885
RASSF2 vs CYTIP: https://hb.flatironinstitute.org/gene/9770+9595
MYOZ1 vs TNNI2: https://hb.flatironinstitute.org/gene/58529+7136
PYGM vs TPM2: https://hb.flatironinstitute.org/gene/5837+7169
 -->
](images/coefs_comp/giant_networks/main.svg "GIANT network interaction"){#fig:giant_gene_pairs width="100%"}


Some of these genes, such as *APOC1* (19q13) and *CCL18* (17q12), were previously found to have an incomplete set of chemical, physical and biological features that correlated well with a lower number of publications [@pmid:30226837].
A gene co-expression analysis beyond linear patterns could shed light on the function of these understudied genes.

Both *IFNG* (12q15) and *SDS* (12q24)

Both *IFNG* (12q15) and *APOC1* (19q13) have been reported to be understudied protein-coding genes, with an incomplete set of measurable features that correlated well with a lower number of publications [@pmid:30226837].
Their potential link with more well-studied genes, such as *SDS* (12q24) and *JUN* (1q32), might provide clues about their function.

*NRADDP* is an RNA gene
AC068... put the ENS id here

- in clustermatch vs Spearman, the composite relationship with twolines are all those 28 pairs

first two pairs:
IFNG and APOC1 are understudied genes
SDS and JUN 

Genes *UTY* (chromosome Y) and *KDM6A* (chromosome X), which are paralogs, show a nonlinear relationship with a subset of samples (males) following a strong linear pattern, and another subset (females) having a constant expression of one gene (*UTY* is zero in this case, as expected).
This combination of linear and constant patterns is captured by Clustermatch ($c=0.29$) but not by Pearson and Spearman ($p=0.24$, $s=0.10$).
Clustermatch also correctly identifies this gene pair pattern in all other tissues in GTEx with the exception of female-specific organs (Supplementary Figures @fig:gtex_tissues:kdm6a_uty).
Another composite relationship is present for genes *RASSF2* (20p13) and *CYTIP* (2q24.1) which show two clear linear patterns.
These two genes are strongly expressed in white blood cells, both in myeloid and lymphoid lineages, and using tissue-specific gene networks from GIANT [@pmcid:PMC4828725] we found strong evidence of interactions in these cell types (Supplementary Figure ADD).

<!--
SDS / IFNG:
    https://hb.flatironinstitute.org/gene/10993+3458
    expressed in leukocyte, blood, lymphocyte, spleen
    serine seems to be essential for T cell expansion: https://www.cell.com/cell-metabolism/pdfExtended/S1550-4131(16)30644-1

JUN / APOC1:
    https://hb.flatironinstitute.org/gene/3725+341
    both expressed in blood, liver, leukocyte
    both seem to be very related to the regulation of immune response

BAG3 / NRADDP:
    maybe remove this because NRADDP is pseudo gene
        and it is not present in GIANT

UTY / KDM6A:
    these are from chr Y and X
    easy explanation

RASSF2 / CYTIP:
    https://hb.flatironinstitute.org/gene/9770+9595
    both expressed in peripheral blood (super strong), leukocyte (super strong), blood, etc etc (super strong)

AC068... / KLHL21:
    maybe remove? I can't fine AC068... gene

the other two gene pairs detected by Pearson do not interact much in GIANT
 -->
