### The CCC reveals linear and nonlinear patterns in human transcriptomic data

We used gene expression data from GTEx v8 across different tissues.
We selected the top 5,000 genes with the largest variance for our initial analyses on whole blood and then computed the pairwise matrix of correlation coefficients using Pearson, Spearman and Clustermatch (see Methods).

![
**Distribution of coefficient values on gene expression (GTEx v8, whole blood).**
**a)** Histogram of coefficient values.
**b)** Corresponding cumulative histogram. The dotted line maps the coefficient value that accumulates 70% of gene pairs.
**c)** 2D histogram plot with hexagonal bins between all coefficients, where a logarithmic scale was used to color each hexagon.
](images/coefs_comp/gtex_whole_blood/dist-main.svg "Distribution of coefficient values"){#fig:dist_coefs width="100%"}


In Figure @fig:dist_coefs a, we show how the pairwise correlation values distribute, where Clustermatch (mean=0.14, median=0.08, sd=0.15) has a much more skewed distribution than Pearson (mean=0.31, median=0.24, sd=0.24) and especially Spearman (mean=0.39, median=0.37, sd=0.26).
Coefficients reach 70% of gene pairs at $c=0.18$, $p=0.44$ and $s=0.56$ (Figure @fig:dist_coefs b).
Clustermatch and Spearman tend to agree more than any of these with Pearson, although many gene pairs seem to have a relatively higher correlation value only with Clustermatch (Figures @fig:dist_coefs c).


![
**Intersection of gene pairs with high and low correlation coefficient values (GTEx v8, whole blood).**
**a)** UpSet plot with six categories (rows) grouping the 30% of the highest (green triangle) and lowest (red triangle) correlation values for each method.
Columns show different intersections of categories grouped by agreements and disagreements.
**b)** Hexagonal binning plots with examples of gene pairs where Clustermatch ($c$) disagrees with Pearson ($p$) and Spearman ($s$).
For each method, green and red triangles indicate if the gene pair is among the top (green) or bottom (red) 30% of correlation values.
No triangle means that the correlation value for the gene pair is between the 30th and 70th percentiles (neither low nor high).
A logarithmic scale was used to color each hexagon.
](images/coefs_comp/gtex_whole_blood/upsetplot-main.svg "Intersection of gene pairs"){#fig:upsetplot_coefs width="100%"}


A closer inspection of gene pairs detected and missed by these coefficients revealed the ability of Clustermatch to capture more complex yet biologically meaningful patterns.<!-- this also isn't reading like objective reporting but more like advocacy-->
For this, we analyzed the agreements and disagreements by obtaining for each coefficient the top 30% of gene pairs with the largest correlation values ("high" set) and the bottom 30% ("low" set), resulting in six potentially overlapping categories.
An UpSet plot [@doi:10.1109/TVCG.2014.2346248] is shown in Figure @fig:upsetplot_coefs a, where the intersections of these six categories allowed to precisely identify the gene expression patterns captured and missed by each coefficient.
For most cases, the three coefficients agreed on whether there is a strong linear correlation (42.1%) and whether there is no relationship (34.3%).
This result is crucial because it implies that the user will not miss important linear patterns in expression data when using Clustermatch.
The figure also confirms that Clustermatch and Spearman agree more on highly correlated pairs (4.0% in "high", and 7.0% in "low") than any of these with Pearson (all between 0.3%-3.5% for "high", and 2.8%-5.5% for "low").
Regarding disagreements (right part of the figure), more than 20 thousand gene pairs (20,987) with a high Clustermatch value are not highly ranked by any other coefficients.
There are also gene pairs with a high Pearson value with either low Clustermatch (1,075) or low Clustermatch and low Spearman values (531).
However, these cases mostly seem to be driven by outliers (Figure @fig:upsetplot_coefs b).
Clustermatch does not miss gene pairs highly ranked by Spearman.


Figure @fig:upsetplot_coefs b shows individual gene pairs among the top five of each intersection category in the "Disagreements" group where Clustermatch disagrees with either Pearson, Spearman or both.
The first three gene pairs at the top (*IFNG* / *SDS*, *JUN* / *APOC1*, and *ZDHHC12* / *CCL18*), with a high Clustermatch and low Pearson coefficient, seem to follow a non-coexistence relationship: in samples where one of the genes is highly (slightly) expressed, the other is slightly (highly) activated, suggesting a potential inhibiting effect.
The next three gene pairs (*UTY* / *KDM6A*, *RASSF2* / *CYTIP*, and *AC068580.6* / *KLHL21*) follow patterns combining either two linear or one linear and one constant relationships.
In particular, genes *UTY* and *KDM6A*, paralogs, show a nonlinear relationship with a subset of samples following a robust linear pattern and another subset having a constant expression of one gene.
This relationship is explained by the fact that *UTY* is in chromosome Y (Yq11) whereas *KDM6A* is in chromosome X (Xp11), and samples with a linear pattern are males, whereas those with no expression for *UTY* are females.<!--opportunity to talk about clustermatch correlation coefficient as a way to probe sex as a biological variable-->
This combination of linear and constant patterns is captured by Clustermatch ($c=0.29$, above the 80th percentile) but not by Pearson ($p=0.24$, below the 55th percentile) or Spearman ($s=0.10$, below the 15th percentile).
Furthermore, the same gene pair pattern is highly ranked by Clustermatch in all other tissues in GTEx, except for female-specific organs (Figure @fig:gtex_tissues:kdm6a_uty).


![
**Scatter plots of genes *KDM6A* and *UTY* across different GTEx tissues.**
Clustermatch correctly captures the relationship in all GTEx tissues, and here we show nine of them in the first three rows.
The last row shows three female-specific organs, where Clustermatch correctly finds no association.
](images/coefs_comp/kdm6a_vs_uty/gtex-KDM6A_vs_UTY.svg "KDM6A and UTY across different GTEx tissues"){#fig:gtex_tissues:kdm6a_uty width="95%"}


To study the other gene pairs found by the correlation coefficients, we used tissue-specific gene networks from GIANT [@pmcid:PMC4828725], where nodes represent genes and each edge a functional relationship weighted with a probability of interaction between two genes.
GIANT networks were built from 987 genome-scale data sets across approximately 38,000 conditions, including expression and different interaction measurements such as gene co-expression (using Pearson correlation), protein-interaction, transcription factor regulation, and chemical and genetic perturbations and microRNA target profiles from the Molecular Signatures Database (MSigDB [@pmid:16199517]).
Figure @fig:giant_gene_pairs shows cell type-specific networks for each gene pair (Figure @fig:upsetplot_coefs b) for which genes are present in GIANT models.
Two large black nodes in each network's top-left and bottom-right corners represent our gene pairs.
A green edge means a close-to-zero probability of interaction, whereas a red edge represents a strong predicted relationship between two genes.
The tissue was automatically selected in each network according to the predicted tissue expression of gene pairs.
Interestingly, gene pairs highly ranked by Clustermatch are part of very cohesive networks and strongly related to blood.
For example, the average probability of gene connections with *IFNG* / *SDS* is very high, at least 0.79 for all other genes shown.
This minimum average with *JUN* / *APOC1* is 0.56, for *ZDHHC12* / *CCL18* is 0.34 (where *ZDHHC12* shows the weakest links although *CCL18* is strongly connected), and for *RASSF2* / *CYTIP* is 0.76.
Predicted networks for the two gene pairs prioritized by Pearson are much less cohesive, suggesting that the high correlation in whole blood is mostly driven by outliers.<!--this is interesting - do you have the pairwise GTEx plots to look at? it is kind of odd to me that PCC is finding things that aren't as aligned with GIANT since that is built with PCC, but that also gets discretized so outlier effects are minimized and consistency across many datasets matters more to get high scores than outliers in a few datasets..-->
For example, the minimum/maximum average of interaction probabilities with *MYOZ1* / *TNNI2* is only 0.11/0.12, and for *PYGM* / *TPM2* is 0.13/0.24.


![
**Predicted tissue-specific networks from GIANT for each gene pairs prioritized by correlation coefficients.**
A node represents a gene and an edge the probability that two genes are part of the same biological process in a specific cell type (indicated at the top-right corner).
A maximum of 15 genes are shown for each subfigure.
The GIANT web application automatically determined a minimum weight for edges to be shown.
These analyses can be performed online using the following links:
*IFNG* / *SDS* [@url:https://hb.flatironinstitute.org/gene/10993+3458];
*JUN* / *APOC1* [@url:https://hb.flatironinstitute.org/gene/3725+341]
*ZDHHC12* / *CCL18* [@url:https://hb.flatironinstitute.org/gene/6362+84885]
*RASSF2* / *CYTIP* [@url:https://hb.flatironinstitute.org/gene/9770+9595]
*MYOZ1* / *TNNI2* [@url:https://hb.flatironinstitute.org/gene/58529+7136]
*PYGM* / *TPM2* [@url:https://hb.flatironinstitute.org/gene/5837+7169]
](images/coefs_comp/giant_networks/main.svg "GIANT network interaction"){#fig:giant_gene_pairs width="100%"}


Some of these genes, such as *SDS* (12q24) and *ZDHHC12* (9q34), were previously found to have a relatively lower number of publications that correlated well with a small set of chemical, physical and biological features [@pmid:30226837].
A gene co-expression analysis on large compendia and beyond linear patterns could shed light on the function of understudied genes.
On the other hand, gene *KLHL21* (1p36) and the novel RNA gene *AC068580.6* (*ENSG00000235027*, in 11p15) have a high Clustermatch correlation and are entirely missed by the other methods.
*KLHL21* was suggested as a potential therapeutic target for hepatocellular carcinoma [@pmid:27769251] and other cancers [@pmid:29574153; @pmid:35084622], and its nonlinear correlation with *AC068580.6* and potentially other genes might unveil other important players in cancer initiation or progression.

<!--
MENTION THE IMPORTANCE OF CLUSTERMATCH TO CONTRUCT COHESIVE GENE CLUSTERS: https://humanbase.readthedocs.io/en/latest/modules.html
-->
