### The CCC reveals linear and nonlinear patterns in human transcriptomic data

We examined the characteristics of the correlation coefficients in gene expression data from GTEx v8 across different tissues.
We selected the top 5,000 genes with the highest variance and computed the correlation matrix between them using Pearson, Spearman, and CCC.
For more information on the data and methods used, please refer to the Methods section.


We examined the distribution of each coefficient's absolute values in GTEx (Figure @fig:dist_coefs).
CCC (mean=0.14, median=0.08, sd=0.15) had a much more skewed distribution than Pearson (mean=0.31, median=0.24, sd=0.24) and Spearman (mean=0.39, median=0.37, sd=0.26).
The coefficients reached a cumulative set containing 70% of gene pairs at different values (Figure @fig:dist_coefs b), $c=0.18$, $p=0.44$ and $s=0.56$, suggesting that for this type of data, the coefficients are not directly comparable by magnitude.
Therefore, we used ranks for further comparisons.
In GTEx v8, CCC values were closer to Spearman and vice versa than either was to Pearson (Figure @fig:dist_coefs c).
We also compared the Maximal Information Coefficient (MIC) in this data (see [Supplementary Note 1](#sec:mic)).
We found that CCC behaved very similarly to MIC, although CCC was up to two orders of magnitude faster to run (see [Supplementary Note 2](#sec:time_test)).
MIC is an advanced correlation coefficient able to capture general patterns beyond linear relationships and has been successfully used in various application domains [@pmid:33972855; @pmid:33001806; @pmid:27006077].
Our results suggest that CCC generalizes to MIC, so in the subsequent analyses we focus on CCC and linear-only coefficients.


![
**Distribution of coefficient values on gene expression (GTEx v8, whole blood).**
**a)** Histogram of coefficient values.
**b)** Corresponding cumulative histogram. The dotted line maps the coefficient value that accumulates 70% of gene pairs.
**c)** 2D histogram plot with hexagonal bins between all coefficients, where a logarithmic scale was used to color each hexagon.
](images/coefs_comp/gtex_whole_blood/dist-main.svg "Distribution of coefficient values"){#fig:dist_coefs width="100%"}


An analysis of gene pairs prioritized and disregarded by the correlation coefficients revealed different patterns.
To compare the agreements and disagreements, we obtained the top 30% of gene pairs with the largest correlation values ("high" set) and the bottom 30% ("low" set).
This resulted in six potentially overlapping categories.
UpSet analysis [@doi:10.1109/TVCG.2014.2346248] (Figure @fig:upsetplot_coefs a) showed that the three coefficients agreed on the presence of a strong correlation (42.1%) or the absence of a relationship (34.3%) in most cases (76.4%).
Since Pearson and Spearman are linear-only, and CCC can capture these patterns, we assume the concordant gene pairs represent clear linear patterns.
CCC and Spearman had more agreement on highly or poorly correlated pairs (4.0% in "high", and 7.0% in "low") than Pearson (all between 0.3%-3.5% for "high", and 2.8%-5.5% for "low").
Overall, CCC agreed with either Pearson or Spearman in 90.5% of gene pairs by assigning a high or a low correlation value.

![
**Intersection of gene pairs with high and low correlation coefficient values (GTEx v8, whole blood).**
**a)** UpSet plot with six categories (rows) grouping the 30% of the highest (green triangle) and lowest (red triangle) values for each coefficient.
Columns show different intersections of categories grouped by agreements and disagreements.
**b)** Hexagonal binning plots with examples of gene pairs where CCC ($c$) disagrees with Pearson ($p$) and Spearman ($s$).
For each method, colors in the triangles indicate if the gene pair is among the top (green) or bottom (red) 30% of coefficient values.
No triangle means that the correlation value for the gene pair is between the 30th and 70th percentiles (neither low nor high).
A logarithmic scale was used to color each hexagon.
](images/coefs_comp/gtex_whole_blood/upsetplot-main.svg "Intersection of gene pairs"){#fig:upsetplot_coefs width="100%"}


Our results showed that more than 20,000 gene pairs had a high CCC value but were not highly ranked by the other correlation coefficients (Figure @fig:upsetplot_coefs a, right).
Additionally, 1,075 gene pairs had a high Pearson value but low CCC values, 87 gene pairs had a high Spearman value but low CCC values, and 531 gene pairs had both low CCC and low Spearman values.
We believe that many of these cases were driven by potential outliers (Figure @fig:upsetplot_coefs b, and analyzed later).
To further investigate, we analyzed gene pairs among the top five of each intersection in the "Disagreements" group (Figure @fig:upsetplot_coefs a, right) where CCC disagreed with Pearson, Spearman or both.

![
**The expression levels of *KDM6A* and *UTY* display sex-specific associations across GTEx tissues.**
CCC captures this nonlinear relationship in all GTEx tissues (nine examples are shown in the first three rows), except in female-specific organs (last row).
](images/coefs_comp/kdm6a_vs_uty/gtex-KDM6A_vs_UTY-main.svg "KDM6A and UTY across different GTEx tissues"){#fig:gtex_tissues:kdm6a_uty width="95%"}

The first three gene pairs (*IFNG* - *SDS*, *JUN* - *APOC1*, and *ZDHHC12* - *CCL18*) had high CCC values and low Pearson values, suggesting a non-coexistence relationship: when one gene was highly expressed, the other was slightly activated, and vice versa.
The following three gene pairs (*UTY* - *KDM6A*, *RASSF2* - *CYTIP*, and *AC068580.6* - *KLHL21*) had patterns combining either two linear or one linear and one independent relationships.
For example, *UTY* and *KDM6A* (paralogs) had a nonlinear relationship, with a subset of samples following a linear pattern and another subset having a constant expression of one gene.
This was explained by the fact that *UTY* is in chromosome Y (Yq11) and *KDM6A* is in chromosome X (Xp11), and males had a linear pattern, whereas females had no expression for *UTY*.
This combination of linear and independent patterns was captured by CCC ($c=0.29$, above the 80th percentile) but not by Pearson ($p=0.24$, below the 55th percentile) or Spearman ($s=0.10$, below the 15th percentile).
Furthermore, the same gene pair pattern was highly ranked by CCC in all other tissues in GTEx, except for female-specific organs (Figure @fig:gtex_tissues:kdm6a_uty).
