### The CCC reveals linear and nonlinear patterns in human transcriptomic data

We analyzed the characteristics of correlation coefficients in gene expression data from GTEx v8 across different tissues.
We first selected the top 5,000 genes with the highest variance in whole blood and computed the correlation matrix between them using Pearson, Spearman, and CCC correlation coefficients (see [Methods](#sec:data_gtex)).


The distribution of the absolute values of each coefficient in GTEx (Figure @fig:dist_coefs) was examined.
CCC had a much more skewed distribution than Pearson and Spearman (mean=0.14, median=0.08, sd=0.15; mean=0.31, median=0.24, sd=0.24; mean=0.39, median=0.37, sd=0.26, respectively).
The coefficients reached a cumulative set containing 70% of gene pairs at different values (Figure @fig:dist_coefs b; $c=0.18$, $p=0.44$ and $s=0.56$).
This suggests that the coefficients are not directly comparable by magnitude, and therefore ranks were used for further comparisons.
In GTEx v8, CCC values were closer to Spearman and vice versa than either was to Pearson (Figure @fig:dist_coefs c).
The Maximal Information Coefficient (MIC) was also compared in this data (see [Supplementary Note 1](#sec:mic)).
It was found that CCC behaved similarly to MIC, although CCC was up to two orders of magnitude faster to run (see [Supplementary Note 2](#sec:time_test)).
MIC is an advanced correlation coefficient used in various application domains [@pmid:33972855; @pmid:33001806; @pmid:27006077], and is capable of capturing general patterns beyond linear relationships.
Therefore, CCC and linear-only coefficients were focused on in the subsequent analyses.


![
**Distribution of coefficient values on gene expression (GTEx v8, whole blood).**
**a)** Histogram of coefficient values.
**b)** Corresponding cumulative histogram. The dotted line maps the coefficient value that accumulates 70% of gene pairs.
**c)** 2D histogram plot with hexagonal bins between all coefficients, where a logarithmic scale was used to color each hexagon.
](images/coefs_comp/gtex_whole_blood/dist-main.svg "Distribution of coefficient values"){#fig:dist_coefs width="100%"}


We analyzed gene pairs that were prioritized or disregarded by the correlation coefficients.
We obtained the top 30% of gene pairs with the largest correlation values ("high" set) and the bottom 30% ("low" set).
We used an UpSet analysis to examine the agreements and disagreements among the coefficients.
The analysis showed that the three coefficients agreed on whether there is a strong correlation (42.1%) or no relationship (34.3%) in most cases (76.4%).
CCC and Spearman agreed more on either highly or poorly correlated pairs (4.0% in "high", and 7.0% in "low") than either with Pearson (ranging from 0.3%-3.5% for "high", and 2.8%-5.5% for "low").
Overall, CCC agreed with either Pearson or Spearman in 90.5% of gene pairs by assigning a high or a low correlation value.
This indicates that CCC can capture both linear and nonlinear patterns.

![
**Intersection of gene pairs with high and low correlation coefficient values (GTEx v8, whole blood).**
**a)** UpSet plot with six categories (rows) grouping the 30% of the highest (green triangle) and lowest (red triangle) values for each coefficient.
Columns show different intersections of categories grouped by agreements and disagreements.
**b)** Hexagonal binning plots with examples of gene pairs where CCC ($c$) disagrees with Pearson ($p$) and Spearman ($s$).
For each method, colors in the triangles indicate if the gene pair is among the top (green) or bottom (red) 30% of coefficient values.
No triangle means that the correlation value for the gene pair is between the 30th and 70th percentiles (neither low nor high).
A logarithmic scale was used to color each hexagon.
](images/coefs_comp/gtex_whole_blood/upsetplot-main.svg "Intersection of gene pairs"){#fig:upsetplot_coefs width="100%"}


Our analysis revealed that more than 20,000 gene pairs had a high CCC value but were not highly ranked by the other correlation coefficients.
We also found 1,075 gene pairs with a high Pearson value, but both low CCC and low Spearman values, and 531 gene pairs with both low CCC and low Spearman values.
However, our examination suggests that many of these cases may be driven by potential outliers.
We analyzed the top five gene pairs in the "Disagreements" group where CCC disagreed with Pearson, Spearman, or both.

![
**The expression levels of *KDM6A* and *UTY* display sex-specific associations across GTEx tissues.**
CCC captures this nonlinear relationship in all GTEx tissues (nine examples are shown in the first three rows), except in female-specific organs (last row).
](images/coefs_comp/kdm6a_vs_uty/gtex-KDM6A_vs_UTY-main.svg "KDM6A and UTY across different GTEx tissues"){#fig:gtex_tissues:kdm6a_uty width="95%"}

The first three gene pairs (*IFNG* - *SDS*, *JUN* - *APOC1*, and *ZDHHC12* - *CCL18*) showed a non-coexistence relationship: when one gene was highly expressed, the other was slightly activated, suggesting an inhibiting effect.
The next three gene pairs (*UTY* - *KDM6A*, *RASSF2* - *CYTIP*, and *AC068580.6* - *KLHL21*) had patterns combining linear and independent relationships.
For example, *UTY* and *KDM6A* (paralogs) had a nonlinear relationship, where some samples followed a linear pattern and others had a constant (independent) expression of one gene.
This combination was captured by CCC ($c=0.29$, above the 80th percentile) but not by Pearson ($p=0.24$, below the 55th percentile) or Spearman ($s=0.10$, below the 15th percentile).
Furthermore, the same gene pair pattern was highly ranked by CCC in all other tissues in GTEx, except for female-specific organs (Figure @fig:gtex_tissues:kdm6a_uty).
