### The CCC reveals linear and nonlinear patterns in human transcriptomic data

We next examined the characteristics of these correlation coefficients in gene expression data from GTEx v8 across different tissues.
We selected the top 5,000 genes with the largest variance for our initial analyses on whole blood and then computed the correlation matrix between genes using Pearson, Spearman and CCC (see [Methods](#sec:data_gtex)).


We examined the distribution of each coefficient's absolute values in GTEx (Figure @fig:dist_coefs).
CCC (mean=0.14, median=0.08, sd=0.15) had a much more skewed distribution than Pearson (mean=0.31, median=0.24, sd=0.24) and Spearman (mean=0.39, median=0.37, sd=0.26).
70% of gene pairs reached cumulative sets at different values (Figure @fig:dist_coefs b): $c=0.18$, $p=0.44$ and $s=0.56$.
This suggests that for this type of data, the coefficients are not directly comparable by magnitude, so we used ranks for further comparisons.
In GTEx v8, CCC values were closer to Spearman and vice versa than either was to Pearson (Figure @fig:dist_coefs c).
We compared the Maximal Information Coefficient (MIC) in this data (see [Supplementary Note 1](#sec:mic)).
CCC behaved similarly to MIC, although it was up to two orders of magnitude faster to run (see [Supplementary Note 2](#sec:time_test)).
MIC, an advanced correlation coefficient able to capture general patterns beyond linear relationships, has been successfully used in various application domains [@pmid:33972855; @pmid:33001806; @pmid:27006077].
These results suggest that our findings for CCC generalize to MIC.
We therefore focus on CCC and linear-only coefficients in the subsequent analyses.


![
**Distribution of coefficient values on gene expression (GTEx v8, whole blood).**
**a)** Histogram of coefficient values.
**b)** Corresponding cumulative histogram. The dotted line maps the coefficient value that accumulates 70% of gene pairs.
**c)** 2D histogram plot with hexagonal bins between all coefficients, where a logarithmic scale was used to color each hexagon.
](images/coefs_comp/gtex_whole_blood/dist-main.svg "Distribution of coefficient values"){#fig:dist_coefs width="100%"}


An UpSet analysis of the three correlation coefficients revealed that 76.4% of gene pairs were concordant in terms of whether there was a strong correlation or no relationship.
The analysis showed that 42.1% of gene pairs had strong correlation and 34.3% had no relationship.
CCC and Spearman agreed more on highly or poorly correlated gene pairs (4.0% in the "high" set and 7.0% in the "low" set) than Pearson (values ranging from 0.3%-3.5% in the "high" set and 2.8%-5.5% in the "low" set).
Overall, CCC agreed with either Pearson or Spearman in 90.5% of gene pairs by assigning a high or a low correlation value.

![
**Intersection of gene pairs with high and low correlation coefficient values (GTEx v8, whole blood).**
**a)** UpSet plot with six categories (rows) grouping the 30% of the highest (green triangle) and lowest (red triangle) values for each coefficient.
Columns show different intersections of categories grouped by agreements and disagreements.
**b)** Hexagonal binning plots with examples of gene pairs where CCC ($c$) disagrees with Pearson ($p$) and Spearman ($s$).
For each method, colors in the triangles indicate if the gene pair is among the top (green) or bottom (red) 30% of coefficient values.
No triangle means that the correlation value for the gene pair is between the 30th and 70th percentiles (neither low nor high).
A logarithmic scale was used to color each hexagon.
](images/coefs_comp/gtex_whole_blood/upsetplot-main.svg "Intersection of gene pairs"){#fig:upsetplot_coefs width="100%"}


Our correlation coefficient based on machine learning showed broad agreement with Pearson and Spearman coefficients, with more than 20,000 gene pairs having a high CCC value.
However, 1,075 gene pairs had a high Pearson value but low CCC, and 87 gene pairs had a high Spearman value but low CCC.
We found that 531 gene pairs had both low CCC and low Spearman values, which may have been driven by outliers.
We analyzed the top five gene pairs in the "Disagreements" group, where CCC disagreed with Pearson, Spearman or both, and found that CCC was more accurate than the other coefficients.

![
**The expression levels of *KDM6A* and *UTY* display sex-specific associations across GTEx tissues.**
CCC captures this nonlinear relationship in all GTEx tissues (nine examples are shown in the first three rows), except in female-specific organs (last row).
](images/coefs_comp/kdm6a_vs_uty/gtex-KDM6A_vs_UTY-main.svg "KDM6A and UTY across different GTEx tissues"){#fig:gtex_tissues:kdm6a_uty width="95%"}

The three gene pairs (*IFNG* - *SDS*, *JUN* - *APOC1*, and *ZDHHC12* - *CCL18*) showed a non-coexistence relationship, suggesting a potentially inhibiting effect.
*UTY* and *KDM6A* (paralogs) followed a nonlinear relationship, with a subset of samples following a linear pattern and another subset having a constant expression of one gene.
This combination of linear and independent patterns was captured by CCC ($c=0.29$, above the 80th percentile) but not by Pearson ($p=0.24$, below the 55th percentile) or Spearman ($s=0.10$, below the 15th percentile).
This same gene pair pattern was highly ranked by CCC in all other tissues in GTEx, except for female-specific organs (Figure @fig:gtex_tissues:kdm6a_uty).
