### The CCC reveals linear and nonlinear patterns in human transcriptomic data

We next examined the characteristics of these correlation coefficients in gene expression data from GTEx v8 across different tissues.
For our initial analyses, we selected the top 5,000 genes with the largest variance on whole blood and then computed the correlation matrix between genes using Pearson, Spearman and CCC (see [Methods](#sec:data_gtex)).
Although we always considered the statistical significance of the coefficients, we focused on the strength of the association (i.e., the coefficient value) for our analyses.


We examined the distribution of each coefficient's absolute values in GTEx (Figure @fig:dist_coefs).
CCC (mean=0.14, median=0.08, sd=0.15) has a much more skewed distribution than Pearson (mean=0.31, median=0.24, sd=0.24) and Spearman (mean=0.39, median=0.37, sd=0.26).
The coefficients reach a cumulative set containing 70% of gene pairs at different values (Figure @{fig:dist_coefs}b), $c=0.18$, $p=0.44$ and $s=0.56$, suggesting that for this type of data, the coefficients are not directly comparable by magnitude, so we used ranks for further comparisons.
In GTEx v8, CCC values were closer to Spearman than either was to Pearson (Figure @{fig:dist_coefs}c).
As shown later, we also compared with the Maximal Information Coefficient (MIC), another advanced, not-only-linear correlation coefficient that has been successfully used in various application domains [@pmid:33972855; @pmid:33001806; @pmid:27006077].
We found that CCC behaved very similarly to MIC, although CCC was up to two orders of magnitude faster to run.
These results suggest that our findings for CCC generalize to MIC, therefore, in the subsequent analyses we focus on CCC and linear-only coefficients.


![
**Distribution of coefficient values on gene expression (GTEx v8, whole blood).**
**a)** Histogram of coefficient values.
**b)** Corresponding cumulative histogram. The dotted line maps the coefficient value that accumulates 70% of gene pairs.
**c)** 2D histogram plot with hexagonal bins between all coefficients, where a logarithmic scale was used to color each hexagon.
](images/coefs_comp/gtex_whole_blood/dist-main.svg "Distribution of coefficient values"){#fig:dist_coefs width="100%"}


A closer inspection of gene pairs that were either prioritized or disregarded by these coefficients revealed that they captured different patterns.
We analyzed the agreements and disagreements by obtaining, for each coefficient, the top 30% of gene pairs with the largest correlation values ("high" set) and the bottom 30% ("low" set), resulting in six potentially overlapping categories (Supplementary Files 1 and 2).
For most cases (76.4%), an UpSet analysis [@doi:10.1109/TVCG.2014.2346248] (Figure @{fig:upsetplot_coefs}a) showed that the three coefficients agreed on whether there is a strong correlation (42.1%) or there is no relationship (34.3%).
Since Pearson and Spearman are linear-only, and CCC can also capture these patterns, we expect that these concordant gene pairs represent clear linear patterns.
CCC and Spearman agree more on either highly or poorly correlated pairs (4.0% in "high", and 7.0% in "low") than any of these with Pearson (all between 0.3%-3.5% for "high", and 2.8%-5.5% for "low").
In summary, CCC agrees with either Pearson or Spearman in 90.5% of gene pairs by assigning a high or a low correlation value.

![
**Intersection of gene pairs with high and low correlation coefficient values (GTEx v8, whole blood).**
**a)** UpSet plot with six categories (rows) grouping the 30% of the highest (green triangle) and lowest (red triangle) values for each coefficient.
Columns show different intersections of categories grouped by agreements and disagreements.
**b)** Hexagonal binning plots with examples of gene pairs where CCC ($c$) disagrees with Pearson ($p$) and Spearman ($s$).
For each method, colors in the triangles indicate if the gene pair is among the top (green) or bottom (red) 30% of coefficient values.
No triangle means that the correlation value for the gene pair is between the 30th and 70th percentiles (neither low nor high).
The statistical significance is indicated with asterisks using the False Discovery Rate (FDR) adjusted $P$-values, calculated using the Benjamini and Hochberg method [@doi:10.1111/j.2517-6161.1995.tb02031.x]: $\mathrm{FDR}<0.05$ (\*), $\mathrm{FDR}<0.01$ (\*\*), and $\mathrm{FDR}<0.001$ (\*\*\*).
A logarithmic scale was used to color each hexagon.
](images/coefs_comp/gtex_whole_blood/upsetplot-main.svg "Intersection of gene pairs"){#fig:upsetplot_coefs width="100%"}


While there was broad agreement, more than 20,000 gene pairs with a high CCC value were not highly ranked by the other coefficients ("Disagreements" group on the right of Figure @{fig:upsetplot_coefs}a).
There were also gene pairs with a high Pearson value and either low CCC (1,075), low Spearman (87) or both low CCC and low Spearman values (531).
No gene pairs were found to have a high Spearman value and a low CCC.
Considering the correlation values and their statistical significance, we analyzed gene pairs among the top ten of each intersection in the "Disagreements" group (Figure @{fig:upsetplot_coefs}a, right) where CCC disagrees with Pearson, Spearman or both.

![
**The expression levels of *KDM6A* and *UTY* display sex-specific associations across GTEx tissues.**
CCC captures this nonlinear relationship in all GTEx tissues (nine examples are shown in the first three rows), except in female-specific organs (last row).
](images/coefs_comp/kdm6a_vs_uty/gtex-KDM6A_vs_UTY-main.svg "KDM6A and UTY across different GTEx tissues"){#fig:gtex_tissues:kdm6a_uty width="95%"}


The first two gene pairs at the top of Figure @{fig:upsetplot_coefs}b (*IFNG* - *SDS*, with high CCC and Spearman, and low Pearson; *PRSS36* - *CCL18*, with high CCC and low Pearson) appear to follow a non-coexistence relationship: in samples where one of the genes is highly expressed, the other is slightly activated, suggesting a potentially inhibiting effect.
The following four gene pairs (*UTY* - *KDM6A*, *DDX3Y* - *KDM6A*, *RASSF2* - *CYTIP*, and *AC068580.6* - *KLHL21*) follow patterns combining either two linear or one linear and one independent relationships.
In particular, genes *UTY* - *KDM6A* (paralogs) and *DDX3Y* - *KDM6A* show a nonlinear relationship where a subset of samples follows a robust linear pattern and another subset has a constant (independent) expression of one gene.
The relationships in these two gene pairs are explained by sex differences in expression: *UTY* and *DDX3Y* are in chromosome Y (Yq11) whereas *KDM6A* is in chromosome X (Xp11), and therefore samples with a linear pattern are males, whereas those with no expression for *UTY* or *DDX3Y* are females.
Furthermore, for this sex-specific gene pair pattern, CCC yields a statistically significant coefficient value in 45 out of 47 tissues in GTEx, except for female-specific organs (Figure @fig:gtex_tissues:kdm6a_uty and @fig:gtex_tissues:kdm6a_ddx3y, and Supplementary File 3).
The gene pair *RASSF2* - *CYTIP* was replicated in an independent dataset as we explain later.
Even though we have not found a biological explanation for gene pair *AC068580.6* - *KLHL21* (there is limited information about *AC068580.6*, ENSG00000235027, a long non-coding RNA), its strong nonlinear connection with *KLHL21* (linked with some cancers [@pmid:27769251]) is robustly captured by CCC only.
Notably, these four gene pairs contain strong linear relationships and CCC is the only coefficient able to consistently capture these nonlinear patterns across a variety of tissues with a statistically significant and high correlation value.
Pearson and Spearman show a statistically significant correlation value for some of these gene pairs, although these values are low and would very likely not be prioritized for further research.
In addition, these two linear-only coefficients are unable to robustly capture the same pattern in other tissues (Figure @fig:gtex_tissues:kdm6a_uty and @fig:gtex_tissues:kdm6a_ddx3y, and Supplementary File 3).
For instance, although the three coefficients are statistically significant in whole blood for the gene pair *UTY* - *KDM6A*, Pearson and Spearman fail to capture the same pattern in the brain cerebellum, and in many cases, such as small intestine, the sign of the coefficient is negative despite the strong positive linear correlation among male samples (Figure @fig:gtex_tissues:kdm6a_uty).


Finally, the last two gene pairs in Figure @{fig:upsetplot_coefs}b are highly ranked by Pearson, but not by CCC or Spearman.
Although all coefficients are significant for the gene pair *MYOZ1* - *TNNI2*, the low CCC ($c=0.03$) and moderate Spearman ($s=0.28$) contrast with Pearson's ($p=0.97$), suggesting a statistically significant but very weak linear relationship.
The high and statistically significant Pearson value for *SCGB3A1* - *C19orf33* seems to be driven by outliers.
