### The CCC reveals linear and nonlinear patterns in human transcriptomic data

We examined the characteristics of the correlation coefficients using gene expression data from GTEx v8, across different tissues.
Initially, we selected the top 5,000 genes with the highest variance from whole blood and computed the correlation matrix between the genes using Pearson, Spearman, and CCC (see [Methods](#sec:data_gtex)).


We examined the distribution of each coefficient's absolute values in GTEx (Figure @fig:dist_coefs).
CCC (mean=0.14, median=0.08, sd=0.15) had a much more skewed distribution than Pearson (mean=0.31, median=0.24, sd=0.24) and Spearman (mean=0.39, median=0.37, sd=0.26).
The coefficients reached a cumulative set containing 70% of gene pairs at different values (Figure @fig:dist_coefs b), $c=0.18$, $p=0.44$ and $s=0.56$, indicating that for this type of data, the coefficients are not directly comparable by magnitude.
Therefore, we used ranks for further comparisons.
In GTEx v8, CCC values were closer to Spearman and vice versa than either was to Pearson (Figure @fig:dist_coefs c).
We also compared the Maximal Information Coefficient (MIC) in this data (see [Supplementary Note 1](#sec:mic)).
We found that CCC behaved very similarly to MIC, although CCC was up to two orders of magnitude faster to run (see [Supplementary Note 2](#sec:time_test)).
MIC is an advanced correlation coefficient able to capture general patterns beyond linear relationships, and it has been successfully used in various application domains [@pmid:33972855; @pmid:33001806; @pmid:27006077].
These results suggest that our findings for CCC generalize to MIC, so in the subsequent analyses we focused on CCC and linear-only coefficients.


![
**Distribution of coefficient values on gene expression (GTEx v8, whole blood).**
**a)** Histogram of coefficient values.
**b)** Corresponding cumulative histogram. The dotted line maps the coefficient value that accumulates 70% of gene pairs.
**c)** 2D histogram plot with hexagonal bins between all coefficients, where a logarithmic scale was used to color each hexagon.
](images/coefs_comp/gtex_whole_blood/dist-main.svg "Distribution of coefficient values"){#fig:dist_coefs width="100%"}


A closer inspection of gene pairs prioritized or disregarded by the correlation coefficients revealed different patterns.
We analyzed the agreements and disagreements by obtaining, for each coefficient, the top 30% of gene pairs with the largest correlation values (the "high" set) and the bottom 30% (the "low" set), resulting in six potentially overlapping categories.
An UpSet analysis (Figure @fig:upsetplot_coefs a) showed that, for most cases (76.4%), the three coefficients agreed on whether there is a strong correlation (42.1%) or no relationship (34.3%).
Since Pearson and Spearman are linear-only, and CCC can also capture these patterns, it is likely that these concordant gene pairs represent clear linear patterns.
CCC and Spearman agreed more on either highly or poorly correlated pairs (4.0% in the "high" set and 7.0% in the "low" set) than either did with Pearson (all between 0.3%-3.5% for the "high" set and 2.8%-5.5% for the "low" set).
In summary, CCC agreed with either Pearson or Spearman in 90.5% of gene pairs by assigning a high or a low correlation value.

![
**Intersection of gene pairs with high and low correlation coefficient values (GTEx v8, whole blood).**
**a)** UpSet plot with six categories (rows) grouping the 30% of the highest (green triangle) and lowest (red triangle) values for each coefficient.
Columns show different intersections of categories grouped by agreements and disagreements.
**b)** Hexagonal binning plots with examples of gene pairs where CCC ($c$) disagrees with Pearson ($p$) and Spearman ($s$).
For each method, colors in the triangles indicate if the gene pair is among the top (green) or bottom (red) 30% of coefficient values.
No triangle means that the correlation value for the gene pair is between the 30th and 70th percentiles (neither low nor high).
A logarithmic scale was used to color each hexagon.
](images/coefs_comp/gtex_whole_blood/upsetplot-main.svg "Intersection of gene pairs"){#fig:upsetplot_coefs width="100%"}


We found that more than 20,000 gene pairs had a high CCC value, but were not highly ranked by the other correlation coefficients (right part of Figure @fig:upsetplot_coefs a).
Additionally, 1,075 gene pairs had a high Pearson value but low CCC, 87 had a high Spearman value but low CCC, and 531 had both low CCC and low Spearman values.
We believe that many of these cases may be due to potential outliers (Figure @fig:upsetplot_coefs b).
We then analyzed the top five gene pairs from each intersection in the "Disagreements" group (Figure @fig:upsetplot_coefs a, right) where CCC disagreed with Pearson, Spearman or both.

![
**The expression levels of *KDM6A* and *UTY* display sex-specific associations across GTEx tissues.**
CCC captures this nonlinear relationship in all GTEx tissues (nine examples are shown in the first three rows), except in female-specific organs (last row).
](images/coefs_comp/kdm6a_vs_uty/gtex-KDM6A_vs_UTY-main.svg "KDM6A and UTY across different GTEx tissues"){#fig:gtex_tissues:kdm6a_uty width="95%"}

The first three gene pairs (*IFNG* - *SDS*, *JUN* - *APOC1*, and *ZDHHC12* - *CCL18*) show a non-coexistence relationship, where one gene is highly expressed while the other is slightly activated, suggesting an inhibiting effect.
The next three gene pairs (*UTY* - *KDM6A*, *RASSF2* - *CYTIP*, and *AC068580.6* - *KLHL21*) combine linear and independent relationships.
For example, *UTY* and *KDM6A* (paralogs) show a nonlinear relationship, where a subset of samples follows a linear pattern while another subset has a constant expression of one gene.
This is because *UTY* is in chromosome Y (Yq11) and *KDM6A* is in chromosome X (Xp11), and samples with a linear pattern are males, whereas those with no expression for *UTY* are females.
The CCC ($c=0.29$) captures this combination of linear and independent patterns, whereas Pearson ($p=0.24$) and Spearman ($s=0.10$) do not.
Furthermore, this gene pair pattern is highly ranked by CCC in all other tissues in GTEx, except for female-specific organs (Figure @fig:gtex_tissues:kdm6a_uty).
