### The CCC reveals linear and nonlinear patterns in human transcriptomic data

We next examined the characteristics of these correlation coefficients in gene expression data from GTEx v8 across different tissues.
We selected the top 5,000 genes with the largest variance for our initial analyses on whole blood and then computed the correlation matrix between genes using Pearson, Spearman and CCC (see [Methods](#sec:data_gtex)).


In our examination of the distribution of each coefficient's absolute values in the GTEx dataset (Figure @fig:dist_coefs), we found that CCC (mean=0.14, median=0.08, sd=0.15) had a much more skewed distribution than Pearson (mean=0.31, median=0.24, sd=0.24) and Spearman (mean=0.39, median=0.37, sd=0.26).
Furthermore, the coefficients reached a cumulative set containing 70% of gene pairs at different values (Figure @fig:dist_coefs b), $c=0.18$, $p=0.44$ and $s=0.56$, suggesting that they are not directly comparable by magnitude.
Thus, we used ranks for further comparisons.
In GTEx v8, CCC values were closer to Spearman and vice versa than either was to Pearson (Figure @fig:dist_coefs c).
Additionally, we compared the Maximal Information Coefficient (MIC) in this data (see [Supplementary Note 1](#sec:mic)) and found that CCC behaved very similarly to MIC, although CCC was up to two orders of magnitude faster to run (see [Supplementary Note 2](#sec:time_test)).
Since MIC is an advanced correlation coefficient, which is able to capture general patterns beyond linear relationships and has been successfully used in various application domains [@pmid:33972855; @pmid:33001806; @pmid:27006077], our findings for CCC may generalize to MIC.
Therefore, in the subsequent analyses, we focus on CCC and linear-only coefficients.


![
**Distribution of coefficient values on gene expression (GTEx v8, whole blood).**
**a)** Histogram of coefficient values.
**b)** Corresponding cumulative histogram. The dotted line maps the coefficient value that accumulates 70% of gene pairs.
**c)** 2D histogram plot with hexagonal bins between all coefficients, where a logarithmic scale was used to color each hexagon.
](images/coefs_comp/gtex_whole_blood/dist-main.svg "Distribution of coefficient values"){#fig:dist_coefs width="100%"}


We performed an UpSet analysis [@doi:10.1109/TVCG.2014.2346248] to analyze the agreements and disagreements between the three correlation coefficients (Figure @fig:upsetplot_coefs a).
The analysis revealed that the coefficients agreed on whether there is a strong correlation (42.1%) or no relationship (34.3%) in most cases (76.4%).
Since Pearson and Spearman are linear-only, we expect that these concordant gene pairs represent clear linear patterns.
CCC and Spearman agreed more on either highly or poorly correlated pairs (4.0% in the "high" set, and 7.0% in the "low" set) than either of them with Pearson (all between 0.3%-3.5% for the "high" set, and 2.8%-5.5% for the "low" set).
In summary, CCC agreed with either Pearson or Spearman in 90.5% of gene pairs by assigning a high or a low correlation value.

![
**Intersection of gene pairs with high and low correlation coefficient values (GTEx v8, whole blood).**
**a)** UpSet plot with six categories (rows) grouping the 30% of the highest (green triangle) and lowest (red triangle) values for each coefficient.
Columns show different intersections of categories grouped by agreements and disagreements.
**b)** Hexagonal binning plots with examples of gene pairs where CCC ($c$) disagrees with Pearson ($p$) and Spearman ($s$).
For each method, colors in the triangles indicate if the gene pair is among the top (green) or bottom (red) 30% of coefficient values.
No triangle means that the correlation value for the gene pair is between the 30th and 70th percentiles (neither low nor high).
A logarithmic scale was used to color each hexagon.
](images/coefs_comp/gtex_whole_blood/upsetplot-main.svg "Intersection of gene pairs"){#fig:upsetplot_coefs width="100%"}


Our analysis revealed that there were more than 20,000 gene pairs with a high CCC value which were not highly ranked by the other correlation coefficients (Figure @fig:upsetplot_coefs a, right).
Additionally, we found 1,075 gene pairs with a high Pearson value and either low CCC or low Spearman values, as well as 531 gene pairs with both low CCC and low Spearman values.
We believe that many of these cases may be driven by potential outliers (Figure @fig:upsetplot_coefs b, and analyzed later).
We further examined the top five gene pairs in the "Disagreements" group (Figure @fig:upsetplot_coefs a, right) where CCC disagreed with Pearson, Spearman or both.

![
**The expression levels of *KDM6A* and *UTY* display sex-specific associations across GTEx tissues.**
CCC captures this nonlinear relationship in all GTEx tissues (nine examples are shown in the first three rows), except in female-specific organs (last row).
](images/coefs_comp/kdm6a_vs_uty/gtex-KDM6A_vs_UTY-main.svg "KDM6A and UTY across different GTEx tissues"){#fig:gtex_tissues:kdm6a_uty width="95%"}

The first three gene pairs (*IFNG* - *SDS*, *JUN* - *APOC1*, and *ZDHHC12* - *CCL18*) had high CCC and low Pearson values, indicating a non-coexistence relationship.
In these cases, samples with high expression of one gene were associated with low expression of the other, suggesting a potential inhibiting effect.
The following three gene pairs (*UTY* - *KDM6A*, *RASSF2* - *CYTIP*, and *AC068580.6* - *KLHL21*) showed patterns combining either two linear or one linear and one independent relationships.
For example, *UTY* and *KDM6A* (paralogs) had a nonlinear relationship in which a subset of samples followed a linear pattern and another subset had a constant (independent) expression of one gene.
This was explained by the fact that *UTY* is on chromosome Y, while *KDM6A* is on chromosome X; samples with a linear pattern were males, and those with no expression for *UTY* were females.
CCC ($c=0.29$, above the 80th percentile) captured this combination of linear and independent patterns, while Pearson ($p=0.24$, below the 55th percentile) and Spearman ($s=0.10$, below the 15th percentile) did not.
Additionally, CCC highly ranked this gene pair pattern in all other tissues in GTEx, except for female-specific organs (Figure @fig:gtex_tissues:kdm6a_uty).
