### Replication of gene associations using tissue-specific gene networks from GIANT

We sought to assess whether associations between gene expression levels in GTEx datasets were replicated in other datasets.
This is challenging because linear-only correlation coefficients are typically used in gene co-expression analyses.
To this end, we used 144 tissue-specific gene networks from the Genome-wide Analysis of gene Networks in Tissues (GIANT) [@pmcid:PMC4828725; @url:https://hb.flatironinstitute.org], which did not include GTEx samples [@url:https://hb.flatironinstitute.org/data].
These networks were built from expression and different interaction measurements, including protein-interaction, transcription factor regulation, chemical/genetic perturbations and microRNA target profiles from the Molecular Signatures Database (MSigDB [@pmid:16199517]).
We reasoned that highly-ranked gene pairs using three different correlation coefficients in a single tissue (whole blood in GTEx) should often replicate in a corresponding tissue or related cell lineage using the multi-cell type functional interaction networks in GIANT.
To test this, we used the GIANT web application to predict a network with interactions for two gene pairs and to automatically detect a relevant tissue or cell type where the genes were predicted to be expressed. 

For example, we obtained the networks in blood and the automatically-predicted cell type for gene pairs *RASSF2* - *CYTIP* (CCC high, Figure @fig:giant_gene_pairs a) and *MYOZ1* - *TNNI2* (Pearson high, Figure @fig:giant_gene_pairs b).
The networks included other genes connected according to their probability of interaction, and two large black nodes in each network's top-left and bottom-right corners represented our gene pairs.
For *RASSF2* and *CYTIP*, with a high CCC value ($c=0.20$, above the 73th percentile) and low Pearson and Spearman ($p=0.16$ and $s=0.11$, below the 38th and 17th percentiles, respectively), the autodetected cell type was leukocytes, and interaction scores were similar to the blood network.
However, for genes *MYOZ1* and *TNNI2*, with a very high Pearson value ($p=0.97$), moderate Spearman ($s=0.28$) and very low CCC ($c=0.03$), the autodetected cell type (skeletal muscle) was not related to blood or one of its cell lineages.
These results suggested that CCC might be capturing blood-specific patterns missed by the other coefficients.

![
**Analysis of GIANT tissue-specific predicted networks for gene pairs prioritized by correlation coefficients.**
**a-b)** Two gene pairs prioritized by correlation coefficients (from Figure @fig:upsetplot_coefs b) with their predicted networks in blood (left) and an automatically selected tissue/cell type (right) using the method described in [@doi:10.1101/gr.155697.113].
A node represents a gene and an edge the probability that two genes are part of the same biological process in a specific cell type.
A maximum of 15 genes are shown for each network.
The GIANT web application automatically determined a minimum interaction confidence (edges' weights) to be shown.
These networks can be analyzed online using the following links:
*RASSF2* - *CYTIP* [@url:https://hb.flatironinstitute.org/gene/9770+9595],
*MYOZ1* - *TNNI2* [@url:https://hb.flatironinstitute.org/gene/58529+7136].
**c)** Summary of predicted tissue/cell type networks for gene pairs exclusively prioritized by CCC and Pearson.
The first row combines all gene pairs where CCC is high and Pearson or Spearman are low.
The second row combines all gene pairs where Pearson is high and CCC or Spearman are low.
Bar plots (left) show the number of gene pairs for each predicted tissue/cell type.
Box plots (right) show the average probability of interaction between genes in these predicted tissue-specific networks.
Red indicates CCC-only tissues/cell types, blue are Pearson-only, and purple are shared.
](images/coefs_comp/giant_networks/top_gene_pairs-main.svg "GIANT network interaction on gene pairs"){#fig:giant_gene_pairs width="100%"}


We evaluated the top 100 discrepant gene pairs between CCC and the other two coefficients using GIANT to autodetect relevant cell types.
The top 5 predicted cell types for gene pairs highly ranked by CCC were all blood-specific, including macrophage, leukocyte, natural killer cell, blood, and mononuclear phagocyte.
The average probability of interaction between genes in these CCC-ranked networks was significantly higher than the other coefficients, with all medians larger than 67% and first quartiles above 41%.
In contrast, most Pearson's gene pairs were predicted to be specific to tissues unrelated to blood, with skeletal muscle being the most commonly predicted tissue.
The interaction probabilities in these Pearson-ranked networks were generally lower than in CCC, except for blood-specific gene pairs.
This suggests that CCC-ranked gene pairs not only had high probabilities of belonging to the same biological process but were also predicted to be specifically expressed in blood cell lineages, while most Pearson-ranked gene pairs were not predicted to be blood-specific and had relatively low interaction probabilities.
This lack of replication in GIANT is consistent with our earlier observations of outlier-driven associations.
