### The CCC reveals linear and nonlinear patterns in human transcriptomic data

We observed a clear increase in correlation coefficients for CCC compared to Pearson and Spearman, suggesting that CCC is better at capturing nonlinear relationships than traditional linear methods.

We next examined the characteristics of these correlation coefficients in gene expression data from GTEx v8 across different tissues.
We selected the top 5,000 genes with the largest variance for our initial analyses.
We then computed the correlation matrix between these genes using Pearson, Spearman, and CCC (see [Methods](#sec:data_gtex)).
Our results showed a clear increase in correlation coefficients when using CCC compared to Pearson and Spearman.
This suggests that CCC is better at capturing nonlinear relationships than traditional linear methods.


We examined the distribution of each coefficient's absolute values in GTEx (see Figure @fig:dist_coefs).
CCC (mean=0.14, median=0.08, sd=0.15) had a much more skewed distribution than Pearson (mean=0.31, median=0.24, sd=0.24) and Spearman (mean=0.39, median=0.37, sd=0.26).
The coefficients reached a cumulative set containing 70% of gene pairs at different values (Figure @fig:dist_coefs b), $c=0.18$, $p=0.44$ and $s=0.56$.
This suggests that for this type of data, the coefficients are not directly comparable by magnitude, so we used ranks for further comparisons.
In GTEx v8, CCC values were closer to Spearman and vice versa than either was to Pearson (Figure @fig:dist_coefs c). 

We also compared the Maximal Information Coefficient (MIC) in this data (see [Supplementary Note 1](#sec:mic)).
We found that CCC behaved very similarly to MIC, although CCC was up to two orders of magnitude faster to run (see [Supplementary Note 2](#sec:time_test)).
MIC is an advanced correlation coefficient that can capture general patterns beyond linear relationships and has been used in various application domains [@pmid:33972855; @pmid:33001806; @pmid:27006077].
Our results suggest that our findings for CCC generalize to MIC.
Therefore, in the subsequent analyses we focus on CCC and linear-only coefficients.


![
**Distribution of coefficient values on gene expression (GTEx v8, whole blood).**
**a)** Histogram of coefficient values.
**b)** Corresponding cumulative histogram. The dotted line maps the coefficient value that accumulates 70% of gene pairs.
**c)** 2D histogram plot with hexagonal bins between all coefficients, where a logarithmic scale was used to color each hexagon.
](images/coefs_comp/gtex_whole_blood/dist-main.svg "Distribution of coefficient values"){#fig:dist_coefs width="100%"}


We conducted a closer inspection of gene pairs prioritized or disregarded by these coefficients.
Our analysis revealed six potentially overlapping categories.
We obtained, for each coefficient, the top 30% of gene pairs with the largest correlation values ("high" set) and the bottom 30% ("low" set). 

An UpSet analysis [@doi:10.1109/TVCG.2014.2346248] (Figure @fig:upsetplot_coefs a) showed that, for most cases (76.4%), the three coefficients agreed on whether there is a strong correlation (42.1%) or no relationship (34.3%).
Since Pearson and Spearman are linear-only, and CCC can also capture these patterns, we expect that these concordant gene pairs represent clear linear patterns. 

CCC and Spearman agree more on either highly or poorly correlated pairs (4.0% in "high", and 7.0% in "low") than either of them with Pearson (all between 0.3%-3.5% for "high", and 2.8%-5.5% for "low").
In summary, CCC agrees with either Pearson or Spearman in 90.5% of gene pairs by assigning a high or a low correlation value.

![
**Intersection of gene pairs with high and low correlation coefficient values (GTEx v8, whole blood).**
**a)** UpSet plot with six categories (rows) grouping the 30% of the highest (green triangle) and lowest (red triangle) values for each coefficient.
Columns show different intersections of categories grouped by agreements and disagreements.
**b)** Hexagonal binning plots with examples of gene pairs where CCC ($c$) disagrees with Pearson ($p$) and Spearman ($s$).
For each method, colors in the triangles indicate if the gene pair is among the top (green) or bottom (red) 30% of coefficient values.
No triangle means that the correlation value for the gene pair is between the 30th and 70th percentiles (neither low nor high).
A logarithmic scale was used to color each hexagon.
](images/coefs_comp/gtex_whole_blood/upsetplot-main.svg "Intersection of gene pairs"){#fig:upsetplot_coefs width="100%"}


There was broad agreement between the correlation coefficients, with more than 20,000 gene pairs having a high CCC value.
However, there were also gene pairs with a high Pearson value and either low CCC (1,075), low Spearman (87) or both low CCC and low Spearman values (531).
Our analysis suggests that many of these cases may be driven by potential outliers (Figure @fig:upsetplot_coefs b).
We analyzed gene pairs among the top five of each intersection in the "Disagreements" group (Figure @fig:upsetplot_coefs a, right) where CCC disagreed with Pearson, Spearman or both.

![
**The expression levels of *KDM6A* and *UTY* display sex-specific associations across GTEx tissues.**
CCC captures this nonlinear relationship in all GTEx tissues (nine examples are shown in the first three rows), except in female-specific organs (last row).
](images/coefs_comp/kdm6a_vs_uty/gtex-KDM6A_vs_UTY-main.svg "KDM6A and UTY across different GTEx tissues"){#fig:gtex_tissues:kdm6a_uty width="95%"}

The first three gene pairs (*IFNG* - *SDS*, *JUN* - *APOC1*, and *ZDHHC12* - *CCL18*) appear to follow a non-coexistence relationship, suggesting a potentially inhibiting effect.
In samples where one of the genes is highly (slightly) expressed, the other is slightly (highly) activated.
The following three gene pairs (*UTY* - *KDM6A*, *RASSF2* - *CYTIP*, and *AC068580.6* - *KLHL21*) present a combination of two linear or one linear and one independent relationships.
For instance, *UTY* and *KDM6A* (paralogs) have a nonlinear relationship in which a subset of samples follows a linear pattern and another subset has a constant expression of one gene.
This is explained by the fact that *UTY* is in chromosome Y (Yq11), while *KDM6A* is in chromosome X (Xp11).
The linear pattern is observed in males, while females have no expression for *UTY*.
The CCC value ($c=0.29$) for this gene pair is above the 80th percentile, while the Pearson ($p=0.24$) and Spearman ($s=0.10$) values are below the 55th and 15th percentiles, respectively.
Furthermore, this gene pair pattern is highly ranked by CCC in all other tissues in GTEx, except for female-specific organs (Figure @fig:gtex_tissues:kdm6a_uty).
