### Clustermatch detects linear and nonlinear patterns in human transcriptomic data

![
**Distribution of coefficient values on gene expression (GTEx v8, whole blood).**
**a)** Histogram of each coefficient values.
**b)** Cumulative histogram.
**c)**/**d)** 2D histogram plot with hexagonal bins between all coefficients, where a logarithmic scale was used to color each hexagon.
](images/coefs_comp/gtex_whole_blood/dist-main.svg "Distribution of coefficient values"){#fig:dist_coefs width="100%"}


We used gene expression data from GTEx v8 and selected the top five tissues with more sample size: muscle (skeletal), whole blood, skin (sun exposed), adipose (subcutaneous) and artery (tibial).
For each of these tissues, we selected the top 5,000 genes with largest variance, and then computed the pairwise correlation matrix using Pearson, Spearman and Clustermatch.
In Figure @fig:dist_coefs a), we show how the pairwise correlation values distribute in whole blood, where Clustermatch (mean=0.14, median=0.08, sd=0.15) has a much more skewed distribution than Pearson (mean=0.31, median=0.24, sd=0.24) and especially Spearman (mean=0.39, median=0.37, sd=0.26).
Each coefficient reaches 70% of gene pairs at $c$=0.18, $r$=0.44 and $r_s$=0.56 (Figure @fig:dist_coefs b)), respectively.
When we directly compare each coefficient with each other, the agreement between Clustermatch and Spearman is higher than any of these with Pearson (Figures @fig:dist_coefs c) and d)).


![
**Intersection of gene pairs with high and low coefficient values (GTEx v8, whole blood).**
High (low) refers to the top (bottom) 30% of each coefficient's values.
](images/coefs_comp/gtex_whole_blood/upsetplot-main.svg "Intersection of gene pairs"){#fig:upsetplot_coefs width="100%"}


When we directly compare each coefficient with each other, the agreement between Clustermatch and Spearman is higher than any of these with Pearson (Figures @fig:dist_coefs c) and d)).
However, although there is also a small subset of gene pairs that are captured by Pearson only (Figure @fig:dist_coefs c)), Clustermatch captures gene pairs that neither Pearson nor Spearman detect, representing potential interesting nonlinear relationships between genes' expression data.
To analyze these agreements and disagreements in more detail, we obtained for each coefficient the top 30% of gene pairs with the largest correlation values (a "high" subset) and the bottom 30% of those ("low" subset), resulting in six potentially overlapping subsets of gene pairs.
Then we used an UpSet plot [@doi:10.1109/TVCG.2014.2346248] to compute the intersections of these subsets and precisely identify the gene expression patterns captured and missed by each coefficient.
The results are in Figure @fig:upsetplot_coefs, which shows the agreements (left) and disagreements (right) between the coefficients.
The three coefficients agree more on gene pairs with a high correlation value (42.1%) than on those with no relationship (34.3%).
The figure also confirms that Clustermatch and Spearman agree more on highly correlated pairs (4.0% in "high", and 7.0% in "low") than any of these with Pearson (all between 0.3%-3.5% for "high", and 2.8%-5.5% for "low").
The same pattern is observed on the disagreements, where almost 10 thousand gene pairs have a high Clustermatch and Spearman coefficient and a low Pearson value, compared with only 531 gene pairs that are high for Pearson and low for Clustermatch and Spearman.
These results are expected in gene expression from GTEx, where samples are relatively homogeneous.
