### Clustermatch detects linear and nonlinear patterns in human transcriptomic data

![
**Distribution of coefficient values on gene expression (GTEx v8, whole blood).**
**a)** Histogram of each coefficient values.
**b)** Cumulative histogram.
**c)**/**d)** 2D histogram plot with hexagonal bins between all coefficients, where a logarithmic scale was used to color each hexagon.
](images/coefs_comp/gtex_whole_blood/dist-main.svg "Distribution of coefficient values"){#fig:dist_coefs width="100%"}


We used gene expression data from GTEx v8 and selected the top five tissues with more sample size: muscle (skeletal), whole blood, skin (sun exposed), adipose (subcutaneous) and artery (tibial).
For each of these tissues, we selected the top 5,000 genes with largest variance, and then computed the pairwise correlation matrix using Pearson, Spearman and Clustermatch.
In Figure @fig:dist_coefs a, we show how the pairwise correlation values distribute in whole blood, where Clustermatch (mean=0.14, median=0.08, sd=0.15) has a much more skewed distribution than Pearson (mean=0.31, median=0.24, sd=0.24) and especially Spearman (mean=0.39, median=0.37, sd=0.26).
Each coefficient reaches 70% of gene pairs at $c$=0.18, $r$=0.44 and $r_s$=0.56 (Figure @fig:dist_coefs b, respectively.
When we directly compare each coefficient with each other, the agreement between Clustermatch and Spearman is higher than any of these with Pearson (Figures @fig:dist_coefs c and d).


![
**Intersection of gene pairs with high and low coefficient values (GTEx v8, whole blood).**
**a)** UpSet plot with six categories (rows) with the highest and lowest 30% of gene pairs for each of the three correlation coefficients.
Columns show different intersections of categories grouped by agreements and disagreements.
**b)** Hexagonal binning plots with gene pair examples where Clustermatch ($c$) disagrees with Pearson ($r$) and Spearman ($r_s$).
Hexagon colors follow a logarithmic scale.
](images/coefs_comp/gtex_whole_blood/upsetplot-main.svg "Intersection of gene pairs"){#fig:upsetplot_coefs width="100%"}


A closer inspection of gene pairs detected and missed by these coefficients revealed the ability of Clustermatch to capture more complex yet biologically meaningful patterns.
For this, we analyzed the agreements and disagreements by obtaining for each coefficient the top 30% of gene pairs with the largest correlation values (a "high" set) and the bottom 30% of those ("low"), resulting in six potentially overlapping categories of gene pairs.
An UpSet plot [@doi:10.1109/TVCG.2014.2346248] is shown in Figure @fig:upsetplot_coefs a, where the intersections of these six categories allowed to precisely identify the gene expression patterns captured and missed by each coefficient.
The three coefficients agree more on gene pairs with a high correlation value (42.1%) than on those with no relationship (34.3%).
The figure also confirms that Clustermatch and Spearman agree more on highly correlated pairs (4.0% in "high", and 7.0% in "low") than any of these with Pearson (all between 0.3%-3.5% for "high", and 2.8%-5.5% for "low").
Regarding disagreements, there are thousands of gene pairs with a high Clustermatch value that are not detected by the other coefficients.
There are also gene pairs with a high Pearson value that have either low Clustermatch (1,075) or low Clustermatch and low Spearman values (531).
However, these cases mostly seem to be driven by outliers (Figure @fig:upsetplot_coefs b, `Supplementary Figures XX, YY, ZZ`{.red}).
No gene pairs highly ranked by Spearman are missed by Clustermatch.


In Figure @fig:upsetplot_coefs b, we focused on individual examples of gene pairs where Clustermatch disagrees with Pearson, Spearman or both.
<!--
SDS / IFNG:
    https://hb.flatironinstitute.org/gene/10993+3458
    expressed in leukocyte, blood, lymphocyte, spleen
    serine seems to be essential for T cell expansion: https://www.cell.com/cell-metabolism/pdfExtended/S1550-4131(16)30644-1

JUN / APOC1:
    https://hb.flatironinstitute.org/gene/3725+341
    both expressed in blood, liver, leukocyte
    both seem to be very related to the regulation of immune response

BAG3 / NRADDP:
    maybe remove this because NRADDP is pseudo gene
        and it is not present in GIANT

UTY / KDM6A:
    these are from chr Y and X
    easy explanation

RASSF2 / CYTIP:
    https://hb.flatironinstitute.org/gene/9770+9595
    both expressed in peripheral blood (super strong), leukocyte (super strong), blood, etc etc (super strong)

AC068... / KLHL21:
    maybe remove? I can't fine AC068... gene

the other two gene pairs detected by Pearson do not interact much in GIANT
 -->
