### The CCC reveals linear and nonlinear patterns in human transcriptomic data

We examined the characteristics of the correlation coefficients in gene expression data from GTEx v8 across different tissues.
We selected the top 5,000 genes with the highest variance for our initial analyses on whole blood and then computed the correlation matrix between these genes using Pearson, Spearman and CCC (see [Methods](#sec:data_gtex)).


We examined the distribution of each coefficient's absolute values in GTEx (Figure @fig:dist_coefs).
The CCC coefficient (mean=0.14, median=0.08, sd=0.15) had a much more skewed distribution than Pearson (mean=0.31, median=0.24, sd=0.24) and Spearman (mean=0.39, median=0.37, sd=0.26).
The coefficients reached a cumulative set containing 70% of gene pairs at different values (Figure @fig:dist_coefs b), with CCC at $c=0.18$, Pearson at $p=0.44$, and Spearman at $s=0.56$.
This suggests that for this type of data, the coefficients are not directly comparable by magnitude, so we used ranks for further comparisons.
In GTEx v8, CCC values were closer to Spearman and vice versa than either was to Pearson (Figure @fig:dist_coefs c).
We also compared the Maximal Information Coefficient (MIC) in this data (see [Supplementary Note 1](#sec:mic)).
We found that CCC behaved very similarly to MIC, although CCC was up to two orders of magnitude faster to run (see [Supplementary Note 2](#sec:time_test)).
MIC is an advanced correlation coefficient which can capture general patterns beyond linear relationships and has been used successfully in various application domains [@pmid:33972855; @pmid:33001806; @pmid:27006077].
These results suggest that our findings for CCC generalize to MIC.
Therefore, in the subsequent analyses we focus on CCC and linear-only coefficients.


![
**Distribution of coefficient values on gene expression (GTEx v8, whole blood).**
**a)** Histogram of coefficient values.
**b)** Corresponding cumulative histogram. The dotted line maps the coefficient value that accumulates 70% of gene pairs.
**c)** 2D histogram plot with hexagonal bins between all coefficients, where a logarithmic scale was used to color each hexagon.
](images/coefs_comp/gtex_whole_blood/dist-main.svg "Distribution of coefficient values"){#fig:dist_coefs width="100%"}


We further analyzed the agreement and disagreement between the three correlation coefficients.
We obtained the top 30% of gene pairs with the largest correlation values ("high" set) and the bottom 30% ("low" set) for each coefficient.
An UpSet analysis showed that the three coefficients agreed on whether there is a strong correlation (42.1%) or no relationship (34.3%) in 76.4% of cases.
Additionally, CCC and Spearman agreed more on either highly or poorly correlated pairs (4.0% in "high", and 7.0% in "low") than either with Pearson (between 0.3%-3.5% for "high", and 2.8%-5.5% for "low").
Overall, CCC agreed with either Pearson or Spearman in 90.5% of gene pairs by assigning a high or a low correlation value.

![
**Intersection of gene pairs with high and low correlation coefficient values (GTEx v8, whole blood).**
**a)** UpSet plot with six categories (rows) grouping the 30% of the highest (green triangle) and lowest (red triangle) values for each coefficient.
Columns show different intersections of categories grouped by agreements and disagreements.
**b)** Hexagonal binning plots with examples of gene pairs where CCC ($c$) disagrees with Pearson ($p$) and Spearman ($s$).
For each method, colors in the triangles indicate if the gene pair is among the top (green) or bottom (red) 30% of coefficient values.
No triangle means that the correlation value for the gene pair is between the 30th and 70th percentiles (neither low nor high).
A logarithmic scale was used to color each hexagon.
](images/coefs_comp/gtex_whole_blood/upsetplot-main.svg "Intersection of gene pairs"){#fig:upsetplot_coefs width="100%"}


We found that more than 20,000 gene pairs had a high CCC value, but were not highly ranked by the other correlation coefficients (Figure @fig:upsetplot_coefs a, right).
Additionally, 1,075 gene pairs had a high Pearson value but low CCC and Spearman values, and 531 gene pairs had both low CCC and low Spearman values.
Our examination suggests that many of these cases may be driven by potential outliers (Figure @fig:upsetplot_coefs b, and analyzed later).
We analyzed the top five gene pairs in the "Disagreements" group (Figure @fig:upsetplot_coefs a, right) where CCC disagreed with Pearson, Spearman, or both.

![
**The expression levels of *KDM6A* and *UTY* display sex-specific associations across GTEx tissues.**
CCC captures this nonlinear relationship in all GTEx tissues (nine examples are shown in the first three rows), except in female-specific organs (last row).
](images/coefs_comp/kdm6a_vs_uty/gtex-KDM6A_vs_UTY-main.svg "KDM6A and UTY across different GTEx tissues"){#fig:gtex_tissues:kdm6a_uty width="95%"}

The first three gene pairs (*IFNG* - *SDS*, *JUN* - *APOC1*, and *ZDHHC12* - *CCL18*) show a non-coexistence relationship, with high CCC and low Pearson values.
This suggests a potentially inhibiting effect, where in samples where one gene is highly expressed, the other is slightly activated and vice versa.
The next three gene pairs (*UTY* - *KDM6A*, *RASSF2* - *CYTIP*, and *AC068580.6* - *KLHL21*) combine either two linear or one linear and one independent relationships.
For example, *UTY* and *KDM6A* (paralogs) show a nonlinear relationship, with a subset of samples following a linear pattern and another subset with a constant expression of one gene.
This is due to the fact that *UTY* is located on chromosome Y (Yq11) and *KDM6A* is located on chromosome X (Xp11), with the linear pattern being specific to males and no expression of *UTY* in females.
CCC captures this combination of linear and independent patterns ($c=0.29$, above the 80th percentile), while Pearson ($p=0.24$, below the 55th percentile) and Spearman ($s=0.10$, below the 15th percentile) do not.
Furthermore, the same gene pair pattern is highly ranked by CCC in most tissues in GTEx, except for female-specific organs (Figure @fig:gtex_tissues:kdm6a_uty).
