## Discussion

We introduced the Clustermatch Correlation Coefficient (CCC), an efficient not-only-linear machine learning-based method able to capture more general patterns in data.
We applied it to gene expression data from GTEx v8 and found that our coefficient is robust to outliers and does not miss strongly linear relationships in gene-gene patterns.
Moreover, CCC also captured complex and biologically meaningful relationships completely missed by standard coefficients.
For instance, CCC detected gene pairs from the sex chromosomes following complex non-linear patterns that were completely missed by linear-only methods.
These examples, in particular, highlight the importance to consider sex as a biological variable and the key role that not-only-linear coefficients can play in capturing sex-specific differences.
We also replicated gene pairs prioritized by CCC in GTEx using tissue-specific gene networks from GIANT trained without GTEx samples.
<!-- Using a computational model from GIANT to predict gene pairs with cell type-specific expression. -->
We found that pairs exclusively prioritized by CCC in whole blood from GTEx were predicted in GIANT to be part of the same biological mechanism and specifically expressed in cell lineages from blood, even though CCC did not have access to any cell lineage-specific information.
<!-- Our analyses showed that directly comparing CCC with linear-only coefficients was an useful approach to highlight the most complex and potentially promising gene pairs. -->
Finally, CCC derives scores very well aligned with MIC while being much more computationally efficient and thus practical for use in large modern datasets.


It is well-known that biomedical research is biased towards a small fraction of human genes [@pmid:17620606; @pmid:17472739].
Researching genes with well-known functions is easier, although this observational bias seems anachronic with current high-throughput technologies.
Several factors explaining this behavior have been identified [@pmid:30226837], such as RNA and protein abundance or gene length.
Another potential explanation could also be a bias towards linear-only statistical methods.
For example, in genome-wide association studies (GWAS), which in most cases employ a linear model, have been successful to identify thousands of genetic variants associated with complex human diseases [@doi:10.1016/j.ajhg.2017.06.005; @doi:10.1038/s41467-020-19653-5].
These findings have been very helpful to understand the molecular basis of common diseases [@doi:10.1038/s41588-018-0241-6] and, more recently, design polygenic scores to estimate an individual's genetic predisposition for a particular trait or disorder [@doi:10.1038/s41588-018-0183-z].
However, the estimated effects of genes identified with GWAS are generally modest and they explain only a fraction of the trait variance, hampering the clinical translation of these findings.
Recent theories, like the omnigenic model [REF], argue that these observations in genetic studies are explained by highly-interconnected gene regulatory networks, with some core genes having a more direct effect on the phenotype than others.
Using this omnigenic perspective, we and others [@doi:10.1101/2021.07.05.450786; @doi:10.1186/s13040-020-00216-9; @doi:10.1101/2021.10.21.21265342] have shown that integrating gene co-expression networks in genetic studies could potentially identify core genes that are missed by linear-only models alone like GWAS.
Our results strongly suggest that building these networks with more advanced and efficient correlation coefficients could provide a better estimation of gene co-expression profiles, and thus more accurately identify these core genes.
We anticipate that approaches like CCC will play a significant role in precision medicine approach by providing the computational tools to focus on less-studied yet potentially more promising genes.


Our analyses have some limitations.
We worked on a sample with the top variable genes to keep computation time feasible.
Although CCC is much faster than MIC, Pearson and Spearman are still the most computationally efficient since they only rely on simple data statistics, although this significantly limits their ability beyond linear patterns.
Even with this small sample of genes, our results confirm that the advantages of using more advanced yet efficient coefficients like CCC can help detect and study more intricate molecular mechanisms.
Although we only used GTEx, with a relatively homogeneous set of samples, we could still find complex and meaningful patterns.
This suggests that the application of CCC on larger compendia, such as recount3 [@pmid:34844637] with thousands of heterogeneous samples across different conditions, can reveal other potentially meaningful gene interactions.


While linear and rank-based correlation coefficients are exceptionally fast to calculate, not all relevant patterns in biological datasets are linear.
For example, patterns associated with sex as a biological variable are not apparent to the linear-only coefficients that we evaluated but are revealed by not-only-linear methods.
Not-only-linear coefficients can also disentangle intrincate yet relevant patterns from gene expression data alone that replicated in models integrating different data modalities.
We find that the CCC is an efficient, next-generation correlation coefficient highly effective in transcriptome analyses but also potentially useful in a broad range of other domains.
