## Discussion

We previously showed that Clustermatch outperformed all other coefficients in a simulated cluster analysis scenario with linear and nonlinear patterns and varying noise levels.
Here we introduced the Clustermatch correlation coefficient, an efficient machine learning-based method and an optimized Python implementation.
We applied it to gene expression data from GTEx v8 and found that our coefficient is robust to outliers and does not miss important linear relationships in gene-gene patterns.
Clustermatch also captured complex and biologically meaningful relationships completely missed by standard coefficients.
<!-- TODO: something about GIANT results? -->
We also showed that directly comparing Clustermatch with linear-only coefficients highlighted the most complex and potentially promising gene pairs.
Finally, Clustermatch derives scores very well aligned with the Maximal Information Coefficient while being much more computationally efficient and thus practical for use in large modern datasets.


Our analyses have some limitations.
We worked on a sample with the top variable genes to keep computation time feasible for Clustermatch.
Although Clustermatch is faster than MIC or DC, Pearson and Spearman are still the fastest since they only rely on simple data statistics.
This smaller sample of genes might not show an accurate picture of all patterns in data.
However, our results confirm that the advantages of using more sophisticated yet efficient methods like Clustermatch can help detect and study more intricate molecular mechanisms.
Although we only used GTEx, with a relatively homogeneous set of samples, we could still find complex and meaningful patterns, suggesting that the application of Clustermatch on larger compendia, such as recount3 with thousands of samples across different conditions, can reveal other potentially important gene interactions.


It is well-known that biomedical research is biased towards a small fraction of human genes [@pmid:17620606; @pmid:17472739].
Researching genes with well-known functions is easier, although this observational bias seems anachronic with current high-throughput technologies.
Although several factors explaining this behavior have been identified [@pmid:30226837], such as RNA and protein abundance or gene length, another potential explanation could also be a bias of current statistical methods towards linear-only patterns.
It is well-known that deep learning approaches have revolutionazed several areas in research and industry, but interpretation is crucial in other fields such as biology and medicine.
In these health-related fields, another breakthrough theoretical advance was the omnigenic model of complex traits, where gene regulatory networks are now first-class players in genetic studies.
This model not only shifted our attention to other, potentially less studied genes that are part of disease-relevant networks, but can also explain why polygenic risk scores perform so poorly across different population ancestries.
The streetlight effect [@url:https://www.discovermagazine.com/the-sciences/why-scientific-studies-are-so-often-wrong-the-streetlight-effect], where we only search where it is easiest to look, is widespread in areas beyond biology.
We anticipate that sophisticated and efficient approaches like Clustermatch will play a significant role in focusing on other less studied areas of the genome and other research fields and applications domains.


Computing a correlation coefficient based on simple data summaries is tempting fast.
However, this study shows that those methods can miss essential patterns to understand the big picture, such as a gene co-expression network representing robust, accurate and not-only-linear interactions.
We provide an efficient method based on machine learning techniques that can process heterogeneous data types seamlessly, dramatically easing preprocessing steps for the end-user.
