## Discussion

We introduce the Clustermatch Correlation Coefficient (CCC), an efficient machine learning-based statistic that is capable of detecting not-only-linear relationships.
Applying CCC to GTEx v8, we found that it was robust to outliers and was able to detect linear relationships as well as complex and biologically meaningful patterns that standard coefficients missed.
Specifically, CCC alone was able to detect gene pairs with complex nonlinear patterns from the sex chromosomes, demonstrating its potential to capture sex-specific differences.
In addition, we found that CCC can also disentangle intricate cell lineage-specific transcriptional patterns that are missed by linear-only coefficients.
This suggests that CCC can be used to detect complex relationships where a subset of samples or conditions are explained by other factors, such as differences between health and disease.
Furthermore, the CCC was more similar to Spearman than Pearson, highlighting their shared robustness to outliers.
Moreover, the CCC results were concordant with MIC, but much faster to compute, making it practical for large datasets.
Additionally, CCC has the advantage of being able to process categorical variables together with numerical values, and has a single parameter that controls the maximum complexity of the detected relationships while also balancing compute time, making it conceptually easy to interpret.


Datasets such as Anscombe or "Datasaurus" show the importance of visualizing data instead of relying on simple summaries.
However, for many datasets, it is not possible to examine every possible relationship, and more sophisticated and robust correlation coefficients are needed.
CCC is an advanced yet interpretable coefficient that can help identify patterns that are more likely to reflect real biology.
For example, genes *UTY* - *KDM6A* (from sex chromosomes) have a strong linear relationship, but only in a subset of samples (males).
This highlights the importance of considering sex as a biological variable (SABV) [@doi:10.1038/509282a] to avoid overlooking differences between men and women, such as in disease manifestations [@doi:10.1210/endrev/bnaa034; @doi:10.1038/s41593-021-00806-8].
A not-only-linear correlation coefficient like CCC could also reveal differences between variables, such as genes, that are explained by a third factor, which would be missed by linear-only coefficients.


It is well-known that biomedical research is biased towards a small fraction of human genes [1,2].
Some genes highlighted in CCC-ranked pairs (Figure b), such as *SDS* (12q24) and *ZDHHC12* (9q34), were previously found to be the focus of fewer than expected publications [3].
Linear correlation coefficients may bias researchers away from genes with complex coexpression patterns.
A beyond-linear gene co-expression analysis on large compendia could help uncover the function of understudied genes.
For example, gene *KLHL21* (1p36) and *AC068580.6* (*ENSG00000235027*, in 11p15) have a high CCC value and are missed by the other coefficients.
*KLHL21* has been suggested as a potential therapeutic target for hepatocellular carcinoma [4] and other cancers [5,6].
Its nonlinear correlation with *AC068580.6* could uncover other important players in cancer initiation or progression, potentially in subsets of samples with specific characteristics (as suggested in Figure b).


Not-only-linear correlation coefficients may be useful in genetic studies.
Genome-wide association studies (GWAS) have been successful in understanding the connection between genotype and phenotype, but the estimated effect sizes of the identified genes are usually small, and they explain only a small part of the phenotype variance [@doi:10.1016/j.ajhg.2017.06.005; @doi:10.1038/s41576-019-0127-1].
This has hindered the clinical translation of these findings.
The omnigenic model for complex traits [@pmid:28622505; @pmid:31051098] suggests that highly-interconnected gene regulatory networks could explain this, with some core genes having a more direct effect on the phenotype than others.
We and others [@doi:10.1101/2021.07.05.450786; @doi:10.1186/s13040-020-00216-9; @doi:10.1101/2021.10.21.21265342] have demonstrated that integrating gene co-expression networks in genetic studies could potentially identify core genes that are not found with linear-only models such as GWAS.
Our results indicate that using more advanced and efficient correlation coefficients to build these networks could better estimate gene co-expression profiles and thus more accurately identify these core genes.
Approaches like CCC could have a significant role in precision medicine by providing the computational tools to focus on more promising genes, which may represent better candidate drug targets.


Our analysis has some limitations.
We worked with a sample of the most variable genes in order to keep computation time manageable.
Although CCC is much faster than MIC, Pearson and Spearman remain the most computationally efficient, as they rely on simple data statistics.
However, our results show the advantages of using more advanced coefficients such as CCC to detect and study more intricate molecular mechanisms that are replicated in independent datasets.
Applying CCC to larger compendia, such as recount3 [@pmid:34844637] with thousands of heterogeneous samples across different conditions, could reveal other potentially meaningful gene interactions.
The single parameter of CCC, $k_{\mathrm{max}}$, controls the maximum complexity of patterns found and also affects compute time.
Our analysis suggested that $k_{\mathrm{max}}=10$ was sufficient to identify both linear and more complex patterns in gene expression.
A more comprehensive analysis of optimal values for this parameter could provide insights for adjusting it for different applications or data types.


While linear and rank-based correlation coefficients are fast to calculate, they are not always sufficient for detecting patterns in biological datasets.
For example, sex differences are not always apparent with linear-only coefficients, but can be revealed by not-only-linear methods.
Not-only-linear coefficients can also identify intricate patterns in expression data that were replicated when different data modalities were integrated.
The CCC correlation coefficient is particularly efficient and highly effective in transcriptome analyses, and could be useful in a range of other domains.
Moreover, it is highly parallelizable, and we anticipate that efficient GPU-based implementations could make it even faster.
