## Discussion

We present the Clustermatch Correlation Coefficient (CCC), a machine learning-based statistic that is efficient and not limited to linear relationships.
Applying CCC to the GTEx v8 dataset revealed it is robust to outliers and can detect linear relationships as well as complex patterns that standard coefficients miss.
CCC was able to capture nonlinear patterns from the sex chromosomes and showed potential to detect complex relationships between a subset of samples or conditions.
For example, top CCC-ranked gene pairs in whole blood from GTEx were replicated in independent tissue-specific networks trained from multiple data types and attributed to cell lineages from blood, despite CCC not having access to any cell lineage-specific information.
CCC was also more similar to Spearman than Pearson, demonstrating their shared robustness to outliers.
Compared to MIC, CCC is much faster to compute and can process categorical variables as well as numerical values.
It is conceptually easy to interpret and has a single parameter that controls the maximum complexity of the detected relationships while also balancing compute time.


Visual analysis can be helpful in examining datasets, but for many datasets it is infeasible to examine each possible relationship.
This is where more sophisticated and robust correlation coefficients, such as CCC, can be useful.
CCC can identify patterns that are more likely to reflect real biology, such as the strong linear relationship between genes *UTY* and *KDM6A* (from sex chromosomes) in a subset of samples (males).
This highlights the importance of considering sex as a biological variable (SABV) [@doi:10.1038/509282a], as differences between men and women can have an impact on disease manifestations [@doi:10.1210/endrev/bnaa034; @doi:10.1038/s41593-021-00806-8].
Additionally, CCC can identify significant differences between variables that are explained by a third factor (beyond sex differences), which would be entirely missed by linear-only coefficients.


It is well-known that biomedical research is biased towards a small fraction of human genes [@pmid:17620606; @pmid:17472739].
Some genes highlighted in CCC-ranked pairs (Figure @fig:upsetplot_coefs b), such as *SDS* (12q24) and *ZDHHC12* (9q34), were found to have fewer publications than expected [@pmid:30226837].
This suggests that the widespread use of linear correlation coefficients may be limiting researchers from discovering complex coexpression patterns.
A beyond-linear analysis of large compendia may help to uncover the function of understudied genes.
For example, gene *KLHL21* (1p36) and *AC068580.6* (*ENSG00000235027*, in 11p15) have a high CCC value and are missed by the other coefficients.
*KLHL21* has been suggested as a potential therapeutic target for hepatocellular carcinoma [@pmid:27769251] and other cancers [@pmid:29574153; @pmid:35084622].
Investigating the nonlinear correlation between *KLHL21* and *AC068580.6* may reveal additional players in cancer initiation or progression, particularly in subsets of samples with specific characteristics (as suggested in Figure @fig:upsetplot_coefs b).


Not-only-linear correlation coefficients have potential to be beneficial in genetic studies.
Genome-wide association studies (GWAS) have been successful in understanding the molecular basis of common diseases by estimating the association between genotype and phenotype [@doi:10.1016/j.ajhg.2017.06.005].
However, the estimated effect sizes of genes identified with GWAS are generally modest, which explains only a fraction of the phenotype variance and hinders the clinical translation of these findings [@doi:10.1038/s41576-019-0127-1].
The omnigenic model for complex traits [@pmid:28622505; @pmid:31051098] suggests that highly-interconnected gene regulatory networks may be at play, with some core genes having a more direct effect on the phenotype than others.
Using this omnigenic perspective, researchers have found that integrating gene co-expression networks in genetic studies could potentially identify core genes that are missed by linear-only models alone like GWAS [@doi:10.1101/2021.07.05.450786; @doi:10.1186/s13040-020-00216-9; @doi:10.1101/2021.10.21.21265342].
Our results indicate that building these networks with more efficient correlation coefficients, such as CCC, could better estimate gene co-expression profiles and thus more accurately identify these core genes.
Such approaches could be of great importance in the precision medicine field, providing the computational tools to focus on more promising genes that could represent better candidate drug targets.


Our analysis has some limitations.
We used a sample with the top variable genes to keep computation time feasible.
Although CCC is much faster than MIC, Pearson and Spearman remain the most computationally efficient due to their reliance on simple data statistics.
However, our results demonstrate the advantages of more advanced coefficients like CCC for detecting and studying more intricate molecular mechanisms that replicate in independent datasets.
Applying CCC to larger compendia, such as recount3 [@pmid:34844637] with thousands of heterogeneous samples across different conditions, could reveal other potentially meaningful gene interactions.
The single parameter of CCC, $k_{\mathrm{max}}$, controls the maximum complexity of patterns found and also affects compute time.
Our analysis suggested that $k_{\mathrm{max}}=10$ was sufficient to identify both linear and more complex patterns in gene expression.
Further research on optimal values for this parameter could provide insights to adjust it for different applications or data types.


Linear and rank-based correlation coefficients are quick to calculate, however, many patterns in biological datasets are nonlinear.
For example, patterns related to sex can only be revealed by not-only-linear methods.
Furthermore, not-only-linear coefficients can identify intricate patterns from expression data that are replicated in models combining different data modalities.
CCC is highly parallelizable, and its efficiency can be further enhanced by GPU-based implementations.
Therefore, the CCC is an efficient correlation coefficient that is useful for transcriptome analyses and potentially other domains.
