## Discussion

We introduced the Clustermatch correlation coefficient, an efficient machine learning-based method along with an optimized implementation in Python.
We applied it to gene expression data from GTEx v8 and found that our coefficient is robust to outliers and does not miss important linear relationships in gene-gene patterns.
Clustermatch also captured complex and biologically meaningful relationships completely missed by standard coefficients.
<!-- TODO: something about GIANT results? -->
We also showed that directly comparing Clustermatch with linear-only coefficients allowed us to highlight the most complex and potentially promising gene pairs.
Finally, Clustermatch derives scores very well aligned with the Maximal Information Coefficient, while being much more computationally efficient and thus practical for use in modern large datasets.


It is well-known that the biomedical research is biased towards a small fraction of human genes [@pmid: 17620606; @pmid:17472739].
- a small portion of human genes are studied more than the rest, the rich gets richer [@pmid:30226837; @pmid:30256779]
- SIMPLIFY ALL THESE IN JUST TWO SENTENCES: the omnigenic model, a modern hypothesis of the genetic architecture of complex traits, postulates that potentially all genes expressed in disease-relevant cell types might contribute to the phenotype
    - the rationale is that gene regulatory networks are so highly interconnected, that the expression of genes more directly affecting a disease is *trans*-regulated (see K99 for wording) by thousands of other peripherial genes
    - GWAS, however, have found only these peripheral genes instead of the core ones
- Robust and sophisticated methods to capture correlations in large datasets can also help in reverting this trend (OF STUDING ONLY WELL KNOWN GENES)


Computing a correlation coefficient based on simple data summaries is tempting fast, but as we have shown in this study, those methods can miss essential patterns to understand the big picture, such as an accurate gene co-expression network.
Instead, we provide an efficient method based on machine learning techniques that can process numerical and categorical data types seamlessly, dramatically easing preprocessing steps for the end user.
