## Discussion

We introduce the Clustermatch Correlation Coefficient (CCC), an efficient machine learning-based statistic that can detect both linear and nonlinear relationships.
Applying CCC to GTEx v8 data revealed that it was robust to outliers and was able to detect complex and biologically meaningful patterns that standard coefficients missed.
In particular, CCC was able to capture nonlinear patterns from the sex chromosomes, highlighting its ability to detect sex-specific differences.
The ability to capture these nonlinear patterns extends beyond sex differences, as it can detect complex relationships where a subset of samples or conditions are explained by other factors (such as differences between health and disease).
We found that top CCC-ranked gene pairs in whole blood from GTEx were replicated in independent tissue-specific networks trained from multiple data types and attributed to cell lineages from blood, even though CCC did not have access to any cell lineage-specific information.
This suggests that CCC can disentangle intricate cell lineage-specific transcriptional patterns missed by linear-only coefficients.
In addition to capturing nonlinear patterns, the CCC was more similar to Spearman than Pearson, highlighting their shared robustness to outliers.
Compared to MIC, CCC is much faster to compute and thus practical for large datasets.
It also has a single parameter that controls the maximum complexity of the detected relationships while also balancing compute time.
Furthermore, CCC can process categorical variables together with numerical values, making it conceptually easy to interpret.


Visual analysis is helpful, but examining each possible relationship in a dataset is often infeasible.
This is why more sophisticated correlation coefficients like CCC are useful.
These coefficients can focus human interpretation on patterns that are more likely to reflect real biology.
For example, CCC detected a strong linear relationship between genes *UTY* - *KDM6A* (from sex chromosomes) in a subset of samples (males).
This highlights the importance of considering sex as a biological variable to avoid overlooking differences between men and women, such as in disease manifestations.
A not-only-linear correlation coefficient like CCC can also identify significant differences between variables that are explained by a third factor, which would be missed by linear-only coefficients.


It is well-known that biomedical research focuses on a small fraction of human genes [@pmid:17620606; @pmid:17472739].
For example, some genes highlighted in CCC-ranked pairs (Figure @fig:upsetplot_coefs b), such as *SDS* (12q24) and *ZDHHC12* (9q34), were found to have fewer publications than expected [@pmid:30226837].
This suggests that the widespread use of linear coefficients may bias researchers away from genes with complex coexpression patterns.
A beyond-linear gene co-expression analysis on large compendia could reveal the function of understudied genes.
For instance, gene *KLHL21* (1p36) and *AC068580.6* (*ENSG00000235027*, in 11p15) have a high CCC value and are missed by the other coefficients.
*KLHL21* has been suggested as a potential therapeutic target for hepatocellular carcinoma [@pmid:27769251] and other cancers [@pmid:29574153; @pmid:35084622].
Its nonlinear correlation with *AC068580.6* may uncover other important players in cancer initiation or progression, potentially in subsets of samples with specific characteristics (as suggested in Figure @fig:upsetplot_coefs b).


Not-only-linear correlation coefficients may be beneficial in genetic studies.
Genome-wide association studies (GWAS) have been successful in understanding the molecular basis of common diseases by estimating the association between genotype and phenotype [@doi:10.1016/j.ajhg.2017.06.005].
However, the estimated effect sizes of genes identified with GWAS are generally modest, and they explain only a fraction of the phenotype variance, hampering the clinical translation of these findings [@doi:10.1038/s41576-019-0127-1].
The omnigenic model for complex traits [@pmid:28622505; @pmid:31051098] suggests that these observations are explained by highly-interconnected gene regulatory networks, with some core genes having a more direct effect on the phenotype than others.
Integrating gene co-expression networks in genetic studies, as done by us and others [@doi:10.1101/2021.07.05.450786; @doi:10.1186/s13040-020-00216-9; @doi:10.1101/2021.10.21.21265342], could potentially identify core genes that are missed by linear-only models alone like GWAS.
Our results suggest that using more advanced and efficient correlation coefficients, such as CCC, to build these networks could better estimate gene co-expression profiles and thus more accurately identify these core genes.
This approach could be instrumental in precision medicine, by providing the computational tools to focus on more promising genes that represent potentially better candidate drug targets.


Our analysis has some limitations.
We worked with a sample of the top variable genes in order to keep computation time feasible.
Although CCC is faster than MIC, Pearson and Spearman are still the most computationally efficient, since they rely on simple data statistics.
However, our results demonstrate the advantages of using more advanced coefficients like CCC to detect and study more intricate molecular mechanisms that are replicated in independent datasets.
Applying CCC to larger compendia, such as recount3 [@pmid:34844637] with thousands of heterogeneous samples across various conditions, could reveal other potentially meaningful gene interactions.
CCC's single parameter, $k_{\mathrm{max}}$, controls the maximum complexity of patterns found and also impacts compute time.
Our analysis suggested that $k_{\mathrm{max}}=10$ was sufficient to identify both linear and more complex patterns in gene expression.
A more comprehensive analysis of optimal values for this parameter could provide insights to adjust it for different applications or data types.


Linear and rank-based correlation coefficients are fast to calculate, but they cannot detect nonlinear patterns in biological datasets.
For example, sex-based patterns are not apparent with linear-only coefficients, but can be revealed by not-only-linear methods.
Furthermore, not-only-linear coefficients can identify intricate patterns in expression data, which can be replicated using models integrating different data modalities.
CCC is particularly advantageous in this regard, as it is highly parallelizable and can be implemented quickly using GPU-based methods.
In conclusion, CCC is an efficient and effective correlation coefficient for transcriptome analyses, and may have broader applications.
