## Discussion

We introduce the Clustermatch Correlation Coefficient (CCC), an efficient machine learning-based statistic that is not limited to linear relationships.
Applying CCC to GTEx v8 revealed that it was robust to outliers and detected linear relationships as well as complex, biologically meaningful patterns that traditional correlation coefficients missed.
CCC was able to detect gene pairs with complex nonlinear patterns from the sex chromosomes, highlighting its ability to capture sex-specific differences.
Moreover, it was effective in detecting complex relationships between samples or conditions that are explained by other factors, such as differences between health and disease.
Our results showed that the top CCC-ranked gene pairs in whole blood from GTEx were replicated in independent tissue-specific networks trained from multiple data types and attributed to cell lineages from blood, despite CCC not having access to any cell lineage-specific information.
This suggests that CCC can detect intricate cell lineage-specific transcriptional patterns that linear-only coefficients miss.
In addition to capturing nonlinear patterns, CCC was more similar to Spearman than Pearson, highlighting their shared robustness to outliers.
Furthermore, CCC was much faster to compute than MIC, making it practical for large datasets.
Additionally, it can process categorical variables together with numerical values.
CCC is conceptually easy to interpret and has a single parameter that controls the maximum complexity of the detected relationships while also balancing compute time.


Datasets such as Anscombe or "Datasaurus" show the value of visual analysis over relying on simple summaries.
However, examining each possible relationship in many datasets is infeasible.
Advanced correlation coefficients like CCC can identify patterns that are likely to reflect real biology.
For example, CCC detected a strong linear relationship between genes *UTY* - *KDM6A* (from sex chromosomes) in a subset of samples (males), which was not captured by linear-only coefficients.
This emphasizes the importance of considering sex as a biological variable (SABV) [@doi:10.1038/509282a] to avoid overlooking differences between men and women, such as in disease manifestations [@doi:10.1210/endrev/bnaa034; @doi:10.1038/s41593-021-00806-8].
A not-only-linear correlation coefficient like CCC could identify significant differences between variables (such as genes) that are explained by a third factor (beyond sex differences), which would be entirely missed by linear-only coefficients.


Biomedical research has been found to focus on a small fraction of human genes [@pmid:17620606; @pmid:17472739], with some genes highlighted in CCC-ranked pairs (Figure @fig:upsetplot_coefs b) receiving fewer publications than expected [@pmid:30226837].
This suggests that the use of linear coefficients may bias researchers away from genes with complex coexpression patterns.
To explore this further, a beyond-linear gene co-expression analysis on large compendia could be conducted.
For example, gene *KLHL21* (1p36) and *AC068580.6* (*ENSG00000235027*, in 11p15) have a high CCC value and are missed by the other coefficients.
*KLHL21* has been suggested as a potential therapeutic target for hepatocellular carcinoma [@pmid:27769251] and other cancers [@pmid:29574153; @pmid:35084622].
Its nonlinear correlation with *AC068580.6* may reveal other important players in cancer initiation or progression, especially in subsets of samples with specific characteristics (as suggested in Figure @fig:upsetplot_coefs b).


Not-only-linear correlation coefficients could be beneficial in the field of genetic studies.
Genome-wide association studies (GWAS) have been successful in understanding the molecular basis of common diseases by estimating the association between genotype and phenotype [@doi:10.1016/j.ajhg.2017.06.005].
However, the estimated effect sizes of genes identified with GWAS are modest, and they explain only a fraction of the phenotype variance, impeding the clinical application of these findings [@doi:10.1038/s41576-019-0127-1].
The omnigenic model for complex traits [@pmid:28622505; @pmid:31051098] suggests that these observations are due to highly-interconnected gene regulatory networks, with some core genes having a more direct effect on the phenotype than others.
We and others [@doi:10.1101/2021.07.05.450786; @doi:10.1186/s13040-020-00216-9; @doi:10.1101/2021.10.21.21265342] have demonstrated that integrating gene co-expression networks in genetic studies could potentially identify core genes that are overlooked by linear-only models like GWAS.
Our results suggest that constructing these networks with more advanced and efficient correlation coefficients could better estimate gene co-expression profiles and thus more accurately identify these core genes.
Approaches like CCC could be beneficial in the precision medicine field by providing the computational tools to focus on more promising genes with potential as better candidate drug targets.


Our analysis has some limitations.
We used a sample of the top variable genes to keep computation time feasible.
Although CCC is faster than MIC, Pearson and Spearman remain the most computationally efficient, as they rely on simple data statistics.
However, our results reveal the advantages of using more advanced coefficients like CCC to detect and study more intricate molecular mechanisms that replicated in independent datasets.
Applying CCC to larger compendia, such as recount3 [@pmid:34844637] with thousands of heterogeneous samples across different conditions, could reveal other potentially meaningful gene interactions.
The single parameter of CCC, $k_{\mathrm{max}}$, controls the maximum complexity of patterns found and also affects compute time.
Our analysis suggested that $k_{\mathrm{max}}=10$ was sufficient to identify linear and more complex patterns in gene expression.
Further research into optimal values for this parameter could provide insights to adjust it for different applications or data types.


Linear and rank-based correlation coefficients are fast to calculate, but not all patterns in biological datasets are linear.
For example, sex-related patterns are not revealed by linear-only coefficients.
Not-only-linear methods can identify patterns driven by other factors and disentangle intricate yet relevant patterns from expression data alone.
The CCC is an efficient correlation coefficient that is highly effective in transcriptome analyses and potentially useful in other domains.
It is also highly parallelizable, and efficient GPU-based implementations could make it even faster.
