## Discussion

We introduce the Clustermatch Correlation Coefficient (CCC), an efficient machine learning-based statistic that captures not only linear relationships, but also complex, nonlinear patterns.
Applying CCC to GTEx v8 revealed it was robust to outliers, detecting linear relationships as well as biologically meaningful patterns that standard coefficients missed.
For example, CCC was able to detect gene pairs with complex nonlinear patterns from the sex chromosomes, which can be used to capture sex-specific differences.
CCC also provides a powerful approach to detect complex relationships between a subset of samples or conditions, such as differences between health and disease.
We found that top CCC-ranked gene pairs in whole blood from GTEx were replicated in independent tissue-specific networks trained from multiple data types and attributed to cell lineages from blood, even though CCC had no access to cell lineage-specific information.
This suggests that it can detect intricate cell lineage-specific transcriptional patterns missed by linear-only coefficients.
Compared to Spearman and Pearson, the CCC was more similar to Spearman in terms of robustness to outliers, but faster to compute.
Additionally, CCC can process categorical variables together with numerical values.
It is conceptually easy to interpret and has a single parameter that controls the maximum complexity of the detected relationships while also balancing compute time.


Datasets such as Anscombe or "Datasaurus" show that visual analysis is essential to uncover patterns in data.
However, for many datasets, examining each possible relationship is infeasible.
This is where more sophisticated and robust correlation coefficients, such as CCC, can be used.
CCC can help identify complex patterns that might reflect heterogeneity in samples and mask clear relationships between variables.
For example, genes *UTY* and *KDM6A* (from sex chromosomes) have a strong linear relationship, but only in a subset of samples (males).
This highlights the importance of considering sex as a biological variable (SABV) [@doi:10.1038/509282a] to avoid overlooking differences between men and women, for instance, in disease manifestations [@doi:10.1210/endrev/bnaa034; @doi:10.1038/s41593-021-00806-8].
A not-only-linear correlation coefficient like CCC could also identify significant differences between variables (such as genes) that are explained by a third factor, which would be entirely missed by linear-only coefficients.


It is well-known that biomedical research focuses on a small fraction of human genes [@pmid:17620606; @pmid:17472739].
For example, some genes highlighted in CCC-ranked pairs (Figure @fig:upsetplot_coefs b), such as *SDS* (12q24) and *ZDHHC12* (9q34), were found to have fewer publications than expected [@pmid:30226837].
This suggests that the use of linear correlation coefficients may bias researchers away from genes with complex coexpression patterns.
A beyond-linear gene co-expression analysis on large datasets may reveal the function of understudied genes.
For example, genes *KLHL21* (1p36) and *AC068580.6* (*ENSG00000235027*, in 11p15) have a high CCC value and are missed by other coefficients.
*KLHL21* has been suggested as a potential therapeutic target for hepatocellular carcinoma [@pmid:27769251] and other cancers [@pmid:29574153; @pmid:35084622].
Its nonlinear correlation with *AC068580.6* may uncover other important players in cancer initiation or progression, especially in subsets of samples with specific characteristics (as suggested in Figure @fig:upsetplot_coefs b).


Not-only-linear correlation coefficients could be useful in genetic studies.
Genome-wide association studies (GWAS) have been successful in understanding the molecular basis of common diseases by estimating the relationship between genotype and phenotype [@doi:10.1016/j.ajhg.2017.06.005].
However, the estimated effect sizes of genes identified with GWAS are generally modest and explain only a fraction of the phenotype variance, hindering clinical translation of these findings [@doi:10.1038/s41576-019-0127-1].
The omnigenic model for complex traits [@pmid:28622505; @pmid:31051098] suggests that this is due to highly-interconnected gene regulatory networks, with some core genes having a more direct effect on the phenotype than others.
We and others [@doi:10.1101/2021.07.05.450786; @doi:10.1186/s13040-020-00216-9; @doi:10.1101/2021.10.21.21265342] have shown that integrating gene co-expression networks into genetic studies can help to identify core genes that are missed by linear-only models like GWAS.
Our results suggest that more advanced and efficient correlation coefficients, such as CCC, could better estimate gene co-expression profiles, leading to more accurate identification of core genes.
This could be beneficial in the precision medicine field, as it could provide the computational tools for focusing on more promising genes, which could represent better candidate drug targets.


Our analysis has some limitations.
We worked with a sample of the top variable genes to keep computation time feasible.
Although CCC is much faster than MIC, Pearson and Spearman remain the most computationally efficient since they only involve simple data statistics.
Our results, however, demonstrate the advantages of using more advanced coefficients such as CCC for detecting and studying more intricate molecular mechanisms that replicate in independent datasets.
Applying CCC to larger compendia, such as recount3 [@pmid:34844637] with thousands of heterogeneous samples across different conditions, can reveal other potentially meaningful gene interactions.
The single parameter of CCC, $k_{\mathrm{max}}$, controls the maximum complexity of patterns found and also affects compute time.
Our analysis suggested that $k_{\mathrm{max}}=10$ was sufficient to identify both linear and more complex patterns in gene expression.
A more comprehensive analysis of optimal values for this parameter could provide insights on how to adjust it for different applications or data types.


Linear and rank-based correlation coefficients are fast to calculate, but they cannot detect patterns that are not linear.
For example, patterns associated with sex are not identifiable with linear-only coefficients.
Not-only-linear methods, such as CCC, can identify these patterns, as well as intricate patterns from expression data alone.
Furthermore, CCC is highly parallelizable, and efficient GPU-based implementations could make it even faster.
This makes it an efficient, next-generation correlation coefficient that is highly effective in transcriptome analyses and potentially useful in a broad range of other domains.
