## Discussion

We introduce the Clustermatch Correlation Coefficient (CCC), an efficient machine learning-based statistic for detecting not-only-linear relationships.
Applying CCC to GTEx v8 revealed that it was robust to outliers and detected linear relationships as well as complex and biologically meaningful patterns that standard coefficients missed.
CCC was particularly effective at capturing nonlinear patterns from the sex chromosomes, highlighting its ability to detect sex-specific differences.
Furthermore, CCC was able to disentangle intricate cell lineage-specific transcriptional patterns missed by linear-only coefficients.
In addition to capturing nonlinear patterns, the CCC was more similar to Spearman than Pearson, indicating its robustness to outliers.
The CCC results were concordant with MIC, but much faster to compute and therefore practical for large datasets.
Moreover, CCC can process both categorical and numerical values.
Finally, CCC has a single parameter that controls the maximum complexity of the detected relationships while also balancing compute time, making it conceptually easy to interpret.


Visual analysis can be helpful for datasets, however, for many datasets examining each possible relationship is infeasible.
In this case, more sophisticated and robust correlation coefficients are needed.
CCC is an advanced yet interpretable coefficient that can focus human interpretation on patterns that are more likely to reflect real biology.
For example, genes *UTY* - *KDM6A* (from sex chromosomes) have a strong linear relationship, but only in males.
This highlights the importance of considering sex as a biological variable (SABV) [@doi:10.1038/509282a], as it can help to identify differences between men and women in disease manifestations [@doi:10.1210/endrev/bnaa034; @doi:10.1038/s41593-021-00806-8].
A not-only-linear correlation coefficient like CCC can also identify significant differences between variables (such as genes) that are explained by a third factor, which would be entirely missed by linear-only coefficients.


Biomedical research is known to be biased towards a small fraction of human genes [@pmid:17620606; @pmid:17472739].
For example, some genes highlighted in CCC-ranked pairs (Figure @fig:upsetplot_coefs b), such as *SDS* (12q24) and *ZDHHC12* (9q34), have been found to have fewer publications than expected [@pmid:30226837].
This suggests that the widespread use of linear coefficients may lead researchers to overlook genes with complex coexpression patterns.
A beyond-linear gene co-expression analysis on large datasets could provide insight into the function of these understudied genes.
For example, *KLHL21* (1p36) and *AC068580.6* (*ENSG00000235027*, in 11p15) have a high CCC value and are missed by the other coefficients.
*KLHL21* has been suggested as a potential therapeutic target for hepatocellular carcinoma [@pmid:27769251] and other cancers [@pmid:29574153; @pmid:35084622].
Its nonlinear correlation with *AC068580.6* might reveal other important players in cancer initiation or progression, particularly in samples with specific characteristics (as suggested in Figure @fig:upsetplot_coefs b).


Not-only-linear correlation coefficients may be beneficial in the field of genetic studies.
For instance, genome-wide association studies (GWAS) have been successful in understanding the molecular basis of common diseases by estimating the association between genotype and phenotype [@doi:10.1016/j.ajhg.2017.06.005].
However, the estimated effect sizes of genes identified with GWAS are generally modest, and they explain only a fraction of the phenotype variance, hindering clinical translation of these findings [@doi:10.1038/s41576-019-0127-1].
The omnigenic model for complex traits [@pmid:28622505; @pmid:31051098] proposes that this is due to highly-interconnected gene regulatory networks, with some core genes having a more direct effect on the phenotype than others.
We and others [@doi:10.1101/2021.07.05.450786; @doi:10.1186/s13040-020-00216-9; @doi:10.1101/2021.10.21.21265342] have demonstrated that integrating gene co-expression networks into genetic studies could potentially identify core genes that are missed by linear-only models like GWAS.
Our results suggest that using more advanced and efficient correlation coefficients to build these networks could better estimate gene co-expression profiles and thus more accurately identify these core genes.
Such approaches, like CCC, could potentially be beneficial in precision medicine by providing computational tools to focus on more promising genes representing potential better candidate drug targets.


Our analysis has some limitations.
We used a sample with the top variable genes to keep computation time feasible.
Although CCC is much faster than MIC, Pearson and Spearman remain the most computationally efficient, as they rely on simple data statistics.
However, our results demonstrate the advantages of using more advanced coefficients such as CCC to detect and study more intricate molecular mechanisms that are replicated in independent datasets.
Applying CCC to larger compendia, such as recount3 [@pmid:34844637] with thousands of heterogeneous samples across different conditions, may reveal other meaningful gene interactions.
The single parameter of CCC, $k_{\mathrm{max}}$, controls the maximum complexity of patterns found and also impacts the compute time.
Our analysis suggested that $k_{\mathrm{max}}=10$ is sufficient to identify both linear and more complex patterns in gene expression.
A more comprehensive analysis of optimal values for this parameter could provide insights to adjust it for different applications or data types.


Linear and rank-based correlation coefficients are fast to calculate, however they cannot identify patterns that are not linear.
For example, sex differences are not apparent to linear-only coefficients.
Not-only-linear methods, such as CCC, can identify patterns driven by other factors and disentangle intricate patterns from expression data alone.
CCC can be made even faster with efficient GPU-based implementations.
Thus, CCC is an efficient, next-generation correlation coefficient that is highly effective in transcriptome analyses and potentially useful in a range of other domains.
