## Discussion

We introduce the Clustermatch Correlation Coefficient (CCC), an efficient machine learning-based statistic for detecting not-only-linear relationships.
Applying CCC to the GTEx v8 dataset revealed that it was robust to outliers and detected both linear relationships and complex, biologically meaningful patterns that standard coefficients missed.
In particular, CCC alone detected gene pairs with complex nonlinear patterns from the sex chromosomes, highlighting the utility of not-only-linear coefficients in capturing sex-specific differences.
This capability, however, extends beyond sex differences, as CCC can detect complex relationships when a subset of samples or conditions are explained by other factors, such as differences between health and disease.
CCC-ranked gene pairs in whole blood from GTEx were replicated in independent tissue-specific networks trained from multiple data types and attributed to cell lineages from blood, even though CCC did not have access to any cell lineage-specific information.
This suggests that CCC can identify intricate cell lineage-specific transcriptional patterns that linear-only coefficients miss.
In addition to capturing nonlinear patterns, the CCC was more similar to Spearman than Pearson in terms of robustness to outliers.
Furthermore, CCC is much faster to compute than MIC and can also process categorical variables together with numerical values.
CCC is conceptually easy to interpret and has a single parameter that controls the maximum complexity of the detected relationships while also balancing compute time.


Datasets such as Anscombe or "Datasaurus" show the importance of visual analysis instead of relying on simple data summaries.
However, for many datasets it is infeasible to examine each possible relationship and this is where more sophisticated and robust correlation coefficients are necessary.
CCC is an advanced yet interpretable coefficient that can help humans identify patterns that are more likely to reflect real biology.
For example, genes *UTY* - *KDM6A* (from sex chromosomes) were detected by CCC to have a strong linear relationship but only in a subset of samples (males).
This highlights the importance of considering sex as a biological variable (SABV) [@doi:10.1038/509282a] to avoid overlooking differences between men and women, such as in disease manifestations [@doi:10.1210/endrev/bnaa034; @doi:10.1038/s41593-021-00806-8].
A not-only-linear correlation coefficient like CCC could identify significant differences between variables (such as genes) that are explained by a third factor (beyond sex differences), which would be missed by linear-only coefficients.


It is well-known that biomedical research is biased towards studying a small fraction of human genes [@pmid:17620606; @pmid:17472739].
This bias is highlighted in the CCC-ranked pairs (Figure @fig:upsetplot_coefs b), such as *SDS* (12q24) and *ZDHHC12* (9q34), which were previously found to be the focus of fewer than expected publications [@pmid:30226837].
It is possible that the widespread use of linear coefficients may lead researchers to overlook genes with complex coexpression patterns.
A beyond-linear gene co-expression analysis on large compendia could help identify the function of understudied genes.
For example, gene *KLHL21* (1p36) and *AC068580.6* (*ENSG00000235027*, in 11p15) have a high CCC value and are missed by the other coefficients.
*KLHL21* has been suggested as a potential therapeutic target for hepatocellular carcinoma [@pmid:27769251] and other cancers [@pmid:29574153; @pmid:35084622].
Its nonlinear correlation with *AC068580.6* could help to uncover other important players in cancer initiation or progression, potentially in subsets of samples with specific characteristics (as suggested in Figure @fig:upsetplot_coefs b).


Not-only-linear correlation coefficients could be useful in the field of genetics.
Genome-wide association studies (GWAS) have been successful in understanding the molecular basis of common diseases by estimating the association between genotype and phenotype [@doi:10.1016/j.ajhg.2017.06.005].
However, the estimated effect sizes of genes identified with GWAS are generally modest, and they explain only a fraction of the phenotype variance, hindering the clinical translation of these findings [@doi:10.1038/s41576-019-0127-1].
The omnigenic model for complex traits [@pmid:28622505; @pmid:31051098] proposes that this is due to highly-interconnected gene regulatory networks, with some core genes having a more direct effect on the phenotype than others.
We and others [@doi:10.1101/2021.07.05.450786; @doi:10.1186/s13040-020-00216-9; @doi:10.1101/2021.10.21.21265342] have shown that integrating gene co-expression networks in genetic studies could potentially help to identify these core genes, which are missed by linear-only models such as GWAS.
Our results suggest that more advanced and efficient correlation coefficients, such as CCC, could better estimate gene co-expression profiles and thus more accurately identify these core genes.
This could potentially play a significant role in the field of precision medicine, by providing the computational tools to focus on more promising genes that could be better candidate drug targets.


Our analyses have some limitations.
We worked on a sample with the top variable genes to keep computation time feasible.
Although the correlation coefficient (CCC) we developed is much faster than the Maximal Information Coefficient (MIC), Pearson and Spearman coefficients are still the most computationally efficient since they rely on simple data statistics.
Our results, however, reveal the advantages of using more advanced coefficients like CCC for detecting and studying more intricate molecular mechanisms that are replicated in independent datasets.
The application of CCC on larger compendia, such as recount3 [@pmid:34844637], which contains thousands of heterogeneous samples across different conditions, can reveal other potentially meaningful gene interactions.
The single parameter of CCC, $k_{\mathrm{max}}$, controls the maximum complexity of patterns found and also impacts the compute time.
Our analysis suggested that $k_{\mathrm{max}}=10$ was sufficient to identify both linear and more complex patterns in gene expression.
A more comprehensive analysis of optimal values for this parameter could provide insights to adjust it for different applications or data types.


Linear and rank-based correlation coefficients are quick to calculate, however, they are not able to detect all the patterns found in biological datasets.
For example, sex-related patterns are not detected by linear-only coefficients, but they can be identified by not-only-linear methods.
Not-only-linear coefficients can also identify complex patterns in expression data that are replicated in models integrating different data modalities.
The CCC correlation coefficient is particularly efficient and can be made even faster with GPU-based implementations.
This next-generation correlation coefficient is highly effective in transcriptome analyses and can also be used in other domains.
