### Tissue-specific network analyses using GIANT {#sec:giant}

We accessed tissue-specific gene networks of GIANT using both the web interface and web services provided by HumanBase [@url:https://hb.flatironinstitute.org].
The GIANT version used in this study included 987 genome-scale datasets with approximately 38,000 conditions from around 14,000 publications.
Details on how these networks were built are described in [@doi:10.1038/ng.3259].
Briefly, tissue-specific gene networks were built using gene expression data (without GTEx samples [@url:https://hb.flatironinstitute.org/data]) from the NCBI's Gene Expression Omnibus (GEO) [@doi:10.1093/nar/gks1193], protein-protein interaction (BioGRID [@pmc:PMC3531226], IntAct [@doi:10.1093/nar/gkr1088], MINT [@doi:10.1093/nar/gkr930] and MIPS [@pmc:PMC148093]), transcription factor regulation using binding motifs from JASPAR [@doi:10.1093/nar/gkp950], and chemical and genetic perturbations from MSigDB [@doi:10.1073/pnas.0506580102].
Gene expression data were log-transformed and the Pearson correlation coefficient was computed for each gene pair, normalized using the Fisher's z transform, and z-scores discretized into different bins.
Gold standards for tissue-specific functional relationships were built using expert curation and experimentally derived gene annotations from the Gene Ontology.
Then, one naive Bayesian classifier (using C++ implementations from the Sleipnir library [@pmid:18499696]) was trained for each of the 144 tissues using these gold standards.
Finally, these classifiers were used to estimate the probability of tissue-specific interactions for each gene pair, which was used to calculate an efficient not-only-linear correlation coefficient (Equation \ref{eq:nonlinear_corr}).

\begin{equation}
    r_{non-linear} = \frac{\sum_{i=1}^{n} \sum_{j=1}^{n} p_{ij}x_iy_j}{\sqrt{\sum_{i=1}^{n}x_i^2\sum_{j=1}^{n}y_j^2}} \label{eq:nonlinear_corr}
\end{equation}

Here, $r_{non-linear}$ is the non-only-linear correlation coefficient, $p_{ij}$ is the probability of a tissue-specific functional relationship between genes $i$ and $j$, and $x_i$ and $y_j$ are the expression values of genes $i$ and $j$, respectively.


We then computed an efficient not-only-linear correlation coefficient (ENLCC) between the two gene pairs using the Equation (@eq:enlcc).

For each pair of genes prioritized in our study using GTEx, we used GIANT through HumanBase to obtain 1) a predicted gene network for blood (manually selected to match whole blood in GTEx) and 2) a gene network with an automatically predicted tissue using the method described in [@doi:10.1101/gr.155697.113].
This tissue prediction approach trains a machine learning model using comprehensive transcriptional data with human-curated markers of different cell lineages (e.g., macrophages) as gold standards.
The models are then used to predict other cell lineage-specific genes.
In addition to reporting this predicted tissue or cell lineage, we computed the average probability of interaction between all genes in the network retrieved from GIANT.
Following the default procedure used in GIANT, we included the top 15 genes with the highest probability of interaction with the queried gene pair for each network.
We then computed an efficient not-only-linear correlation coefficient (ENLCC) between the two gene pairs using the following equation:

$$ENLCC=\frac{\sum_{i=1}^{n} (x_i-\bar{x})(y_i-\bar{y})}{\sqrt{\sum_{i=1}^{n}(x_i-\bar{x})^2}\sqrt{\sum_{i=1}^{n}(y_i-\bar{y})^2}} {#eq:enlcc}$$
