### Tissue-specific network analyses using GIANT {#sec:giant}

We used the GIANT database to access tissue-specific gene networks.
This version of GIANT included 987 genome-scale datasets with 38,000 conditions from 14,000 publications.
We obtained gene expression data from the NCBI Gene Expression Omnibus, protein-protein interactions from BioGRID, IntAct, MINT and MIPS, transcription factor regulation from JASPAR, and chemical and genetic perturbations from MSigDB.
The gene expression data were log-transformed and the Pearson correlation was computed for each gene pair, normalized via the Fisher's z transform, and z-scores discretized into different bins.
Gold standards for tissue-specific functional relationships were built using expert curation and experimentally derived gene annotations from the Gene Ontology.
We used naive Bayesian classifiers (from the Sleipnir library) to estimate the probability of tissue-specific interactions for each gene pair.


For each gene pair prioritized in our study using GTEx, we used GIANT through HumanBase to obtain a predicted gene network for blood (specifically matching whole blood in GTEx) and an automatically predicted gene network for a tissue.
This tissue was predicted using a machine learning model trained on comprehensive transcriptional data and human-curated markers of different cell lineages (e.g., macrophages).
The model was then used to predict other cell lineage-specific genes.
In addition to the predicted tissue or cell lineage, we calculated the average probability of interaction between all genes in the network retrieved from GIANT.
Following the default procedure used in GIANT, we included the top 15 genes with the highest probability of interaction with the queried gene pair for each network.
