### Tissue-specific network analyses using GIANT {#sec:giant}

We accessed tissue-specific gene networks of GIANT using both the web interface and web services provided by HumanBase.
We used the GIANT version that included 987 genome-scale datasets with approximately 38,000 conditions from around 14,000 publications.
Details on how these networks were built can be found in the publication [@doi:10.1038/ng.3259].
We used gene expression data from the NCBI's Gene Expression Omnibus (GEO) [@doi:10.1093/nar/gks1193], protein-protein interaction data from BioGRID [@pmc:PMC3531226], IntAct [@doi:10.1093/nar/gkr1088], MINT [@doi:10.1093/nar/gkr930] and MIPS [@pmc:PMC148093], transcription factor regulation data from the JASPAR database [@doi:10.1093/nar/gkp950], and chemical and genetic perturbations from MSigDB [@doi:10.1073/pnas.0506580102].
We log-transformed the gene expression data and computed the Pearson correlation for each gene pair.
The z-scores were then normalized using the Fisher's z transform and discretized into different bins.
We built gold standards for tissue-specific functional relationships using expert curation and experimentally derived gene annotations from the Gene Ontology.
We trained one naive Bayesian classifier (using C++ implementations from the Sleipnir library [@pmid:18499696]) for each of the 144 tissues using these gold standards.
Finally, we used these classifiers to estimate the probability of tissue-specific interactions for each gene pair.


For each pair of genes prioritized in our study using GTEx, we obtained a predicted gene network for blood (manually selected to match whole blood in GTEx) and an automatically predicted tissue network using a machine learning method provided by HumanBase web interfaces/services.
This method trains a model using transcriptional data and human-curated markers of different cell lineages as gold standards, which are then used to predict other cell lineage-specific genes.
In addition to reporting this predicted tissue or cell lineage, we computed the average probability of interaction between all genes in the network retrieved from GIANT.
We included the top 15 genes with the highest probability of interaction with the queried gene pair for each network, following the default procedure used in GIANT.
