### Tissue-specific network analyses using GIANT {#sec:giant}

We accessed tissue-specific gene networks of GIANT using the web interface and web services provided by HumanBase [@url:https://hb.flatironinstitute.org].
This version included 987 genome-scale datasets with approximately 38,000 conditions from around 14,000 publications.
The networks were built using gene expression data (without GTEx samples [@url:https://hb.flatironinstitute.org/data]) from NCBI's Gene Expression Omnibus (GEO) [@doi:10.1093/nar/gks1193], protein-protein interactions (BioGRID [@pmc:PMC3531226], IntAct [@doi:10.1093/nar/gkr1088], MINT [@doi:10.1093/nar/gkr930] and MIPS [@pmc:PMC148093]), transcription factor regulation using binding motifs from JASPAR [@doi:10.1093/nar/gkp950], and chemical and genetic perturbations from MSigDB [@doi:10.1073/pnas.0506580102].
The gene expression data were log-transformed, and Pearson correlation was computed for each gene pair, normalized using Fisher's z-transform, and z-scores discretized into different bins.
Gold standards for tissue-specific functional relationships were built using expert curation and experimentally derived gene annotations from the Gene Ontology.
A naive Bayesian classifier (C++ implementations from the Sleipnir library [@pmid:18499696]) was trained for each of the 144 tissues using these gold standards.
Finally, these classifiers were used to estimate the probability of tissue-specific interactions for each gene pair.
Further details on the building of these networks are described in [@doi:10.1038/ng.3259].


For each gene pair prioritized in our study using GTEx, we used GIANT and HumanBase to obtain a predicted gene network for blood and an automatically predicted gene network for a specific cell lineage.
This cell lineage prediction was done by training a machine learning model using comprehensive transcriptional data, with human-curated markers of different cell lineages as gold standards.
We then used these models to predict other cell lineage-specific genes.
In addition to reporting the predicted tissue or cell lineage, we computed the average probability of interaction between all genes in the network retrieved from GIANT.
We included the top 15 genes with the highest probability of interaction with the queried gene pair for each network, following the default procedure used in GIANT.
