### Tissue-specific network analyses using GIANT {#sec:giant}

We accessed tissue-specific gene networks of GIANT using both the web interface and web services provided by HumanBase [@url:https://hb.flatironinstitute.org].
The version of GIANT used in this study included 987 genome-scale datasets with 38,000 conditions from 14,000 publications.
We used gene expression data (without GTEx samples [@url:https://hb.flatironinstitute.org/data]) from the NCBI's Gene Expression Omnibus (GEO) [@doi:10.1093/nar/gks1193], protein-protein interaction data from BioGRID [@pmc:PMC3531226], IntAct [@doi:10.1093/nar/gkr1088], MINT [@doi:10.1093/nar/gkr930] and MIPS [@pmc:PMC148093], transcription factor regulation using binding motifs from JASPAR [@doi:10.1093/nar/gkp950], and chemical and genetic perturbations from MSigDB [@doi:10.1073/pnas.0506580102] to build tissue-specific gene networks. 

We log-transformed gene expression data, computed Pearson correlation for each gene pair, normalized it using the Fisher's z transform, and discretized z-scores into different bins.
Gold standards for tissue-specific functional relationships were created using expert curation and experimentally derived gene annotations from the Gene Ontology.
We then trained one naive Bayesian classifier (using C++ implementations from the Sleipnir library [@pmid:18499696]) for each of the 144 tissues using these gold standards.
Finally, these classifiers were used to estimate the probability of tissue-specific interactions for each gene pair.
Details on how these networks were built are described in [@doi:10.1038/ng.3259].


For each pair of genes prioritized in our study using GTEx, we obtained a predicted gene network for blood (manually selected to match whole blood in GTEx) and a gene network with an automatically predicted tissue from HumanBase web interfaces/services.
The tissue prediction approach trains a machine learning model using comprehensive transcriptional data with human-curated markers of different cell lineages (e.g., macrophages) as gold standards.
Then, these models are used to predict other cell lineage-specific genes.
In addition to reporting the predicted tissue or cell lineage, we computed the average probability of interaction between all genes in the network retrieved from GIANT.
Following the default procedure used in GIANT, we included the top 15 genes with the highest probability of interaction with the queried gene pair for each network.
