### Tissue-specific network analyses using GIANT {#sec:giant}

We used gene networks from the GIANT database, which included 987 genome-scale datasets from 14,000 publications.
The networks were built using gene expression data from the NCBI's Gene Expression Omnibus, protein-protein interaction data from BioGRID, IntAct, MINT and MIPS, transcription factor regulation from JASPAR, and chemical and genetic perturbations from MSigDB.
The data was log-transformed, and the Pearson correlation was computed and normalized using the Fisher's z transform.
Gold standards for tissue-specific functional relationships were built using expert curation and experimentally derived gene annotations from the Gene Ontology.
We then used one naive Bayesian classifier (implemented in C++ from the Sleipnir library) for each of the 144 tissues, trained with the gold standards, to estimate the probability of tissue-specific interactions for each gene pair.


For each pair of genes prioritized in our study using GTEx, we used GIANT through HumanBase to obtain a gene network and the average probability of interaction between all genes in the network.
For the gene network, we manually selected one that matches whole blood in GTEx, and for the probability of interaction, we used the method described in [@doi:10.1101/gr.155697.113], which uses a machine learning model trained with human-curated markers of different cell lineages (e.g., macrophages) as gold standards.
We included the top 15 genes with the highest probability of interaction with the queried gene pair for each network.
