<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Milton Pividori" />
  <meta name="author" content="Marylyn D. Ritchie" />
  <meta name="author" content="Diego H. Milone" />
  <meta name="author" content="Casey S. Greene" />
  <meta name="dcterms.date" content="2024-08-12" />
  <meta name="keywords" content="correlation coefficient, nonlinear relationships, gene expression" />
  <title>An efficient not-only-linear correlation coefficient based on clustering</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/main/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta property="og:type" content="article" />
  <meta name="dc.title" content="An efficient not-only-linear correlation coefficient based on clustering" />
  <meta name="citation_title" content="An efficient not-only-linear correlation coefficient based on clustering" />
  <meta property="og:title" content="An efficient not-only-linear correlation coefficient based on clustering" />
  <meta property="twitter:title" content="An efficient not-only-linear correlation coefficient based on clustering" />
  <meta name="dc.date" content="2024-08-12" />
  <meta name="citation_publication_date" content="2024-08-12" />
  <meta property="article:published_time" content="2024-08-12" />
  <meta name="dc.modified" content="2024-08-12T21:43:52+00:00" />
  <meta property="article:modified_time" content="2024-08-12T21:43:52+00:00" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="Milton Pividori" />
  <meta name="citation_author_institution" content="Department of Biomedical Informatics, University of Colorado School of Medicine, Aurora, CO 80045, USA" />
  <meta name="citation_author_institution" content="Department of Genetics, Perelman School of Medicine, University of Pennsylvania, Philadelphia, PA 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0002-3035-4403" />
  <meta name="twitter:creator" content="@miltondp" />
  <meta name="citation_author" content="Marylyn D. Ritchie" />
  <meta name="citation_author_institution" content="Department of Genetics, Perelman School of Medicine, University of Pennsylvania, Philadelphia, PA 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0002-1208-1720" />
  <meta name="twitter:creator" content="@MarylynRitchie" />
  <meta name="citation_author" content="Diego H. Milone" />
  <meta name="citation_author_institution" content="Research Institute for Signals, Systems and Computational Intelligence (sinc(i)), Universidad Nacional del Litoral, Consejo Nacional de Investigaciones Científicas y Técnicas (CONICET), Santa Fe CP3000, Argentina" />
  <meta name="citation_author_orcid" content="0000-0003-2182-4351" />
  <meta name="twitter:creator" content="@d1001" />
  <meta name="citation_author" content="Casey S. Greene" />
  <meta name="citation_author_institution" content="Center for Health AI, University of Colorado School of Medicine, Aurora, CO 80045, USA" />
  <meta name="citation_author_institution" content="Department of Biomedical Informatics, University of Colorado School of Medicine, Aurora, CO 80045, USA" />
  <meta name="citation_author_orcid" content="0000-0001-8713-9213" />
  <meta name="twitter:creator" content="@GreeneScientist" />
  <link rel="canonical" href="https://greenelab.github.io/ccc-manuscript/" />
  <meta property="og:url" content="https://greenelab.github.io/ccc-manuscript/" />
  <meta property="twitter:url" content="https://greenelab.github.io/ccc-manuscript/" />
  <meta name="citation_fulltext_html_url" content="https://greenelab.github.io/ccc-manuscript/" />
  <meta name="citation_pdf_url" content="https://greenelab.github.io/ccc-manuscript/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://greenelab.github.io/ccc-manuscript/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://greenelab.github.io/ccc-manuscript/v/1b71ed008e4c7b962a64754ba2c9a013060ff779/" />
  <meta name="manubot_html_url_versioned" content="https://greenelab.github.io/ccc-manuscript/v/1b71ed008e4c7b962a64754ba2c9a013060ff779/" />
  <meta name="manubot_pdf_url_versioned" content="https://greenelab.github.io/ccc-manuscript/v/1b71ed008e4c7b962a64754ba2c9a013060ff779/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header id="title-block-header">
<h1 class="title">An efficient not-only-linear correlation coefficient based on clustering</h1>
</header>
<p><em>A DOI-citable version of this manuscript is available at <a href="https://doi.org/10.1101/2022.06.15.496326" class="uri">https://doi.org/10.1101/2022.06.15.496326</a></em>.</p>
<!-- 
<small><em>
This manuscript
([permalink](https://greenelab.github.io/ccc-manuscript/v/1b71ed008e4c7b962a64754ba2c9a013060ff779/))
was automatically generated
from [greenelab/ccc-manuscript@1b71ed0](https://github.com/greenelab/ccc-manuscript/tree/1b71ed008e4c7b962a64754ba2c9a013060ff779)
on August 12, 2024.
</em></small> -->
<h2 id="authors">Authors</h2>
<ul>
<li><p><strong>Milton Pividori</strong>
<sup><a href="#correspondence">✉</a></sup><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-3035-4403">0000-0002-3035-4403</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/miltondp">miltondp</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/miltondp">miltondp</a>
· <img src="images/mastodon.svg" class="inline_icon" width="16" height="16" alt="Mastodon icon" />
<a href="https://genomic.social/@miltondp">@miltondp@genomic.social</a>
<br>
<small>
Department of Biomedical Informatics, University of Colorado School of Medicine, Aurora, CO 80045, USA; Department of Genetics, Perelman School of Medicine, University of Pennsylvania, Philadelphia, PA 19104, USA
· Funded by The Gordon and Betty Moore Foundation GBMF 4552; The National Human Genome Research Institute (R01 HG010067); The National Human Genome Research Institute (K99/R00 HG011898); The Eunice Kennedy Shriver National Institute of Child Health and Human Development (R01 HD109765)
</small></p></li>
<li><p><strong>Marylyn D. Ritchie</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-1208-1720">0000-0002-1208-1720</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/MarylynRitchie">MarylynRitchie</a>
<br>
<small>
Department of Genetics, Perelman School of Medicine, University of Pennsylvania, Philadelphia, PA 19104, USA
</small></p></li>
<li><p><strong>Diego H. Milone</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-2182-4351">0000-0003-2182-4351</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/dmilone">dmilone</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/d1001">d1001</a>
<br>
<small>
Research Institute for Signals, Systems and Computational Intelligence (sinc(i)), Universidad Nacional del Litoral, Consejo Nacional de Investigaciones Científicas y Técnicas (CONICET), Santa Fe CP3000, Argentina
</small></p></li>
<li><p><strong>Casey S. Greene</strong>
<sup><a href="#correspondence">✉</a></sup><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-8713-9213">0000-0001-8713-9213</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/cgreene">cgreene</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/GreeneScientist">GreeneScientist</a>
· <img src="images/mastodon.svg" class="inline_icon" width="16" height="16" alt="Mastodon icon" />
<a href="https://genomic.social/@greenescientist">@greenescientist@genomic.social</a>
<br>
<small>
Center for Health AI, University of Colorado School of Medicine, Aurora, CO 80045, USA; Department of Biomedical Informatics, University of Colorado School of Medicine, Aurora, CO 80045, USA
· Funded by The Gordon and Betty Moore Foundation (GBMF 4552); The National Human Genome Research Institute (R01 HG010067); The National Cancer Institute (R01 CA237170); The Eunice Kennedy Shriver National Institute of Child Health and Human Development (R01 HD109765)
</small></p></li>
</ul>
<div id="correspondence">
<p>✉ — Correspondence possible via <a href="https://github.com/greenelab/ccc-manuscript/issues">GitHub Issues</a>
or email to
Milton Pividori &lt;milton.pividori@cuanschutz.edu&gt;,
Casey S. Greene &lt;casey.s.greene@cuanschutz.edu&gt;.</p>
</div>
<h2 class="page_break_before" id="abstract">Abstract</h2>
<p>Correlation coefficients are widely used to identify patterns in data that may be of particular interest.
In transcriptomics, genes with correlated expression often share functions or are part of disease-relevant biological processes.
Here we introduce the Clustermatch Correlation Coefficient (CCC), an efficient, easy-to-use and not-only-linear coefficient based on clustering.
CCC assumes that if two features are similar, then the clustering of objects using each feature separately should match.
We show that CCC reveals biologically meaningful linear and nonlinear patterns missed by standard, linear-only correlation coefficients.
Moreover, CCC is much faster than state-of-the-art coefficients such as the Maximal Information Coefficient.
When applied to human gene expression data, CCC identifies robust linear relationships while detecting nonlinear patterns associated, for example, with sex differences that are not captured by linear-only coefficients.
Gene pairs highly ranked by CCC were enriched for interactions in integrated networks built from protein-protein interaction, transcription factor regulation, and chemical and genetic perturbations, suggesting that CCC could detect functional relationships that linear-only methods missed.
CCC is a highly-efficient, next-generation not-only-linear correlation coefficient that can readily be applied to genome-scale data and other domains across different data types.</p>
<h2 id="introduction">Introduction</h2>
<p>New technologies have vastly improved data collection, generating a deluge of information across different disciplines.
This large amount of data provides new opportunities to address unanswered scientific questions, provided we have efficient tools capable of identifying multiple types of underlying patterns.
Correlation analysis is an essential statistical technique for discovering relationships between variables<span class="citation" data-cites="DUeOcxpt"><sup><a href="#ref-DUeOcxpt" role="doc-biblioref">1</a></sup></span>.
Correlation coefficients are often used in exploratory data mining techniques, such as clustering or community detection algorithms, to compute a similarity value between a pair of objects of interest such as genes<span class="citation" data-cites="yvIastvc"><sup><a href="#ref-yvIastvc" role="doc-biblioref">2</a></sup></span> or disease-relevant lifestyle factors<span class="citation" data-cites="14q47vZTA"><sup><a href="#ref-14q47vZTA" role="doc-biblioref">3</a></sup></span>.
These coefficients are also used in supervised tasks, for example, for feature selection to improve prediction accuracy<span class="citation" data-cites="iZjSr8s2 1FQ1FgsCm"><sup><a href="#ref-iZjSr8s2" role="doc-biblioref">4</a>,<a href="#ref-1FQ1FgsCm" role="doc-biblioref">5</a></sup></span>.
The Pearson correlation coefficient is ubiquitously deployed across application domains and diverse scientific areas.
Thus, even minor and significant improvements in these techniques could have enormous consequences in industry and research.</p>
<p>In transcriptomics, many analyses start with estimating the correlation between genes.
More sophisticated approaches built on correlation analysis can suggest gene function<span class="citation" data-cites="16aFivIcr"><sup><a href="#ref-16aFivIcr" role="doc-biblioref">6</a></sup></span>, aid in discovering common and cell lineage-specific regulatory networks<span class="citation" data-cites="q5QZW3xH"><sup><a href="#ref-q5QZW3xH" role="doc-biblioref">7</a></sup></span>, and capture important interactions in a living organism that can uncover molecular mechanisms in other species<span class="citation" data-cites="17eqTELga x9k6ERo9"><sup><a href="#ref-17eqTELga" role="doc-biblioref">8</a>,<a href="#ref-x9k6ERo9" role="doc-biblioref">9</a></sup></span>.
The analysis of large RNA-seq datasets<span class="citation" data-cites="vsBvj38s TryC3OjB"><sup><a href="#ref-vsBvj38s" role="doc-biblioref">10</a>,<a href="#ref-TryC3OjB" role="doc-biblioref">11</a></sup></span> can also reveal complex transcriptional mechanisms underlying human diseases<span class="citation" data-cites="yvIastvc zwyRqHIg 13mfjo6Kw 12H2nr6Rc qAH2cGOH"><sup><a href="#ref-yvIastvc" role="doc-biblioref">2</a>,<a href="#ref-zwyRqHIg" role="doc-biblioref">12</a>–<a href="#ref-qAH2cGOH" role="doc-biblioref">15</a></sup></span>.
Since the introduction of the omnigenic model of complex traits<span class="citation" data-cites="g7Id81Dt sfMulPrl"><sup><a href="#ref-g7Id81Dt" role="doc-biblioref">16</a>,<a href="#ref-sfMulPrl" role="doc-biblioref">17</a></sup></span>, gene-gene relationships are playing an increasingly important role in genetic studies of human diseases<span class="citation" data-cites="1EN7sv6pQ NM3rHx1i 17f9RJ1ul sGgkLsyy"><sup><a href="#ref-1EN7sv6pQ" role="doc-biblioref">18</a>–<a href="#ref-sGgkLsyy" role="doc-biblioref">21</a></sup></span>, even in specific fields such as polygenic risk scores<span class="citation" data-cites="14ZMJkeR2"><sup><a href="#ref-14ZMJkeR2" role="doc-biblioref">22</a></sup></span>.
In this context, recent approaches combine disease-associated genes from genome-wide association studies (GWAS) with gene co-expression networks to prioritize “core” genes directly affecting diseases<span class="citation" data-cites="57TjOMEA NM3rHx1i 17f9RJ1ul"><sup><a href="#ref-NM3rHx1i" role="doc-biblioref">19</a>,<a href="#ref-17f9RJ1ul" role="doc-biblioref">20</a>,<a href="#ref-57TjOMEA" role="doc-biblioref">23</a></sup></span>.
These core genes are not captured by standard statistical methods but are believed to be part of highly-interconnected, disease-relevant regulatory networks.
Therefore, advanced correlation coefficients could immediately find wide applications across many areas of biology, including the prioritization of candidate drug targets in the precision medicine field.</p>
<p>The Pearson and Spearman correlation coefficients are widely used because they reveal intuitive relationships and can be computed quickly.
However, they are designed to capture linear or monotonic patterns (referred to as linear-only) and may miss complex yet critical relationships.
Novel coefficients have been proposed as metrics that capture nonlinear patterns such as the Maximal Information Coefficient (MIC)<span class="citation" data-cites="tbcQL0eR"><sup><a href="#ref-tbcQL0eR" role="doc-biblioref">24</a></sup></span> and the Distance Correlation (DC)<span class="citation" data-cites="IvOGGmxI"><sup><a href="#ref-IvOGGmxI" role="doc-biblioref">25</a></sup></span>.
MIC, in particular, is one of the most commonly used statistics to capture more complex relationships, with successful applications across several domains<span class="citation" data-cites="NnsqWXcp g8HrFUfP iZjSr8s2"><sup><a href="#ref-iZjSr8s2" role="doc-biblioref">4</a>,<a href="#ref-NnsqWXcp" role="doc-biblioref">26</a>,<a href="#ref-g8HrFUfP" role="doc-biblioref">27</a></sup></span>.
However, the computational complexity makes them impractical for even moderately sized datasets<span class="citation" data-cites="NnsqWXcp ztrPHUjX"><sup><a href="#ref-NnsqWXcp" role="doc-biblioref">26</a>,<a href="#ref-ztrPHUjX" role="doc-biblioref">28</a></sup></span>.
Recent implementations of MIC, for example, take several seconds to compute on a single variable pair across a few thousand objects or conditions<span class="citation" data-cites="NnsqWXcp"><sup><a href="#ref-NnsqWXcp" role="doc-biblioref">26</a></sup></span>.
We previously developed a clustering method for highly diverse datasets that outperformed approaches based on Pearson, Spearman, DC and MIC in detecting clusters of simulated linear and nonlinear relationships with varying noise levels<span class="citation" data-cites="VCRLONE"><sup><a href="#ref-VCRLONE" role="doc-biblioref">29</a></sup></span>.</p>
<p>Here we introduce the Clustermatch Correlation Coefficient (CCC), an efficient not-only-linear coefficient that works across quantitative and qualitative variables.
CCC has a single parameter that limits the maximum complexity of relationships found (from linear to more general patterns) and computation time.
CCC provides a high level of flexibility to detect specific types of patterns that are more important for the user, while providing safe defaults to capture general relationships.
We also provide an efficient CCC implementation that is highly parallelizable, allowing to speed up computation across variable pairs with millions of objects or conditions.
To assess its performance, we applied our method to gene expression data from the Genotype-Tissue Expression v8 (GTEx) project across different tissues<span class="citation" data-cites="9Pr9idng"><sup><a href="#ref-9Pr9idng" role="doc-biblioref">30</a></sup></span>.
CCC captured both strong linear relationships and novel nonlinear patterns, which were entirely missed by standard coefficients.
For example, some of these nonlinear patterns were associated with sex differences in gene expression, suggesting that CCC can capture strong relationships present only in a subset of samples.
We also found that the CCC behaves similarly to MIC in several cases, although it is much faster to compute.
Gene pairs detected in expression data by CCC had higher interaction probabilities in tissue-specific gene networks from the Genome-wide Analysis of gene Networks in Tissues (GIANT)<span class="citation" data-cites="CVF61Un5"><sup><a href="#ref-CVF61Un5" role="doc-biblioref">31</a></sup></span>.
Furthermore, its ability to efficiently handle diverse data types (including numerical and categorical features) reduces preprocessing steps and makes it appealing to analyze large and heterogeneous repositories.</p>
<h2 id="results">Results</h2>
<h3 id="overview-of-ccc-a-not-only-linear-correlation-coefficient">Overview of CCC: a not-only-linear correlation coefficient</h3>
<div id="fig:datasets_rel" class="fignos">
<figure>
<img src="images/intro/relationships.svg" style="width:100.0%" alt="Figure 1: Different patterns across data types. Each panel contains a set of simulated datasets described by two generic variables. a) The Anscombe’s quartet with four different datasets (from Anscombe I to IV) with numerical variables x and y, and 11 data points; b) Four datasets with 100 data points; c) Two datasets with categorical variables w (with values “Orange” and “Blue”) and z (with values “A”, “B” and “C”), and 100 data points; d) Two datasets with categorical and numerical variables, and 100 data points. Each panel shows the correlation value using: Pearson (p) and Spearman (s) for numerical variables only, and CCC (c) for both numerical and categorical; their statistical significance is indicated with asterisks: P&lt;0.05 (*), P&lt;0.01 (**), and P&lt;0.001 (***). For CCC, we computed the p-value using 10,000 permutations. Vertical and horizontal red lines show how CCC clustered data points using x and y, respectively. For categorical variables, CCC uses the categories to cluster data points." />
<figcaption aria-hidden="true"><span>Figure 1:</span> <strong>Different patterns across data types.</strong>
Each panel contains a set of simulated datasets described by two generic variables.
<strong>a)</strong> The Anscombe’s quartet with four different datasets (from Anscombe I to IV) with numerical variables <span class="math inline">\(x\)</span> and <span class="math inline">\(y\)</span>, and 11 data points;
<strong>b)</strong> Four datasets with 100 data points;
<strong>c)</strong> Two datasets with categorical variables <span class="math inline">\(w\)</span> (with values “Orange” and “Blue”) and <span class="math inline">\(z\)</span> (with values “A”, “B” and “C”), and 100 data points;
<strong>d)</strong> Two datasets with categorical and numerical variables, and 100 data points.
Each panel shows the correlation value using: Pearson (<span class="math inline">\(p\)</span>) and Spearman (<span class="math inline">\(s\)</span>) for numerical variables only, and CCC (<span class="math inline">\(c\)</span>) for both numerical and categorical;
their statistical significance is indicated with asterisks: <span class="math inline">\(P&lt;0.05\)</span> (*), <span class="math inline">\(P&lt;0.01\)</span> (**), and <span class="math inline">\(P&lt;0.001\)</span> (***).
For CCC, we computed the p-value using 10,000 permutations.
Vertical and horizontal red lines show how CCC clustered data points using <span class="math inline">\(x\)</span> and <span class="math inline">\(y\)</span>, respectively.
For categorical variables, CCC uses the categories to cluster data points.</figcaption>
</figure>
</div>
<p>The CCC provides a similarity measure between any pair of variables, either with numerical or categorical values.
The method assumes that if there is a relationship between two variables/features describing <span class="math inline">\(n\)</span> data points/objects, then the <strong>cluster</strong>ings of those objects using each variable should <strong>match</strong>.
In the case of numerical values, CCC uses quantiles to efficiently separate data points into different clusters (e.g., the median separates numerical data into two clusters).
For categorical values, CCC uses the categories themselves to separate data points into different clusters (e.g., if feature “color” has three values, “red”, “green”, and “blue”, then data will be clustered into three clusters defined by those colors).
Once all clusterings are generated according to each variable, we define the CCC as the maximum adjusted Rand index (ARI)<span class="citation" data-cites="e4AuoW8N"><sup><a href="#ref-e4AuoW8N" role="doc-biblioref">32</a></sup></span> between them, ranging between 0 and 1.
Details of the CCC algorithm can be found in <a href="#sec:ccc_algo">Methods</a>.</p>
<p>We examined how the Pearson (<span class="math inline">\(p\)</span>), Spearman (<span class="math inline">\(s\)</span>) and CCC (<span class="math inline">\(c\)</span>) correlation coefficients behaved on different simulated data patterns.
Figure <a href="#fig:datasets_rel">1</a> shows different types of relationships between two variables of different data types, where <span class="math inline">\(x\)</span> and <span class="math inline">\(y\)</span> are numerical and <span class="math inline">\(w\)</span> and <span class="math inline">\(z\)</span> are categorical.
For each variable pair, we show the coefficient values and their statistical significance, where asterisks indicate different <span class="math inline">\(P\)</span>-values (<span class="math inline">\(P\)</span>).
The red lines show how CCC clustered numerical data points using <span class="math inline">\(x\)</span> (vertical lines) and <span class="math inline">\(y\)</span> (horizontal lines).</p>
<p>In Figure <a href="#fig:datasets_rel">1</a>a, we examine the classic Anscombe’s quartet<span class="citation" data-cites="WP7yhZGS"><sup><a href="#ref-WP7yhZGS" role="doc-biblioref">33</a></sup></span>, which comprises four synthetic datasets with different patterns but the same data statistics (mean, standard deviation and Pearson’s correlation).
This kind of simulated data, recently revisited with the “Datasaurus”<span class="citation" data-cites="LW7cVCGa dvyMwrz9 FhufXP6P"><sup><a href="#ref-LW7cVCGa" role="doc-biblioref">34</a>–<a href="#ref-FhufXP6P" role="doc-biblioref">36</a></sup></span>, is used as a reminder of the importance of going beyond simple statistics, where either undesirable patterns (such as outliers) or desirable ones (such as biologically meaningful nonlinear relationships) can be masked by summary statistics alone.
Anscombe I contains a noisy but clear linear pattern, similar to Anscombe III where the linearity is perfect besides one outlier.
In these two examples, CCC separates data points using two clusters (one red line for each variable <span class="math inline">\(x\)</span> and <span class="math inline">\(y\)</span>), yielding a statistically significant value of <span class="math inline">\(1.0\)</span> (the maximum for CCC) and thus indicating a strong relationship.
Anscombe II seems to follow a partially quadratic relationship interpreted as linear by Pearson and Spearman.
In contrast, for this potentially undersampled quadratic pattern, CCC yields a lower and not statistically significant value of <span class="math inline">\(0.34\)</span>, reflecting a more complex relationship than a linear pattern.
Anscombe IV shows a vertical line of data points where <span class="math inline">\(x\)</span> values are almost constant except for one outlier.
This outlier does not influence CCC (which correctly identifies no relationship) as it does for Pearson or Spearman, although only Pearson yields a statistically significant result.
Thus <span class="math inline">\(c=0.00\)</span> (the minimum value) correctly indicates no association for this variable pair because, besides the outlier, for a single value of <span class="math inline">\(x\)</span> there are ten different values for <span class="math inline">\(y\)</span>.
This pair of variables does not fit the CCC assumption: the two clusters formed with <span class="math inline">\(x\)</span> (approximately separated by <span class="math inline">\(x=13\)</span>) do not match the three clusters formed with <span class="math inline">\(y\)</span>.
The Pearson’s correlation coefficient is the same across all these Anscombe’s examples (<span class="math inline">\(p=0.82\)</span>), whereas Spearman is <span class="math inline">\(0.50\)</span> or greater.</p>
<p>We also simulated additional types of numerical relationships (Figure <a href="#fig:datasets_rel">1</a>b), including some previously described from gene expression data<span class="citation" data-cites="yCHLC0oS 1AMcxTw9R 16ddFzCqe"><sup><a href="#ref-yCHLC0oS" role="doc-biblioref">37</a>–<a href="#ref-16ddFzCqe" role="doc-biblioref">39</a></sup></span>.
For the random/independent pair of variables, all coefficients correctly agree with a value close to zero and <span class="math inline">\(P&gt;0.05\)</span>.
The non-coexistence pattern, captured by all coefficients, represents a case where one gene (<span class="math inline">\(x\)</span>) might be expressed while the other one (<span class="math inline">\(y\)</span>) is inhibited, highlighting a potentially strong biological relationship (such as a microRNA negatively regulating another gene).
For the other two examples (quadratic and two-lines), only CCC is able to yield a high and statistically significant correlation value, whereas Pearson and Spearman fail to capture these nonlinear patterns.
These relationships also show how CCC uses different degrees of complexity to capture the relationships.
For the quadratic pattern, for example, CCC separates <span class="math inline">\(x\)</span> into more clusters (four in this case) to reach the maximum ARI.
The two-lines example shows two embedded linear relationships with different slopes, which neither Pearson nor Spearman detect (<span class="math inline">\(p=-0.12\)</span> and <span class="math inline">\(s=0.05\)</span>, respectively, both with <span class="math inline">\(P&gt;0.05\)</span>).
Here, CCC increases the complexity of the model by using eight clusters for <span class="math inline">\(x\)</span> and six for <span class="math inline">\(y\)</span>, resulting in <span class="math inline">\(c=0.31\)</span> (<span class="math inline">\(P&lt;0.001\)</span>).</p>
<p>Furthermore, we also simulated categorical variables, which only CCC can handle.
Figure <a href="#fig:datasets_rel">1</a>c shows two patterns between variables <span class="math inline">\(w\)</span> (with categories “Orange” and “Blue”) and <span class="math inline">\(z\)</span> (with categories “A”, “B” and “C”).
The first case (Two-Categorical I) represents a random/independent pattern where categorical values in one variable are approximately uniformly distributed across the categorical values of the other variable.
Here, as expected, CCC yield a very low and non-significant value.
In the second case (Two-Categorical II), the category “Blue” of <span class="math inline">\(w\)</span> is overrepresented in data points with <span class="math inline">\(z\)</span> equal to “A” and, less strongly, the category “Orange” of <span class="math inline">\(w\)</span> is overrepresented in data points with <span class="math inline">\(z\)</span> equal to “B”.
In this case, since CCC clusters data points using the categorical values, it detects that clusters of data points with <span class="math inline">\(w\)</span>=“Blue” match clusters with <span class="math inline">\(z\)</span>=“A”, yielding a statistically significant <span class="math inline">\(c=0.21\)</span>.
Figure <a href="#fig:datasets_rel">1</a>d mixes a categorical variable (<span class="math inline">\(z\)</span>) with a numerical one (<span class="math inline">\(y\)</span>).
The first case (Categorical-Numerical I) represents a random/independent pattern where numerical values in <span class="math inline">\(y\)</span> are approximately uniformly distributed across the categorical values in <span class="math inline">\(z\)</span>.
Similarly as in the other random/independent cases, CCC yields a very low and non-significant value, since the clusters formed by <span class="math inline">\(y\)</span> do not match the clusters (given by the categorical values) formed by <span class="math inline">\(z\)</span>.
Conversely, in the second case (Categorical-Numerical II), clusters of data points with similar values in <span class="math inline">\(y\)</span> tend to have also similar categorical values in <span class="math inline">\(z\)</span>.
In this example, for data points with <span class="math inline">\(z\)</span>=“A”, we assigned <span class="math inline">\(y \sim \mathcal{N}(0, 0.5^2)\)</span>, whereas for <span class="math inline">\(z\)</span>=“B” and “C”, we assigned <span class="math inline">\(y \sim \mathcal{N}(1, 0.25^2)\)</span> and <span class="math inline">\(y \sim \mathcal{N}(1, 0.75^2)\)</span>, respectively.
Here, CCC uses <span class="math inline">\(y\)</span> values to group data points into two clusters, and these clusters match the clusters obtained from <span class="math inline">\(z\)</span>, yielding a statistically significant <span class="math inline">\(c=0.30\)</span>.</p>
<h3 id="the-ccc-reveals-linear-and-nonlinear-patterns-in-human-transcriptomic-data">The CCC reveals linear and nonlinear patterns in human transcriptomic data</h3>
<p>We next examined the characteristics of these correlation coefficients in gene expression data from GTEx v8 across different tissues.
For our initial analyses, we selected the top 5,000 genes with the largest variance on whole blood and then computed the correlation matrix between genes using Pearson, Spearman and CCC (see <a href="#sec:data_gtex">Methods</a>).
Although we always considered the statistical significance of the coefficients, we focused on the strength of the association (i.e., the coefficient value) for our analyses.</p>
<p>We examined the distribution of each coefficient’s absolute values in GTEx (Figure <a href="#fig:dist_coefs">2</a>).
CCC (mean=0.14, median=0.08, sd=0.15) has a much more skewed distribution than Pearson (mean=0.31, median=0.24, sd=0.24) and Spearman (mean=0.39, median=0.37, sd=0.26).
The coefficients reach a cumulative set containing 70% of gene pairs at different values (Figure <a href="#fig:dist_coefs">2</a>b), <span class="math inline">\(c=0.18\)</span>, <span class="math inline">\(p=0.44\)</span> and <span class="math inline">\(s=0.56\)</span>, suggesting that for this type of data, the coefficients are not directly comparable by magnitude, so we used ranks for further comparisons.
In GTEx v8, CCC values were closer to Spearman than either was to Pearson (Figure <a href="#fig:dist_coefs">2</a>c).
We also compared with the Maximal Information Coefficient (MIC) (see <a href="#sec:mic">Methods</a>), another advanced, not-only-linear correlation coefficient that has been successfully used in various application domains<span class="citation" data-cites="NnsqWXcp g8HrFUfP iZjSr8s2"><sup><a href="#ref-iZjSr8s2" role="doc-biblioref">4</a>,<a href="#ref-NnsqWXcp" role="doc-biblioref">26</a>,<a href="#ref-g8HrFUfP" role="doc-biblioref">27</a></sup></span>.
We found that CCC behaved very similarly to MIC, although CCC was up to two orders of magnitude faster to run (see <a href="#sec:time_test">Methods</a>).
These results suggest that our findings for CCC generalize to MIC, therefore, in the subsequent analyses we focus on CCC and linear-only coefficients.</p>
<div id="fig:dist_coefs" class="fignos">
<figure>
<img src="images/coefs_comp/gtex_whole_blood/dist-main.svg" style="width:100.0%" alt="Figure 2: Distribution of coefficient values on gene expression (GTEx v8, whole blood). a) Histogram of coefficient values. b) Corresponding cumulative histogram. The dotted line maps the coefficient value that accumulates 70% of gene pairs. c) 2D histogram plot with hexagonal bins between all coefficients, where a logarithmic scale was used to color each hexagon." />
<figcaption aria-hidden="true"><span>Figure 2:</span> <strong>Distribution of coefficient values on gene expression (GTEx v8, whole blood).</strong>
<strong>a)</strong> Histogram of coefficient values.
<strong>b)</strong> Corresponding cumulative histogram. The dotted line maps the coefficient value that accumulates 70% of gene pairs.
<strong>c)</strong> 2D histogram plot with hexagonal bins between all coefficients, where a logarithmic scale was used to color each hexagon.</figcaption>
</figure>
</div>
<p>A closer inspection of gene pairs that were either prioritized or disregarded by these coefficients revealed that they captured different patterns.
We analyzed the agreements and disagreements by obtaining, for each coefficient, the top 30% of gene pairs with the largest correlation values (“high” set) and the bottom 30% (“low” set), resulting in six potentially overlapping categories (Supplementary Files 1 and 2).
For most cases (76.4%), an UpSet analysis<span class="citation" data-cites="ZhJ7VK5R"><sup><a href="#ref-ZhJ7VK5R" role="doc-biblioref">40</a></sup></span> (Figure <a href="#fig:upsetplot_coefs">3</a>a) showed that the three coefficients agreed on whether there is a strong correlation (42.1%) or there is no relationship (34.3%).
Since Pearson and Spearman are linear-only, and CCC can also capture these patterns, we expect that these concordant gene pairs represent clear linear patterns.
CCC and Spearman agree more on either highly or poorly correlated pairs (4.0% in “high”, and 7.0% in “low”) than any of these with Pearson (all between 0.3%-3.5% for “high”, and 2.8%-5.5% for “low”).
In summary, CCC agrees with either Pearson or Spearman in 90.5% of gene pairs by assigning a high or a low correlation value.</p>
<div id="fig:upsetplot_coefs" class="fignos">
<figure>
<img src="images/coefs_comp/gtex_whole_blood/upsetplot-main.svg" style="width:100.0%" alt="Figure 3: Intersection of gene pairs with high and low correlation coefficient values (GTEx v8, whole blood). a) UpSet plot with six categories (rows) grouping the 30% of the highest (green triangle) and lowest (red triangle) values for each coefficient. Columns show different intersections of categories grouped by agreements and disagreements. b) Hexagonal binning plots with examples of gene pairs where CCC (c) disagrees with Pearson (p) and Spearman (s). For each method, colors in the triangles indicate if the gene pair is among the top (green) or bottom (red) 30% of coefficient values. No triangle means that the correlation value for the gene pair is between the 30th and 70th percentiles (neither low nor high). The statistical significance is indicated with asterisks using the False Discovery Rate (FDR) adjusted P-values, calculated using the Benjamini and Hochberg method41: \mathrm{FDR}&lt;0.05 (*), \mathrm{FDR}&lt;0.01 (**), and \mathrm{FDR}&lt;0.001 (***). A logarithmic scale was used to color each hexagon." />
<figcaption aria-hidden="true"><span>Figure 3:</span> <strong>Intersection of gene pairs with high and low correlation coefficient values (GTEx v8, whole blood).</strong>
<strong>a)</strong> UpSet plot with six categories (rows) grouping the 30% of the highest (green triangle) and lowest (red triangle) values for each coefficient.
Columns show different intersections of categories grouped by agreements and disagreements.
<strong>b)</strong> Hexagonal binning plots with examples of gene pairs where CCC (<span class="math inline">\(c\)</span>) disagrees with Pearson (<span class="math inline">\(p\)</span>) and Spearman (<span class="math inline">\(s\)</span>).
For each method, colors in the triangles indicate if the gene pair is among the top (green) or bottom (red) 30% of coefficient values.
No triangle means that the correlation value for the gene pair is between the 30th and 70th percentiles (neither low nor high).
The statistical significance is indicated with asterisks using the False Discovery Rate (FDR) adjusted <span class="math inline">\(P\)</span>-values, calculated using the Benjamini and Hochberg method<span class="citation" data-cites="UeFCwwR7"><sup><a href="#ref-UeFCwwR7" role="doc-biblioref">41</a></sup></span>: <span class="math inline">\(\mathrm{FDR}&lt;0.05\)</span> (*), <span class="math inline">\(\mathrm{FDR}&lt;0.01\)</span> (**), and <span class="math inline">\(\mathrm{FDR}&lt;0.001\)</span> (***).
A logarithmic scale was used to color each hexagon.</figcaption>
</figure>
</div>
<p>While there was broad agreement, more than 20,000 gene pairs with a high CCC value were not highly ranked by the other coefficients (“Disagreements” group on the right of Figure <a href="#fig:upsetplot_coefs">3</a>a).
There were also gene pairs with a high Pearson value and either low CCC (1,075), low Spearman (87) or both low CCC and low Spearman values (531).
No gene pairs were found to have a high Spearman value and a low CCC.
Considering the correlation values and their statistical significance, we analyzed gene pairs among the top ten of each intersection in the “Disagreements” group (Figure <a href="#fig:upsetplot_coefs">3</a>a, right) where CCC disagrees with Pearson, Spearman or both.</p>
<div id="fig:gtex_tissues:kdm6a_uty" class="fignos">
<figure>
<img src="images/coefs_comp/kdm6a_vs_uty/gtex-KDM6A_vs_UTY-main.svg" style="width:95.0%" alt="Figure 4: The expression levels of KDM6A and UTY display sex-specific associations across GTEx tissues. CCC captures this nonlinear relationship in all GTEx tissues (nine examples are shown in the first three rows), except in female-specific organs (last row)." />
<figcaption aria-hidden="true"><span>Figure 4:</span> <strong>The expression levels of <em>KDM6A</em> and <em>UTY</em> display sex-specific associations across GTEx tissues.</strong>
CCC captures this nonlinear relationship in all GTEx tissues (nine examples are shown in the first three rows), except in female-specific organs (last row).</figcaption>
</figure>
</div>
<p>The first two gene pairs at the top of Figure <a href="#fig:upsetplot_coefs">3</a>b (<em>IFNG</em> - <em>SDS</em>, with high CCC and Spearman, and low Pearson; <em>PRSS36</em> - <em>CCL18</em>, with high CCC and low Pearson) appear to follow a non-coexistence relationship: in samples where one of the genes is highly expressed, the other is slightly activated, suggesting a potentially inhibiting effect.
The following four gene pairs (<em>UTY</em> - <em>KDM6A</em>, <em>DDX3Y</em> - <em>KDM6A</em>, <em>RASSF2</em> - <em>CYTIP</em>, and <em>AC068580.6</em> - <em>KLHL21</em>) follow patterns combining either two linear or one linear and one independent relationships.
In particular, genes <em>UTY</em> - <em>KDM6A</em> (paralogs) and <em>DDX3Y</em> - <em>KDM6A</em> show a nonlinear relationship where a subset of samples follows a robust linear pattern and another subset has a constant (independent) expression of one gene.
The relationships in these two gene pairs are explained by sex differences in expression: <em>UTY</em> and <em>DDX3Y</em> are in chromosome Y (Yq11) whereas <em>KDM6A</em> is in chromosome X (Xp11), and therefore samples with a linear pattern are males, whereas those with no expression for <em>UTY</em> or <em>DDX3Y</em> are females.
Furthermore, for this sex-specific gene pair pattern, CCC yields a statistically significant coefficient value in 45 out of 47 tissues in GTEx, except for female-specific organs (Figure <a href="#fig:gtex_tissues:kdm6a_uty">4</a> and <a href="#fig:gtex_tissues:kdm6a_ddx3y">S1</a>, and Supplementary File 3).
The gene pair <em>RASSF2</em> - <em>CYTIP</em> was replicated in an independent dataset as we explain later.
Even though we have not found a biological explanation for gene pair <em>AC068580.6</em> - <em>KLHL21</em> (there is limited information about <em>AC068580.6</em>, ENSG00000235027, a long non-coding RNA), its strong nonlinear connection with <em>KLHL21</em> (linked with some cancers<span class="citation" data-cites="9i58Tfs4"><sup><a href="#ref-9i58Tfs4" role="doc-biblioref">42</a></sup></span>) is robustly captured by CCC only.
Notably, these four gene pairs contain strong linear relationships and CCC is the only coefficient able to consistently capture these nonlinear patterns across a variety of tissues with a statistically significant and high correlation value.
Pearson and Spearman show a statistically significant correlation value for some of these gene pairs, although these values are low and would very likely not be prioritized for further research.
In addition, these two linear-only coefficients are unable to robustly capture the same pattern in other tissues (Figure <a href="#fig:gtex_tissues:kdm6a_uty">4</a> and <a href="#fig:gtex_tissues:kdm6a_ddx3y">S1</a>, and Supplementary File 3).
For instance, although the three coefficients are statistically significant in whole blood for the gene pair <em>UTY</em> - <em>KDM6A</em>, Pearson and Spearman fail to capture the same pattern in the brain cerebellum, and in many cases, such as small intestine, the sign of the coefficient is negative despite the strong positive linear correlation among male samples (Figure <a href="#fig:gtex_tissues:kdm6a_uty">4</a>).</p>
<p>Finally, the last two gene pairs in Figure <a href="#fig:upsetplot_coefs">3</a>b are highly ranked by Pearson, but not by CCC or Spearman.
Although all coefficients are significant for the gene pair <em>MYOZ1</em> - <em>TNNI2</em>, the low CCC (<span class="math inline">\(c=0.03\)</span>) and moderate Spearman (<span class="math inline">\(s=0.28\)</span>) contrast with Pearson’s (<span class="math inline">\(p=0.97\)</span>), suggesting a statistically significant but very weak linear relationship.
The high and statistically significant Pearson value for <em>SCGB3A1</em> - <em>C19orf33</em> seems to be driven by outliers.</p>
<h3 id="replication-of-gene-associations-using-tissue-specific-gene-networks-from-giant">Replication of gene associations using tissue-specific gene networks from GIANT</h3>
<p>We sought to systematically analyze discrepant scores to assess whether associations were replicated in other datasets besides GTEx.
This is challenging and prone to bias because linear-only correlation coefficients are usually used in gene co-expression analyses.
Therefore, we used 144 tissue-specific gene networks from the Genome-wide Analysis of gene Networks in Tissues (GIANT) project<span class="citation" data-cites="9ZXy7Ut2 ZSXf7Pdq"><sup><a href="#ref-9ZXy7Ut2" role="doc-biblioref">43</a>,<a href="#ref-ZSXf7Pdq" role="doc-biblioref">44</a></sup></span>, where nodes represent genes and each edge a functional relationship weighted with a probability of interaction between two genes (see <a href="#sec:giant">Methods</a>).
The version of GIANT used in this study did not include GTEx samples<span class="citation" data-cites="BgCQnLU6"><sup><a href="#ref-BgCQnLU6" role="doc-biblioref">45</a></sup></span>, making it an ideal case for replication.
These networks were built from expression and different interaction measurements, including protein-interaction, transcription factor regulation, chemical/genetic perturbations and microRNA target profiles from the Molecular Signatures Database (MSigDB<span class="citation" data-cites="tt0Vdf4C"><sup><a href="#ref-tt0Vdf4C" role="doc-biblioref">46</a></sup></span>).
We reasoned that statistically significant and highly-ranked gene pairs using three different coefficients in a single tissue (whole blood in GTEx, Figure <a href="#fig:upsetplot_coefs">3</a>) that represented real patterns should often replicate in a corresponding tissue or related cell lineage using the multi-cell type functional interaction networks in GIANT.
In addition to predicting a network with interactions for a pair of genes, the GIANT web application can also automatically detect a relevant tissue or cell type where genes are predicted to be specifically expressed (the approach uses a machine learning method introduced in<span class="citation" data-cites="mk4SLpDo"><sup><a href="#ref-mk4SLpDo" role="doc-biblioref">47</a></sup></span> and described in <a href="#sec:giant">Methods</a>).</p>
<p>As an example of our evaluation procedure, we obtained the networks in blood and the automatically-predicted cell type for gene pairs <em>RASSF2</em> - <em>CYTIP</em> (strong nonlinear pattern and CCC high, Figure <a href="#fig:giant_gene_pairs">5</a>a) and <em>MYOZ1</em> - <em>TNNI2</em> (weak linear pattern and Pearson high, Figure <a href="#fig:giant_gene_pairs">5</a>b).
In addition to the gene pair, the networks include other genes connected according to their probability of interaction (up to 15 additional genes are shown), which allows estimating whether genes are part of the same tissue-specific biological process.
Two large black nodes in each network’s top-left and bottom-right corners represent our gene pairs.
A green edge means a close-to-zero probability of interaction, whereas a red edge represents a strong predicted relationship between the two genes.
In this example, genes <em>RASSF2</em> and <em>CYTIP</em> (Figure <a href="#fig:giant_gene_pairs">5</a>a), with a high CCC value (<span class="math inline">\(c=0.20\)</span>, above the 73th percentile) and low Pearson and Spearman (<span class="math inline">\(p=0.16\)</span> and <span class="math inline">\(s=0.11\)</span>, below the 38th and 17th percentiles, respectively), were both strongly connected to the blood network, with interaction scores of at least 0.69 and an average of 0.77 and 0.85, respectively (Supplementary Table <a href="#tbl:giant:weights">S1</a>).
The autodetected cell type for this pair was leukocytes, and interaction scores were similar to the blood network (Supplementary Table <a href="#tbl:giant:weights">S1</a>).
However, genes <em>MYOZ1</em> and <em>TNNI2</em>, with a very high Pearson value (<span class="math inline">\(p=0.97\)</span>), moderate Spearman (<span class="math inline">\(s=0.28\)</span>) and very low CCC (<span class="math inline">\(c=0.03\)</span>), were predicted to belong to much less cohesive networks (Figure <a href="#fig:giant_gene_pairs">5</a>b), with average interaction scores of 0.17 and 0.22 with the rest of the genes, respectively.
Additionally, the autodetected cell type (skeletal muscle) is not related to blood or one of its cell lineages.
These preliminary results suggested that CCC might be capturing blood-specific patterns missed by the other coefficients.</p>
<div id="fig:giant_gene_pairs" class="fignos">
<figure>
<img src="images/coefs_comp/giant_networks/top_gene_pairs-main.svg" style="width:100.0%" alt="Figure 5: Analysis of GIANT tissue-specific predicted networks for gene pairs prioritized by correlation coefficients. a-b) Two gene pairs prioritized by correlation coefficients (from Figure 3b) with their predicted networks in blood (left) and an automatically selected tissue/cell type (right) using the method described in47. A node represents a gene and an edge the probability that two genes are part of the same biological process in a specific cell type. A maximum of 15 genes are shown for each network. The GIANT web application automatically determined a minimum interaction confidence (edges’ weights) to be shown. These networks can be analyzed online using the following links: RASSF2 - CYTIP48, MYOZ1 - TNNI249. c) Summary of predicted tissue/cell type networks for gene pairs exclusively prioritized by CCC and Pearson (\mathrm{FDR}&lt;0.05). The first row combines all gene pairs where CCC is high, and Pearson and Spearman are not. The second row combines all gene pairs where Pearson is high, and CCC and Spearman are not. Bar plots (left) show the number of gene pairs for each predicted tissue/cell type. Box plots (right) show the average probability of interaction between genes in these predicted tissue-specific networks. Red indicates CCC-only tissues/cell types, blue are Pearson-only, and purple are shared." />
<figcaption aria-hidden="true"><span>Figure 5:</span> <strong>Analysis of GIANT tissue-specific predicted networks for gene pairs prioritized by correlation coefficients.</strong>
<strong>a-b)</strong> Two gene pairs prioritized by correlation coefficients (from Figure <a href="#fig:upsetplot_coefs">3</a>b) with their predicted networks in blood (left) and an automatically selected tissue/cell type (right) using the method described in<span class="citation" data-cites="mk4SLpDo"><sup><a href="#ref-mk4SLpDo" role="doc-biblioref">47</a></sup></span>.
A node represents a gene and an edge the probability that two genes are part of the same biological process in a specific cell type.
A maximum of 15 genes are shown for each network.
The GIANT web application automatically determined a minimum interaction confidence (edges’ weights) to be shown.
These networks can be analyzed online using the following links:
<em>RASSF2</em> - <em>CYTIP</em><span class="citation" data-cites="FKQEHB1F"><sup><a href="#ref-FKQEHB1F" role="doc-biblioref">48</a></sup></span>,
<em>MYOZ1</em> - <em>TNNI2</em><span class="citation" data-cites="JpfbR0un"><sup><a href="#ref-JpfbR0un" role="doc-biblioref">49</a></sup></span>.
<strong>c)</strong> Summary of predicted tissue/cell type networks for gene pairs exclusively prioritized by CCC and Pearson (<span class="math inline">\(\mathrm{FDR}&lt;0.05\)</span>).
The first row combines all gene pairs where CCC is high, and Pearson and Spearman are not.
The second row combines all gene pairs where Pearson is high, and CCC and Spearman are not.
Bar plots (left) show the number of gene pairs for each predicted tissue/cell type.
Box plots (right) show the average probability of interaction between genes in these predicted tissue-specific networks.
Red indicates CCC-only tissues/cell types, blue are Pearson-only, and purple are shared.</figcaption>
</figure>
</div>
<p>We next performed a systematic evaluation using the top 100 discrepant gene pairs between CCC and the other two coefficients.
For each gene pair prioritized in GTEx (whole blood), we autodetected a relevant cell type using GIANT to assess whether genes were predicted to be specifically expressed in a blood-relevant cell lineage.
For this, we used the top five most commonly autodetected cell types for each coefficient and assessed connectivity in the resulting networks (see <a href="#sec:giant">Methods</a>).
The top 5 predicted cell types for gene pairs highly ranked by CCC and not by the rest were all blood-specific (Figure <a href="#fig:giant_gene_pairs">5</a>c, top left), including macrophage, leukocyte, natural killer cell, blood and mononuclear phagocyte.
The average probability of interaction between genes in these CCC-ranked networks was significantly higher than the other coefficients (Figure <a href="#fig:giant_gene_pairs">5</a>c, top right), with all medians larger than 67% and first quartiles above 41% across predicted cell types.
In contrast, most Pearson’s gene pairs were predicted to be specific to tissues unrelated to blood (Figure <a href="#fig:giant_gene_pairs">5</a>c, bottom left), with skeletal muscle being the most commonly predicted tissue.
The interaction probabilities in these Pearson-ranked networks were also generally lower than in CCC, except for blood-specific gene pairs (Figure <a href="#fig:giant_gene_pairs">5</a>c, bottom right).</p>
<p>The associations exclusively detected by CCC in whole blood from GTEx were more strongly replicated in these independent networks that incorporated multiple data modalities.
CCC-ranked gene pairs not only had higher probabilities of belonging to the same biological process but were also predicted to be specifically expressed in blood cell lineages.
Conversely, most Pearson-ranked gene pairs were not predicted to be blood-specific, and their interaction probabilities were relatively much lower.
This lack of replication in GIANT suggests that top Pearson-exclusive-ranked gene pairs in GTEx might be driven mainly by outliers, which is consistent with our earlier observations of outlier-driven associations (Figure <a href="#fig:upsetplot_coefs">3</a>b).</p>
<h2 id="discussion">Discussion</h2>
<p>We introduce the Clustermatch Correlation Coefficient (CCC), an efficient not-only-linear clustering-based statistic.
Applying CCC to GTEx v8 revealed that it was robust to outliers and detected linear relationships as well as complex and biologically meaningful patterns that standard coefficients missed.
In particular, CCC alone detected gene pairs with complex nonlinear patterns from the sex chromosomes, highlighting the way that not-only-linear coefficients can play in capturing sex-specific differences.
The ability to capture these nonlinear patterns, however, extends beyond sex differences: it provides a powerful approach to detect potentially complex relationships where a subset of samples or conditions are explained by other factors (such as differences between health and disease).
We found that top CCC-ranked gene pairs in whole blood from GTEx were replicated in independent tissue-specific networks trained from multiple data types and attributed to cell lineages from blood, even though CCC did not have access to any cell lineage-specific information.
This suggests that CCC can disentangle intricate cell lineage-specific transcriptional patterns missed by linear-only coefficients.
In addition to capturing nonlinear patterns, the CCC was more similar to Spearman than Pearson, highlighting their shared robustness to outliers.
The CCC results were concordant with MIC, but much faster to compute and thus practical for large datasets.
Another advantage over MIC and standard coefficients is that CCC can also process categorical variables together with numerical values.
CCC is conceptually easy to interpret and has a single parameter that controls the maximum complexity of the detected relationships while also balancing compute time.</p>
<p>Datasets such as Anscombe or “Datasaurus” highlight the value of visualization instead of relying on simple data summaries.
While visual analysis is helpful, for many datasets examining each possible relationship is infeasible, and this is where more sophisticated and robust correlation coefficients are necessary.
Advanced yet interpretable coefficients like CCC can focus human interpretation on patterns that are more likely to reflect real biology.
The complexity of these patterns might reflect heterogeneity in samples that mask clear relationships between variables.
For example, genes <em>UTY</em> - <em>KDM6A</em> (from sex chromosomes), detected by CCC, have a strong linear relationship but only in a subset of samples (males), which was not captured by linear-only coefficients.
This example, in particular, highlights the importance of considering sex as a biological variable (SABV)<span class="citation" data-cites="4QgDZ8bF"><sup><a href="#ref-4QgDZ8bF" role="doc-biblioref">50</a></sup></span> to avoid overlooking important differences between men and women, for instance, in disease manifestations<span class="citation" data-cites="iox25BGw 1GpiBQk8d"><sup><a href="#ref-iox25BGw" role="doc-biblioref">51</a>,<a href="#ref-1GpiBQk8d" role="doc-biblioref">52</a></sup></span>.
More generally, a not-only-linear correlation coefficients that support categorical variables like CCC could identify significant differences between variables (such as genes) that are explained by a third factor (beyond sex differences), that would be entirely missed by linear-only coefficients.</p>
<p>It is well-known that biomedical research is biased towards a small fraction of human genes<span class="citation" data-cites="1CZbj0k1F 104MiBf8H"><sup><a href="#ref-1CZbj0k1F" role="doc-biblioref">53</a>,<a href="#ref-104MiBf8H" role="doc-biblioref">54</a></sup></span>.
Some genes highlighted in CCC-ranked pairs (Figure <a href="#fig:upsetplot_coefs">3</a>b), such as <em>SDS</em> (12q24) or <em>PRSS36</em> (16p11), were previously found to be the focus of fewer than expected publications<span class="citation" data-cites="IlPwHaTE"><sup><a href="#ref-IlPwHaTE" role="doc-biblioref">55</a></sup></span>.
It is possible that the widespread use of linear coefficients may bias researchers away from genes with complex coexpression patterns.
A beyond-linear gene co-expression analysis on large compendia might shed light on the function of understudied genes.
For example, gene <em>KLHL21</em> (1p36) and <em>AC068580.6</em> (a long non-coding RNA gene in 11p15) have a high CCC value and are missed by the other coefficients.
<em>KLHL21</em> was suggested as a potential therapeutic target for hepatocellular carcinoma<span class="citation" data-cites="9i58Tfs4"><sup><a href="#ref-9i58Tfs4" role="doc-biblioref">42</a></sup></span> and other cancers<span class="citation" data-cites="efGVmXWt 1FQgFkJoJ"><sup><a href="#ref-efGVmXWt" role="doc-biblioref">56</a>,<a href="#ref-1FQgFkJoJ" role="doc-biblioref">57</a></sup></span>.
Its nonlinear correlation with <em>AC068580.6</em> might unveil other important players in cancer initiation or progression, potentially in subsets of samples with specific characteristics (as suggested in Figure <a href="#fig:upsetplot_coefs">3</a>b).</p>
<p>Not-only-linear correlation coefficients might also be helpful in the field of genetic studies.
In this context, genome-wide association studies (GWAS) have been successful in understanding the molecular basis of common diseases by estimating the association between genotype and phenotype<span class="citation" data-cites="yqBcaXHD"><sup><a href="#ref-yqBcaXHD" role="doc-biblioref">58</a></sup></span>.
However, the estimated effect sizes of genes identified with GWAS are generally modest, and they explain only a fraction of the phenotype variance, hampering the clinical translation of these findings<span class="citation" data-cites="17B4AigZT"><sup><a href="#ref-17B4AigZT" role="doc-biblioref">59</a></sup></span>.
Recent theories, like the omnigenic model for complex traits<span class="citation" data-cites="g7Id81Dt sfMulPrl"><sup><a href="#ref-g7Id81Dt" role="doc-biblioref">16</a>,<a href="#ref-sfMulPrl" role="doc-biblioref">17</a></sup></span>, argue that these observations are explained by highly-interconnected gene regulatory networks, with some core genes having a more direct effect on the phenotype than others.
Using this omnigenic perspective, we and others<span class="citation" data-cites="NM3rHx1i 57TjOMEA 17f9RJ1ul"><sup><a href="#ref-NM3rHx1i" role="doc-biblioref">19</a>,<a href="#ref-17f9RJ1ul" role="doc-biblioref">20</a>,<a href="#ref-57TjOMEA" role="doc-biblioref">23</a></sup></span> have shown that integrating gene co-expression networks with genetic studies could potentially identify core genes that are missed by linear-only models alone like GWAS.
Our results suggest that building these networks with the latest approaches<span class="citation" data-cites="12OuQVjYy"><sup><a href="#ref-12OuQVjYy" role="doc-biblioref">60</a></sup></span> and advanced and efficient correlation coefficients could better estimate gene co-expression profiles and thus more accurately identify these core genes.
Approaches like CCC could play a significant role in the precision medicine field by providing the computational tools to focus on more promising genes representing potentially better candidate drug targets.</p>
<p>Our analyses have some limitations.
We worked on a sample with the top variable genes in a single tissue from GTEx to keep computation time feasible.
Although CCC is much faster than MIC, Pearson and Spearman are still the most computationally efficient since they only rely on simple data statistics.
Our results, however, reveal the advantages of using more advanced coefficients like CCC for detecting and studying more intricate molecular mechanisms that are replicated in independent datasets.
The application of CCC on larger compendia, such as recount3<span class="citation" data-cites="TryC3OjB"><sup><a href="#ref-TryC3OjB" role="doc-biblioref">11</a></sup></span> with thousands of heterogeneous samples across different conditions, can reveal other potentially meaningful gene interactions.
We compute <span class="math inline">\(P\)</span>-values using computationally intensive permutation tests; in the future, we plan to explore efficient permutation approaches such as those based on extreme value theory<span class="citation" data-cites="I46r0ipf"><sup><a href="#ref-I46r0ipf" role="doc-biblioref">61</a></sup></span>.
The single parameter of CCC, <span class="math inline">\(k_{\mathrm{max}}\)</span>, controls the maximum complexity of patterns found and also impacts the compute time.
Our analysis suggested that <span class="math inline">\(k_{\mathrm{max}}=10\)</span> was sufficient to identify both linear and more complex patterns in gene expression.
A more comprehensive analysis of optimal values for this parameter could provide insights to adjust it for different applications or data types.
Finally, computing the correlation between a gene pair represents only the first step of the analysis.
Controlling for known confounders, integrating with other data types, and replicating in independent datasets are some of the other important steps to ensure the biological relevance of the detected patterns.</p>
<p>While linear and rank-based correlation coefficients are exceptionally fast to calculate, not all relevant patterns in biological datasets are linear.
For example, patterns associated with sex as a biological variable are not apparent to the linear-only coefficients that we evaluated but are revealed by not-only-linear methods.
Beyond sex differences, being able to use a method that inherently identifies patterns driven by other factors is likely to be desirable.
Not-only-linear coefficients can also disentangle intricate yet relevant patterns from expression data alone that were replicated in models integrating different data modalities.
CCC, in particular, is highly parallelizable, and we anticipate efficient GPU-based implementations that could make it even faster.
The CCC is an efficient, next-generation correlation coefficient that is highly effective in transcriptome analyses and potentially useful in a broad range of other domains.</p>
<h2 id="acknowledgments">Acknowledgments</h2>
<p>This work has been supported by
the Gordon and Betty Moore Foundation (GBMF 4552 to C.S. Greene),
the National Human Genome Research Institute (R01 HG010067 to C.S. Greene, K99 HG011898 and R00 HG011898 to M. Pividori),
the National Cancer Institute (R01 CA237170 to C.S. Greene),
the Eunice Kennedy Shriver National Institute of Child Health and Human Development (R01 HD109765 to C.S. Greene).</p>
<h3 id="author-contributions">Author contributions</h3>
<p>C.S.G. and D.H.M. supervised the project.
M.P. conceived the study, developed the methodology, performed the analysis, and drafted the manuscript.
C.S.G., D.H.M. and M.D.R. provided critical feedback about the methodology, analyses performed and manuscript writing.</p>
<h3 id="declaration-of-interests">Declaration of interests</h3>
<p>The authors declare no competing interests.</p>
<h2 class="page_break_before" id="sec:methods">STAR★Methods</h2>
<h3 id="resource-availability">Resource availability</h3>
<h4 id="lead-contact">Lead contact</h4>
<p>Further information and requests for resources and reagents should be directed to and will be fulfilled by the Lead Contact, Milton Pividori (milton.pividori@cuanschutz.edu).</p>
<h4 id="materials-availability">Materials availability</h4>
<p>This study did not generate new materials.</p>
<h4 id="data-and-code-availability">Data and code availability</h4>
<ul>
<li>This paper analyzes existing, publicly available data. These accession numbers for the datasets are listed in the key resources table.</li>
<li>All original code has been deposited at <a href="https://github.com/greenelab/ccc">https://github.com/greenelab/ccc</a> and Zenodo: 13304247, and is publicly available as of the date of publication. DOIs are listed in the key resources table.</li>
<li>Any additional information required to reanalyze the data reported in this paper is available from the lead contact upon request.</li>
</ul>
<h3 id="method-details">Method details</h3>
<h4 id="sec:data_gtex">Data preprocessing</h4>
<p>We downloaded gene expression data from GTEx v8 (<a href="https://gtexportal.org/">https://gtexportal.org/</a>) for all tissues, normalized using TPM (transcripts per million), and focused our primary analysis on whole blood, which has a good sample size (755).
We selected the top 5,000 genes from whole blood with the largest variance after standardizing with <span class="math inline">\(log(x + 1)\)</span> to avoid a bias toward highly expressed genes.
We then computed Pearson<span class="citation" data-cites="8Miti2Gz GZZZRhEY"><sup><a href="#ref-8Miti2Gz" role="doc-biblioref">62</a>,<a href="#ref-GZZZRhEY" role="doc-biblioref">63</a></sup></span>, Spearman<span class="citation" data-cites="8Miti2Gz GZZZRhEY"><sup><a href="#ref-8Miti2Gz" role="doc-biblioref">62</a>,<a href="#ref-GZZZRhEY" role="doc-biblioref">63</a></sup></span>, the Maximal Information Coefficient (MIC)<span class="citation" data-cites="AcVZZdTz"><sup><a href="#ref-AcVZZdTz" role="doc-biblioref">64</a></sup></span> and CCC on these 5,000 genes across all 755 samples, generating a pairwise similarity matrix of size 5,000 x 5,000.</p>
<h4 id="sec:ccc_algo">The Clustermatch Correlation Coefficient (CCC)</h4>
<h5 id="definitions">Definitions</h5>
<p><strong>Definition 1.1.</strong> Given a data vector <span class="math inline">\(\mathbf{x}=(x_{1},x_{2},\dots,x_{n}) \in \mathbb{R}^n\)</span> then define</p>
<p><span class="math display">\[\pi_{\ell} = \{i \mid \rho_\ell &lt; x_{i} \leq \rho_{\ell+1}\}, \forall \ell \in \left[1,k\right]\]</span></p>
<p>as a <em>partition</em> of the <span class="math inline">\(n\)</span> objects of <span class="math inline">\(\mathbf{x}\)</span> into <span class="math inline">\(\left\vert\pi\right\vert=k\)</span> clusters, where <span class="math inline">\(\boldsymbol{\rho}\)</span> is a set of <span class="math inline">\(k+1\)</span> cutpoints (e.g., quantiles) that define the clusters, with <span class="math inline">\(\rho_{1} = \min(\mathbf{x})\)</span> and <span class="math inline">\(\rho_{k+1} = \max(\mathbf{x})\)</span>.
If <span class="math inline">\(\mathbf{x}\)</span> is a categorical vector with no intrinsic ordering, then a partition is defined as</p>
<p><span class="math display">\[\pi_{\ell} = \{i \mid x_{i} = \mathcal{C}_{\ell}\}, \forall \ell \in \left[1,\lvert\mathcal{C}\rvert\right]\]</span></p>
<p>where <span class="math inline">\(\mathcal{C} = \{c_1, c_2,\dots,c_m\}\)</span> is a set of unique values in <span class="math inline">\(\mathbf{x}\)</span> corresponding to the <span class="math inline">\(m = \lvert\mathcal{C}\rvert\)</span> categorical values that define the clusters.</p>
<p><strong>Definition 1.2.</strong> Given two partitions <span class="math inline">\(\pi\)</span> and <span class="math inline">\(\pi^{\prime}\)</span> of <span class="math inline">\(n\)</span> objects, the <em>adjusted Rand Index (ARI)</em><span class="citation" data-cites="e4AuoW8N"><sup><a href="#ref-e4AuoW8N" role="doc-biblioref">32</a></sup></span> is given by</p>
<p><span class="math display">\[\textrm{ARI}(\pi, \pi^{\prime}) = \frac{2(n_{0}n_{1} - n_{2} n_{3})}{(n_0 + n_2)(n_2 + n_1) + (n_0 + n_3)(n_3 + n_1)},\]</span></p>
<p>where <span class="math inline">\(n_{0}\)</span> is the number of object pairs that are in the same cluster in both partitions <span class="math inline">\(\pi\)</span> and <span class="math inline">\(\pi^{\prime}\)</span>,
<span class="math inline">\(n_{1}\)</span> is the number of object pairs that are in different clusters,
<span class="math inline">\(n_{2}\)</span> is the number of object pairs that are in the same cluster in <span class="math inline">\(\pi\)</span> but in different clusters in <span class="math inline">\(\pi^{\prime}\)</span>,
and <span class="math inline">\(n_{3}\)</span> is the number of object pairs that are in different clusters in <span class="math inline">\(\pi\)</span> but in the same cluster in <span class="math inline">\(\pi^{\prime}\)</span>.
Intuitively, <span class="math inline">\(n_0 + n_1\)</span> reflects the number of object pairs where both partitions agree, and <span class="math inline">\(n_2 + n_3\)</span> are those in which they disagree.</p>
<p><strong>Definition 1.3.</strong> The <em>Clustermatch Correlation Coefficient (CCC)</em> between <span class="math inline">\(\mathbf{x}\)</span> and <span class="math inline">\(\mathbf{y}\)</span> is defined as the maximum ARI between all possible partitions of <span class="math inline">\(\mathbf{x}\)</span> and <span class="math inline">\(\mathbf{y}\)</span></p>
<p><span class="math display">\[\textrm{CCC}(\mathbf{x}, \mathbf{y}) = \max \lbrace 0, \max_{\substack{\pi_j \in \Pi^{\mathbf{x}} \\ \pi_l \in \Pi^{\mathbf{y}}}} \lbrace \textrm{ARI}(\pi_j, \pi_l) \rbrace \rbrace, \forall \left\vert\pi\right\vert \in [2, k_{\mathrm{max}}]\]</span></p>
<p>where <span class="math inline">\(\Pi^{\mathbf{x}}\)</span> is a set of partitions derived from <span class="math inline">\(\mathbf{x}\)</span>, <span class="math inline">\(\Pi^{\mathbf{y}}\)</span> is a set of partitions derived from <span class="math inline">\(\mathbf{y}\)</span>, and <span class="math inline">\(k_{\mathrm{max}}\)</span> specifies the maximum number of clusters allowed.
The ARI has an upper bound of 1 (achieved when both partitions are identical), and although it does not have a well-defined lower bound, values equal or less than zero are achieved when partitions are independent.
Therefore, <span class="math inline">\(\textrm{CCC}(\mathbf{x}, \mathbf{y}) \in \left[0,1\right]\)</span>.
In the special case where all <span class="math inline">\(n\)</span> objects in either <span class="math inline">\(\mathbf{x}\)</span> or <span class="math inline">\(\mathbf{y}\)</span> have the same value, the CCC is undefined.</p>
<p>The CCC has the following basic properties derived from the ARI:
1) symmetry, since <span class="math inline">\(\mathrm{ARI}(\pi, \pi^{\prime}) = \mathrm{ARI}(\pi^{\prime}, \pi)\)</span>;
2) normalization, since it takes a minimum value of zero and a maximum of one since <span class="math inline">\(\mathrm{ARI}(\pi, \pi) = 1\)</span>;
3) constant baseline, since the ARI is adjusted-for-chance<span class="citation" data-cites="e4AuoW8N"><sup><a href="#ref-e4AuoW8N" role="doc-biblioref">32</a></sup></span>, it returns a value close to zero for independently drawn partitions, and this also holds when partitions have different number of clusters<span class="citation" data-cites="1HaffUr3r"><sup><a href="#ref-1HaffUr3r" role="doc-biblioref">65</a></sup></span>.
This is an important property, since CCC compares partitions with different numbers of clusters, and relationships between two variables (such as linear or quadratic) might be better represented with different numbers of clusters as shown in Figure <a href="#fig:datasets_rel">1</a>.</p>
<h5 id="the-maximum-number-of-clusters-k_mathrmmax">The maximum number of clusters <span class="math inline">\(k_{\mathrm{max}}\)</span></h5>
<p>The parameter <span class="math inline">\(k_{\mathrm{max}}\)</span> is the maximum number of clusters <span class="math inline">\(k\)</span> allowed for any partition derived from <span class="math inline">\(\mathbf{x}\)</span> or <span class="math inline">\(\mathbf{y}\)</span>.
On one hand, note that the same value of <span class="math inline">\(k\)</span> might not be the right one to find a relationship between any two variables.
For instance, in the quadratic example in Figure <a href="#fig:datasets_rel">1</a>, CCC returns a value of 0.36 (grouping objects in four clusters using one variable and two using the other).
If we used only two clusters instead, CCC would return a similarity value of 0.02.
On the other hand, computational time increases quadratically with <span class="math inline">\(k_{\mathrm{max}}\)</span>.
In addition, it is important to note that given the constant baseline property of the ARI, the CCC returns a value close to zero for independent variables regardless of the value of <span class="math inline">\(k_{\mathrm{max}}\)</span>.
As shown in Figure <a href="#fig:constant_baseline:k_max">S2</a>, this holds even for very large values of <span class="math inline">\(k_{\mathrm{max}}\)</span>, approaching the number of objects <span class="math inline">\(n\)</span>.
Note that as <span class="math inline">\(k_{\mathrm{max}}\)</span> approaches <span class="math inline">\(n\)</span>, the number of singleton clusters (i.e., clusters with only one object) increases, which would not be useful for finding relationships between variables.
Therefore, given the constant baseline property, <span class="math inline">\(k_{\mathrm{max}}\)</span> only represents a tradeoff between the ability to capture complex patterns and the computational cost, with random/independent variables having a CCC value close to zero regardless of the value of <span class="math inline">\(k_{\mathrm{max}}\)</span>; we found that <span class="math inline">\(k_{\mathrm{max}}=10\)</span> works well in practice, and it was used as the default maximum number of clusters across all our analyses.</p>
<h5 id="statistical-significance">Statistical significance</h5>
<p>Our null hypothesis is that the variables represented by <span class="math inline">\(\mathbf{x}\)</span> and <span class="math inline">\(\mathbf{y}\)</span> are independent.
To compute a <span class="math inline">\(P\)</span>-value, we perform a set of permutations of values in <span class="math inline">\(\mathbf{y}\)</span> and compute the CCC between <span class="math inline">\(\mathbf{x}\)</span> and each permuted vector.
The <span class="math inline">\(P\)</span>-value is the proportion of CCC values using the permuted data that are greater than or equal to the CCC value between <span class="math inline">\(\mathbf{x}\)</span> and <span class="math inline">\(\mathbf{y}\)</span>.
We used 1 million permutations in all our analyses, and we adjusted the <span class="math inline">\(P\)</span>-values using the Benjamini and Hochberg procedure<span class="citation" data-cites="UeFCwwR7"><sup><a href="#ref-UeFCwwR7" role="doc-biblioref">41</a></sup></span> to control the false discovery rate (FDR);
given the computational cost, we computed a <span class="math inline">\(P\)</span>-value only for gene pairs from the “Disagreements” group in Figure <a href="#fig:upsetplot_coefs">3</a>, which contains gene pairs ranked differently by the correlation coefficients.</p>
<h5 id="algorithm">Algorithm</h5>
<!-- The Latex code for the algorithm is here: https://www.overleaf.com/project/61b8c643eb0ed41c2d8aaadc -->
<figure>
<img src="images/intro/ccc_algorithm/ccc_algorithm.svg" style="width:75.0%" />
</figure>
<p>The main function of the algorithm, <code>ccc</code>, generates a set of partitions <span class="math inline">\(\Pi^{\mathbf{x}}\)</span> for variable <span class="math inline">\(\mathbf{x}\)</span> (line 16), and another set of partitions <span class="math inline">\(\Pi^{\mathbf{y}}\)</span> for variable <span class="math inline">\(\mathbf{y}\)</span> (line 17).
Then, it computes the ARI between each partition <span class="math inline">\(\pi_j \in \Pi^{\mathbf{x}}\)</span> and <span class="math inline">\(\pi_l \in \Pi^{\mathbf{y}}\)</span> and gets the maximum (line 18), returning either this value or zero if this is negative (line 19).</p>
<p>Since CCC only needs a set of partitions to compute a correlation value, any type of variable that can be used to perform clustering is supported.
If variable <span class="math inline">\(\mathbf{v}\)</span> is numerical (lines 2 to 6 in the <code>get_partitions</code> function), each partition <span class="math inline">\(\pi\)</span> is generated using a set of quantiles <span class="math inline">\(\boldsymbol{\rho}\)</span>.
For example, if <span class="math inline">\(k=2\)</span>, then <span class="math inline">\(\boldsymbol{\rho}=(\rho_1, \rho_2, \rho_3)\)</span>, where <span class="math inline">\(\rho_1\)</span> is the minimum value of <span class="math inline">\(\mathbf{v}\)</span>, <span class="math inline">\(\rho_2\)</span> is the median, and <span class="math inline">\(\rho_3\)</span> is the maximum value of <span class="math inline">\(\mathbf{v}\)</span>.
Then, the first cluster <span class="math inline">\(\pi_{1}\)</span> contains all values of <span class="math inline">\(\mathbf{v}\)</span> that are less than or equal to <span class="math inline">\(\rho_2\)</span>, and <span class="math inline">\(\pi_2\)</span> contains all values of <span class="math inline">\(\mathbf{v}\)</span> that are greater than <span class="math inline">\(\rho_2\)</span>.
If variable <span class="math inline">\(\mathbf{v}\)</span> is categorical (lines 8 to 11), we compute a single partition <span class="math inline">\(\pi\)</span> with <span class="math inline">\(m=\left\vert\mathcal{C}\right\vert\)</span> clusters, where <span class="math inline">\(\mathcal{C} = \{c_1, c_2,\dots,c_m\}\)</span> is a set of unique categorical values in <span class="math inline">\(\mathbf{v}\)</span>.
Therefore, all variable types are internally represented as partitions and it is not necessary to access the original data values to compute the ARI.
Consequently, numerical and categorical variables can be naturally integrated.</p>
<p>Our algorithm implementation uses <span class="math inline">\(k_{\mathrm{max}}=10\)</span> as the default.
This means that for a variable pair, 18 partitions are generated (9 for each variable, from <span class="math inline">\(k=2\)</span> to <span class="math inline">\(k=10\)</span>), and 81 ARI comparisons are performed.
Smaller values of <span class="math inline">\(k_{\mathrm{max}}\)</span> can reduce computation time, although at the expense of missing more complex/general relationships.
Our examples in Figure <a href="#fig:datasets_rel">1</a> suggest that using <span class="math inline">\(k_{\mathrm{max}}=2\)</span> would force CCC to find linear-only patterns, which could be a valid use case scenario where only this kind of relationships are desired.
In addition, <span class="math inline">\(k_{\mathrm{max}}=2\)</span> implies that only two partitions are generated, and only one ARI comparison is performed.</p>
<p>As a final remark, generating partitions (lines 15 and 16) and computing their similarity (line 17) can be easily parallelized.
We used three CPU cores in our analyses to speed up the computation of CCC and this could be potentially extended to a large number of processors using a GPU.</p>
<h5 id="strength-of-linear-correlation">Strength of linear correlation</h5>
<p>Figure <a href="#fig:gtex_tissues:kdm6a_uty">4</a> shows the relationships between <em>UTY</em> (chromosome Y) and <em>KDM6A</em> (chromosome X) across tissues in GTEx.
For this gene pair, CCC can find a complex pattern where a subset of samples (males) follows a clear linear relationship, and there is no relationship in the rest of the samples (females).
As it can be seen, there is a difference in the strength of the linear correlation between male samples across different tissues.
For example, in brain cerebellum, the linear correlation among male samples is stronger than in small intestine (terminal ileum).
As shown in Figure <a href="#fig:linear_in_nonlinear">S3</a>a, this difference is reflected by all coefficients when only male samples are considered.</p>
<p>However, when we consider all samples (males and females), there is no longer a linear relationship between <em>UTY</em> and <em>KDM6A</em>.
Therefore, while a subset of the data displays linear relationships, overall, it is no longer true that there is a linear correlation.
CCC assumes that if two variables (genes in our case) are similar, the clustering of objects (samples) using each variable separately should match.
As shown in Figure <a href="#fig:linear_in_nonlinear">S3</a>a with red lines, this clustering of samples and their matching can be seen for the gene pair <em>UTY</em> / <em>KDM6A</em>: when we only consider male samples, CCC finds clusterings in brain cerebellum with a larger matching than in small intestine because the linear strength differs.
But when we consider all samples together (males and females, as shown in Figure <a href="#fig:linear_in_nonlinear">S3</a>b), the pattern is nonlinear, the distribution of all the data is different, and so are the clusterings found by CCC.</p>
<p>The effect of analyzing <em>all</em> the data (males and females) in this nonlinear pattern (Figure <a href="#fig:gtex_tissues:kdm6a_uty">4</a>) is also clear in the negative sign of Pearson and Spearman coefficients in small intestine or even other tissues with a very strong and clear linear pattern among male samples such as breast mammary tissue.
This case indicates that Pearson and Spearman, although statistically significant, are capturing the wrong pattern.
Therefore, the fact that CCC yields a similar value (0.19) for these nonlinear patterns in brain cerebellum and small intestine (Figure <a href="#fig:linear_in_nonlinear">S3</a>b) reflects a similar clustering matching when considering all the samples.
When applied only to the data with linear relationships of varying strength, CCC performs consistently with other coefficients.</p>
<h5 id="presence-of-substructure-in-the-data">Presence of substructure in the data</h5>
<p>Consider a scenario where there are known and undesirable substructures in the data.
In the example in Figure <a href="#fig:substructure">S4</a>, we have simulated two distinct clusters (normally distributed) placed diagonally, horizontally, and vertically.
The only case where the CCC is close to 1.0 (Diagonal, left) is when the clusterings/partitions of objects using each variable (<span class="math inline">\(x\)</span> and <span class="math inline">\(y\)</span>) match, which coincides with a linear pattern.
In the other two cases (Horizontal and Vertical), clusterings of objects do not match, leading to a CCC value of zero.
We note that MIC has the same behavior.</p>
<h4 id="sec:giant">Tissue-specific network analyses using GIANT</h4>
<p>We accessed tissue-specific gene networks of GIANT using both the web interface and web services provided by HumanBase<span class="citation" data-cites="ZSXf7Pdq"><sup><a href="#ref-ZSXf7Pdq" role="doc-biblioref">44</a></sup></span>.
The GIANT version used in this study included 987 genome-scale datasets with approximately 38,000 conditions from around 14,000 publications.
Details on how these networks were built are described in<span class="citation" data-cites="CVF61Un5"><sup><a href="#ref-CVF61Un5" role="doc-biblioref">31</a></sup></span>.
Briefly, tissue-specific gene networks were built using gene expression data (without GTEx samples<span class="citation" data-cites="BgCQnLU6"><sup><a href="#ref-BgCQnLU6" role="doc-biblioref">45</a></sup></span>) from the NCBI’s Gene Expression Omnibus (GEO)<span class="citation" data-cites="tGvcfcsl"><sup><a href="#ref-tGvcfcsl" role="doc-biblioref">66</a></sup></span>, protein-protein interaction (BioGRID<span class="citation" data-cites="JHRpiznV"><sup><a href="#ref-JHRpiznV" role="doc-biblioref">67</a></sup></span>, IntAct<span class="citation" data-cites="7fExfA0i"><sup><a href="#ref-7fExfA0i" role="doc-biblioref">68</a></sup></span>, MINT<span class="citation" data-cites="ODdRgjHY"><sup><a href="#ref-ODdRgjHY" role="doc-biblioref">69</a></sup></span> and MIPS<span class="citation" data-cites="2AZQGpIZ"><sup><a href="#ref-2AZQGpIZ" role="doc-biblioref">70</a></sup></span>), transcription factor regulation using binding motifs from JASPAR<span class="citation" data-cites="h6BVQcKz"><sup><a href="#ref-h6BVQcKz" role="doc-biblioref">71</a></sup></span>, and chemical and genetic perturbations from MSigDB<span class="citation" data-cites="15p5LWIVP"><sup><a href="#ref-15p5LWIVP" role="doc-biblioref">72</a></sup></span>.
Gene expression data were log-transformed, and the Pearson correlation was computed for each gene pair, normalized using the Fisher’s z transform, and z-scores discretized into different bins.
Gold standards for tissue-specific functional relationships were built using expert curation and experimentally derived gene annotations from the Gene Ontology.
Then, one naive Bayesian classifier (using C++ implementations from the Sleipnir library<span class="citation" data-cites="Diiyst6u"><sup><a href="#ref-Diiyst6u" role="doc-biblioref">73</a></sup></span>) for each of the 144 tissues was trained using these gold standards.
Finally, these classifiers were used to estimate the probability of tissue-specific interactions for each gene pair.</p>
<p>For each pair of genes prioritized in our study using GTEx, we used GIANT through HumanBase to obtain
1) a predicted gene network for blood (manually selected to match whole blood in GTEx) and
2) a gene network with an automatically predicted tissue using the method described in<span class="citation" data-cites="mk4SLpDo"><sup><a href="#ref-mk4SLpDo" role="doc-biblioref">47</a></sup></span> and provided by HumanBase web interfaces/services.
Briefly, the tissue prediction approach trains a machine learning model using comprehensive transcriptional data with human-curated markers of different cell lineages (e.g., macrophages) as gold standards.
Then, these models are used to predict other cell lineage-specific genes.
In addition to reporting this predicted tissue or cell lineage, we computed the average probability of interaction between all genes in the network retrieved from GIANT.
Following the default procedure used in GIANT, we included the top 15 genes with the highest probability of interaction with the queried gene pair for each network.</p>
<h4 id="sec:mic">Comparison with the Maximal Information Coefficient (MIC)</h4>
<p>We used the Python package <code>minepy</code><span class="citation" data-cites="wY5WU58s ah0tGOQ0"><sup><a href="#ref-wY5WU58s" role="doc-biblioref">74</a>,<a href="#ref-ah0tGOQ0" role="doc-biblioref">75</a></sup></span> (version 1.2.5) to estimate the MIC coefficient.
In GTEx v8 (whole blood), we used MIC<sub>e</sub> (an improved implementation of the original MIC introduced in<span class="citation" data-cites="AcVZZdTz"><sup><a href="#ref-AcVZZdTz" role="doc-biblioref">64</a></sup></span>) with the default parameters <code>alpha=0.6</code>, <code>c=15</code> and <code>estimator='mic_e'</code>.
We used the <code>pairwise_distances</code> function from <code>scikit-learn</code><span class="citation" data-cites="GZZZRhEY"><sup><a href="#ref-GZZZRhEY" role="doc-biblioref">63</a></sup></span> to parallelize the computation of MIC on GTEx.
For our computational complexity analyses, we ran the original MIC (using parameter <code>estimator='mic_approx'</code>) and MIC<sub>e</sub> (<code>estimator='mic_e'</code>).</p>
<h5 id="conceptual-and-statistical-differences-between-ccc-and-mic">Conceptual and statistical differences between CCC and MIC</h5>
<p>The Clustermatch Correlation Coefficient (CCC) and the Maximal Information Coefficient (MIC)<span class="citation" data-cites="tbcQL0eR"><sup><a href="#ref-tbcQL0eR" role="doc-biblioref">24</a></sup></span> are measures designed to capture non-linear relationships between variables.
While they share certain similarities, there are also notable differences between them.</p>
<p>Conceptually, CCC is grounded in clustering input data using each variable separately.
This process effectively transforms each variable into a set of partitions, each containing a different number of clusters.
The CCC then quantifies the correlation between variables by assessing the similarity of these partitions.
This allows to process of various types of variables, including both numerical and categorical variables, even when the categories are nominal (i.e., they lack intrinsic order), as explained in <a href="#sec:ccc_algo">Methods</a>.
MIC, however, is specifically designed for numerical variables.
Additionally, in theory, CCC should also support correlating variables with different dimensions.
For 1-dimensional variables (such as genes), CCC obtains partitions using a quantiles-based approach.
For multidimensional variables, CCC could potentially use a standard clustering algorithm (such as <span class="math inline">\(k\)</span>-means) to obtain partitions.</p>
<p>Now, consider two variables with <span class="math inline">\(n\)</span> data points on a scatterplot.
We can overlay a grid on this scatterplot with <span class="math inline">\(x\)</span> columns and <span class="math inline">\(y\)</span> rows, where each cell of this grid contains a portion of the data points, thereby defining a bivariate probability distribution.
The MIC algorithm seeks an optimal grid configuration that maximizes the ratio of mutual information to <span class="math inline">\(\log \min \{x, y\}\)</span>, subject to the constraint that <span class="math inline">\(xy &lt; n^{0.6}\)</span>.
This normalization process using <span class="math inline">\(\log \min \{x, y\}\)</span> scales the MIC score between zero and one.
The CCC, as defined in <a href="#sec:ccc_algo">Methods</a>, also generates a symmetric, normalized score between zero and one.
However, unlike MIC which utilizes normalized mutual information, CCC employs the Adjusted Rand Index (ARI).
The ARI has an advantageous property: it consistently returns a baseline (zero) for independently drawn partitions, irrespective of the number of clusters (see Figure <a href="#fig:constant_baseline:k_max">S2</a>).
This property is not inherent in mutual information, which can produce varied values for independent variables if the grid dimensions vary.
MIC mitigates this by limiting the grid size with the constraint <span class="math inline">\(xy &lt; n^{0.6}\)</span>, which could also limit its ability to detect complex relationships.</p>
<p>Both CCC and MIC involve binning the input data vectors, aiming to maximize the mutual information and the ARI, respectively.
However, their approaches differ in complexity and execution.
MIC utilizes a sophisticated dynamic programming algorithm to identify the optimal grid.
In contrast, CCC employs a more straightforward and faster method, partitioning the data points separately using the two vectors.
While CCC might benefit from adopting MIC’s more complex grid search approach, it remains uncertain if MIC could maintain its performance using CCC’s simpler partitioning strategy.</p>
<p>Regarding their parameters, CCC’s <span class="math inline">\(k_{\mathrm{max}}\)</span> (maximum number of clusters) and MIC’s <span class="math inline">\(B(n)\)</span> (maximum grid size) serve similar purposes.
They control both the complexity of the patterns detected and the computational time.
For example, as illustrated in Figure <a href="#fig:datasets_rel">1</a> (Anscombe I and III), a <span class="math inline">\(k_{\mathrm{max}}\)</span> of 2 is adequate for identifying linear patterns but insufficient for more complex patterns like quadratic or two-lines patterns.
A similar principle applies to MIC’s <span class="math inline">\(B(n)\)</span>.
However, a critical distinction exists between the two: the constant baseline property of ARIs ensures that CCC returns a value close to zero for independent variables, regardless of <span class="math inline">\(k_{\mathrm{max}}\)</span>.
In contrast, MIC may produce non-zero scores for independent data if <span class="math inline">\(B(n)\)</span> is set too high, as discussed in Section 2.2.1 of the supplementary material in<span class="citation" data-cites="tbcQL0eR"><sup><a href="#ref-tbcQL0eR" role="doc-biblioref">24</a></sup></span>.
The authors of MIC suggest that a value of <span class="math inline">\(B(n) = n^{0.6}\)</span> is generally effective in practice.</p>
<!-- - put here all that i mentioned to the reviewer -->
<h5 id="comparison-in-gene-expression-data">Comparison in gene expression data</h5>
<p>We compared all the coefficients in this study with MIC, a popular nonlinear method that can find complex relationships in data, although very computationally intensive<span class="citation" data-cites="CHkDwNWM"><sup><a href="#ref-CHkDwNWM" role="doc-biblioref">76</a></sup></span>.
We ran MIC<sub>e</sub> (see Methods) on all possible pairwise comparisons of our 5,000 highly variable genes from whole blood in GTEx v8.
<!-- This took 4 days and 19 hours to finish (compared with 9 hours for CCC). -->
Then we performed the analysis on the distribution of coefficients (the same as in the main text), shown in Figure <a href="#fig:dist_coefs_mic">6</a>.
We verified that CCC and MIC behave similarly in this dataset, with essentially the same distribution but only shifted.
Figure <a href="#fig:dist_coefs_mic">6</a>c shows that these two coefficients relate almost linearly, and both compare very similarly with Pearson and Spearman.</p>
<div id="fig:dist_coefs_mic" class="fignos">
<figure>
<img src="images/coefs_comp/gtex_whole_blood/mic/dist-main.svg" style="width:100.0%" alt="Figure 6: Distribution of MIC values on gene expression (GTEx v8, whole blood) and comparison with other methods. a) Histogram of coefficient values. b) Corresponding cumulative histogram. The dotted line maps the coefficient value that accumulates 70% of gene pairs. c) 2D histogram plot with hexagonal bins between all coefficients, where a logarithmic scale was used to color each hexagon." />
<figcaption aria-hidden="true"><span>Figure 6:</span> <strong>Distribution of MIC values on gene expression (GTEx v8, whole blood) and comparison with other methods.</strong>
<strong>a)</strong> Histogram of coefficient values.
<strong>b)</strong> Corresponding cumulative histogram. The dotted line maps the coefficient value that accumulates 70% of gene pairs.
<strong>c)</strong> 2D histogram plot with hexagonal bins between all coefficients, where a logarithmic scale was used to color each hexagon.</figcaption>
</figure>
</div>
<h4 class="page_break_before" id="sec:time_test">Computational complexity of coefficients</h4>
<p>We also compared CCC with the other coefficients in terms of computational complexity.
Although CCC and MIC might identify similar gene pairs in gene expression data (see <a href="#sec:mic">here</a>), the use of MIC in large datasets remains limited due to its very long computation time, despite some methodological/implementation improvements<span class="citation" data-cites="wY5WU58s NRWTH7z2 hjqdXcdK HLCXXMYj CHkDwNWM"><sup><a href="#ref-wY5WU58s" role="doc-biblioref">74</a>,<a href="#ref-CHkDwNWM" role="doc-biblioref">76</a>–<a href="#ref-HLCXXMYj" role="doc-biblioref">79</a></sup></span>.
The original MIC implementation uses ApproxMaxMI, a computationally demanding heuristic estimator<span class="citation" data-cites="yCHLC0oS"><sup><a href="#ref-yCHLC0oS" role="doc-biblioref">37</a></sup></span>.
Recently, a more efficient implementation called MIC<sub>e</sub> was proposed<span class="citation" data-cites="AcVZZdTz"><sup><a href="#ref-AcVZZdTz" role="doc-biblioref">64</a></sup></span>.
These two MIC estimators are provided by the <code>minepy</code> package<span class="citation" data-cites="wY5WU58s"><sup><a href="#ref-wY5WU58s" role="doc-biblioref">74</a></sup></span>, a C implementation available for Python.
We compared all these coefficients in terms of computation time on randomly generated variables of different sizes, which simulates a scenario of gene expression data with different numbers of conditions.
Differently from the rest, CCC allows us to easily parallelize the computation of a single gene pair (see <a href="#sec:ccc_algo">Methods</a>), so we also tested the cases using 1 and 3 CPU cores.
Figure <a href="#fig:time_test">7</a> shows the time in seconds in log scale.</p>
<div id="fig:time_test" class="fignos">
<figure>
<img src="images/coefs_comp/time_test/time_test-main.svg" style="width:55.0%" alt="Figure 7: Computational complexity of all correlation coefficients on simulated data. We simulated variables/features with varying data sizes (from 100 to a million, x-axis). The plot shows the average time in seconds (log-scale) taken for each coefficient on ten repetitions (1000 repetitions were performed for data size 100). CCC was run using 1 and 3 CPU cores. MIC and MICe did not finish running in a reasonable amount of time for data sizes of 10,000 and 100,000, respectively." />
<figcaption aria-hidden="true"><span>Figure 7:</span> <strong>Computational complexity of all correlation coefficients on simulated data.</strong>
We simulated variables/features with varying data sizes (from 100 to a million, <span class="math inline">\(x\)</span>-axis).
The plot shows the average time in seconds (log-scale) taken for each coefficient on ten repetitions (1000 repetitions were performed for data size 100).
CCC was run using 1 and 3 CPU cores.
MIC and MIC<sub>e</sub> did not finish running in a reasonable amount of time for data sizes of 10,000 and 100,000, respectively.</figcaption>
</figure>
</div>
<p>As we already expected, Pearson and Spearman were the fastest, given that they only need to compute basic summary statistics from the data.
For example, Pearson is three orders of magnitude faster than CCC.
Among the nonlinear coefficients, CCC was faster than the two MIC variations (up to two orders of magnitude), with the only exception in very small data sizes.
The difference is important because both MIC variants were implemented in C<span class="citation" data-cites="wY5WU58s"><sup><a href="#ref-wY5WU58s" role="doc-biblioref">74</a></sup></span>, a high-performance programming language, whereas CCC was implemented in Python (optimized with <code>numba</code>).
For a data size of a million, the multi-core CCC was twice as fast as the single-core CCC.
This suggests that new implementations using more advanced processing units (such as GPUs) are feasible and could make CCC reach speeds closer to Pearson.</p>
<h2 class="page_break_before" id="references">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs" class="references csl-bib-body" role="doc-bibliography">
<div id="ref-DUeOcxpt" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">Hanson, B., Sugden, A., and Alberts, B. (2011). Making data maximally available. Science <em>331</em>, 649. <a href="https://doi.org/10.1126/science.1203354">https://doi.org/10.1126/science.1203354</a>.</div>
</div>
<div id="ref-yvIastvc" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Krishnan, A., Zhang, R., Yao, V., Theesfeld, C.L., Wong, A.K., Tadych, A., Volfovsky, N., Packer, A., Lash, A., and Troyanskaya, O.G. (2016). Genome-wide prediction and functional characterization of the genetic basis of autism spectrum disorder. Nat Neurosci <em>19</em>, 1454–1462. <a href="https://doi.org/10.1038/nn.4353">https://doi.org/10.1038/nn.4353</a>.</div>
</div>
<div id="ref-14q47vZTA" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">Kong, J., Klein, B.E.K., Klein, R., Lee, K.E., and Wahba, G. (2012). Using distance correlation and SS-ANOVA to assess associations of familial relationships, lifestyle factors, diseases, and mortality. Proc. Natl. Acad. Sci. U.S.A. <em>109</em>, 20352–20357. <a href="https://doi.org/10.1073/pnas.1217269109">https://doi.org/10.1073/pnas.1217269109</a>.</div>
</div>
<div id="ref-iZjSr8s2" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">Ge, R., Zhou, M., Luo, Y., Meng, Q., Mai, G., Ma, D., Wang, G., and Zhou, F. (2016). McTwo: a two-step feature selection algorithm based on maximal information coefficient. BMC Bioinformatics <em>17</em>, 142. <a href="https://doi.org/10.1186/s12859-016-0990-0">https://doi.org/10.1186/s12859-016-0990-0</a>.</div>
</div>
<div id="ref-1FQ1FgsCm" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline">Song, X.-F., Zhang, Y., Gong, D.-W., and Gao, X.-Z. (2022). A Fast Hybrid Feature Selection Based on Correlation-Guided Clustering and Particle Swarm Optimization for High-Dimensional Data. IEEE Trans Cybern <em>52</em>, 9573–9586. <a href="https://doi.org/10.1109/tcyb.2021.3061152">https://doi.org/10.1109/tcyb.2021.3061152</a>.</div>
</div>
<div id="ref-16aFivIcr" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline">Novershtern, N., Subramanian, A., Lawton, L.N., Mak, R.H., Haining, W.N., McConkey, M.E., Habib, N., Yosef, N., Chang, C.Y., Shay, T., et al. (2011). Densely interconnected transcriptional circuits control cell states in human hematopoiesis. Cell <em>144</em>, 296–309. <a href="https://doi.org/10.1016/j.cell.2011.01.004">https://doi.org/10.1016/j.cell.2011.01.004</a>.</div>
</div>
<div id="ref-q5QZW3xH" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline">Greene, C.S., Krishnan, A., Wong, A.K., Ricciotti, E., Zelaya, R.A., Himmelstein, D.S., Zhang, R., Hartmann, B.M., Zaslavsky, E., Sealfon, S.C., et al. (2015). Understanding multicellular function and disease with human tissue-specific networks. Nat Genet <em>47</em>, 569–576. <a href="https://doi.org/10.1038/ng.3259">https://doi.org/10.1038/ng.3259</a>.</div>
</div>
<div id="ref-17eqTELga" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline">Ficklin, S.P., and Feltus, F.A. (2011). Gene coexpression network alignment and conservation of gene modules between two grass species: maize and rice. Plant Physiol <em>156</em>, 1244–1256. <a href="https://doi.org/10.1104/pp.111.173047">https://doi.org/10.1104/pp.111.173047</a>.</div>
</div>
<div id="ref-x9k6ERo9" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline">Tsaparas, P., Mariño-Ramírez, L., Bodenreider, O., Koonin, E.V., and Jordan, I.K. (2006). Global similarity and local divergence in human and mouse gene co-expression networks. BMC Evol Biol <em>6</em>, 70. <a href="https://doi.org/10.1186/1471-2148-6-70">https://doi.org/10.1186/1471-2148-6-70</a>.</div>
</div>
<div id="ref-vsBvj38s" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline">(2020). The GTEx Consortium atlas of genetic regulatory effects across human tissues. Science <em>369</em>, 1318–1330. <a href="https://doi.org/10.1126/science.aaz1776">https://doi.org/10.1126/science.aaz1776</a>.</div>
</div>
<div id="ref-TryC3OjB" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline">Wilks, C., Zheng, S.C., Chen, F.Y., Charles, R., Solomon, B., Ling, J.P., Imada, E.L., Zhang, D., Joseph, L., Leek, J.T., et al. (2021). recount3: summaries and queries for large-scale RNA-seq expression and splicing. Genome Biol <em>22</em>, 323. <a href="https://doi.org/10.1186/s13059-021-02533-6">https://doi.org/10.1186/s13059-021-02533-6</a>.</div>
</div>
<div id="ref-zwyRqHIg" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline">Taroni, J.N., Grayson, P.C., Hu, Q., Eddy, S., Kretzler, M., Merkel, P.A., and Greene, C.S. (2019). MultiPLIER: A Transfer Learning Framework for Transcriptomics Reveals Systemic Features of Rare Disease. Cell Syst <em>8</em>, 380–394.e4. <a href="https://doi.org/10.1016/j.cels.2019.04.003">https://doi.org/10.1016/j.cels.2019.04.003</a>.</div>
</div>
<div id="ref-13mfjo6Kw" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline">Barbeira, A.N., Pividori, M., Zheng, J., Wheeler, H.E., Nicolae, D.L., and Im, H.K. (2019). Integrating predicted transcriptome from multiple tissues improves association detection. PLoS Genet <em>15</em>, e1007889. <a href="https://doi.org/10.1371/journal.pgen.1007889">https://doi.org/10.1371/journal.pgen.1007889</a>.</div>
</div>
<div id="ref-12H2nr6Rc" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline">Yao, D.W., O'Connor, L.J., Price, A.L., and Gusev, A. (2020). Quantifying genetic effects on disease mediated by assayed gene expression levels. Nat Genet <em>52</em>, 626–633. <a href="https://doi.org/10.1038/s41588-020-0625-2">https://doi.org/10.1038/s41588-020-0625-2</a>.</div>
</div>
<div id="ref-qAH2cGOH" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline">Võsa, U., Claringbould, A., Westra, H.-J., Bonder, M.J., Deelen, P., Zeng, B., Kirsten, H., Saha, A., Kreuzhuber, R., Yazar, S., et al. (2021). Large-scale cis- and trans-eQTL analyses identify thousands of genetic loci and polygenic scores that regulate blood gene expression. Nat Genet <em>53</em>, 1300–1310. <a href="https://doi.org/10.1038/s41588-021-00913-z">https://doi.org/10.1038/s41588-021-00913-z</a>.</div>
</div>
<div id="ref-g7Id81Dt" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline">Boyle, E.A., Li, Y.I., and Pritchard, J.K. (2017). An Expanded View of Complex Traits: From Polygenic to Omnigenic. Cell <em>169</em>, 1177–1186. <a href="https://doi.org/10.1016/j.cell.2017.05.038">https://doi.org/10.1016/j.cell.2017.05.038</a>.</div>
</div>
<div id="ref-sfMulPrl" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">17. </div><div class="csl-right-inline">Liu, X., Li, Y.I., and Pritchard, J.K. (2019). Trans Effects on Gene Expression Can Drive Omnigenic Inheritance. Cell <em>177</em>, 1022–1034.e6. <a href="https://doi.org/10.1016/j.cell.2019.04.014">https://doi.org/10.1016/j.cell.2019.04.014</a>.</div>
</div>
<div id="ref-1EN7sv6pQ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">18. </div><div class="csl-right-inline">Jagadeesh, K.A., Dey, K.K., Montoro, D.T., Mohan, R., Gazal, S., Engreitz, J.M., Xavier, R.J., Price, A.L., and Regev, A. (2021). Identifying disease-critical cell types and cellular processes across the human body by integration of single-cell profiles and human genetics. bioRxiv. <a href="https://doi.org/10.1101/2021.03.19.436212">https://doi.org/10.1101/2021.03.19.436212</a>.</div>
</div>
<div id="ref-NM3rHx1i" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">19. </div><div class="csl-right-inline">Pividori, M., Lu, S., Li, B., Su, C., Johnson, M.E., Wei, W.-Q., Feng, Q., Namjou, B., Kiryluk, K., Kullo, I.J., et al. (2023). Projecting genetic associations through gene expression patterns highlights disease etiology and drug mechanisms. Nat Commun <em>14</em>. <a href="https://doi.org/10.1038/s41467-023-41057-4">https://doi.org/10.1038/s41467-023-41057-4</a>.</div>
</div>
<div id="ref-17f9RJ1ul" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">20. </div><div class="csl-right-inline">Bakker, O.B., Claringbould, A., Westra, H.-J., Wiersma, H., Boulogne, F., Võsa, U., Symmons, S.M., Jonkers, I.H., Franke, L., and Deelen, P. (2021). Linking common and rare disease genetics through gene regulatory networks. <a href="https://doi.org/10.1101/2021.10.21.21265342">https://doi.org/10.1101/2021.10.21.21265342</a>.</div>
</div>
<div id="ref-sGgkLsyy" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">21. </div><div class="csl-right-inline">Võsa, U., Claringbould, A., Westra, H.-J., Bonder, M.J., Deelen, P., Zeng, B., Kirsten, H., Saha, A., Kreuzhuber, R., Yazar, S., et al. (2021). Large-scale cis- and trans-eQTL analyses identify thousands of genetic loci and polygenic scores that regulate blood gene expression. Nat Genet <em>53</em>, 1300–1310. <a href="https://doi.org/10.1038/s41588-021-00913-z">https://doi.org/10.1038/s41588-021-00913-z</a>.</div>
</div>
<div id="ref-14ZMJkeR2" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline">Mathieson, I. (2021). The omnigenic model and polygenic prediction of complex traits. The American Journal of Human Genetics <em>108</em>, 1558–1563. <a href="https://doi.org/10.1016/j.ajhg.2021.07.003">https://doi.org/10.1016/j.ajhg.2021.07.003</a>.</div>
</div>
<div id="ref-57TjOMEA" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline">Lee, H.-C., Ichikawa, O., Glicksberg, B.S., Divaraniya, A.A., Becker, C.E., Agarwal, P., and Dudley, J.T. (2020). Identification of therapeutic targets from genetic association studies using hierarchical component analysis. BioData Mining <em>13</em>. <a href="https://doi.org/10.1186/s13040-020-00216-9">https://doi.org/10.1186/s13040-020-00216-9</a>.</div>
</div>
<div id="ref-tbcQL0eR" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline">Reshef, D.N., Reshef, Y.A., Finucane, H.K., Grossman, S.R., McVean, G., Turnbaugh, P.J., Lander, E.S., Mitzenmacher, M., and Sabeti, P.C. (2011). Detecting novel associations in large data sets. Science <em>334</em>, 1518–1524. <a href="https://doi.org/10.1126/science.1205438">https://doi.org/10.1126/science.1205438</a>.</div>
</div>
<div id="ref-IvOGGmxI" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">25. </div><div class="csl-right-inline">Székely, G.J., Rizzo, M.L., and Bakirov, N.K. (2007). Measuring and testing dependence by correlation of distances. Ann. Statist. <em>35</em>. <a href="https://doi.org/10.1214/009053607000000505">https://doi.org/10.1214/009053607000000505</a>.</div>
</div>
<div id="ref-NnsqWXcp" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">26. </div><div class="csl-right-inline">Cao, D., Chen, Y., Chen, J., Zhang, H., and Yuan, Z. (2021). An improved algorithm for the maximal information coefficient and its application. R Soc Open Sci <em>8</em>, 201424. <a href="https://doi.org/10.1098/rsos.201424">https://doi.org/10.1098/rsos.201424</a>.</div>
</div>
<div id="ref-g8HrFUfP" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">27. </div><div class="csl-right-inline">Liang, T., Zhang, Q., Liu, X., Lou, C., Liu, X., and Wang, H. (2020). Time-Frequency Maximal Information Coefficient Method and its Application to Functional Corticomuscular Coupling. IEEE Trans Neural Syst Rehabil Eng <em>28</em>, 2515–2524. <a href="https://doi.org/10.1109/tnsre.2020.3028199">https://doi.org/10.1109/tnsre.2020.3028199</a>.</div>
</div>
<div id="ref-ztrPHUjX" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">28. </div><div class="csl-right-inline">Chen, Y., Zeng, Y., Luo, F., and Yuan, Z. (2016). A New Algorithm to Optimize Maximal Information Coefficient. PLoS One <em>11</em>, e0157567. <a href="https://doi.org/10.1371/journal.pone.0157567">https://doi.org/10.1371/journal.pone.0157567</a>.</div>
</div>
<div id="ref-VCRLONE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">29. </div><div class="csl-right-inline">Pividori, M., Cernadas, A., de Haro, L.A., Carrari, F., Stegmayer, G., and Milone, D.H. (2018). Clustermatch: discovering hidden relations in highly diverse kinds of qualitative and quantitative data without standardization. Bioinformatics <em>35</em>, 1931–1939. <a href="https://doi.org/10.1093/bioinformatics/bty899">https://doi.org/10.1093/bioinformatics/bty899</a>.</div>
</div>
<div id="ref-9Pr9idng" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">30. </div><div class="csl-right-inline">, Aguet, F., Anand, S., Ardlie, K.G., Gabriel, S., Getz, G.A., Graubert, A., Hadley, K., Handsaker, R.E., Huang, K.H., et al. (2020). The GTEx Consortium atlas of genetic regulatory effects across human tissues. Science <em>369</em>, 1318–1330. <a href="https://doi.org/10.1126/science.aaz1776">https://doi.org/10.1126/science.aaz1776</a>.</div>
</div>
<div id="ref-CVF61Un5" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">31. </div><div class="csl-right-inline">Greene, C.S., Krishnan, A., Wong, A.K., Ricciotti, E., Zelaya, R.A., Himmelstein, D.S., Zhang, R., Hartmann, B.M., Zaslavsky, E., Sealfon, S.C., et al. (2015). Understanding multicellular function and disease with human tissue-specific networks. Nat Genet <em>47</em>, 569–576. <a href="https://doi.org/10.1038/ng.3259">https://doi.org/10.1038/ng.3259</a>.</div>
</div>
<div id="ref-e4AuoW8N" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">32. </div><div class="csl-right-inline">Hubert, L., and Arabie, P. (1985). Comparing partitions. Journal of Classification <em>2</em>, 193–218. <a href="https://doi.org/10.1007/bf01908075">https://doi.org/10.1007/bf01908075</a>.</div>
</div>
<div id="ref-WP7yhZGS" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">33. </div><div class="csl-right-inline">Anscombe, F.J. (1973). Graphs in Statistical Analysis. The American Statistician <em>27</em>, 17–21. <a href="https://doi.org/10.1080/00031305.1973.10478966">https://doi.org/10.1080/00031305.1973.10478966</a>.</div>
</div>
<div id="ref-LW7cVCGa" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">34. </div><div class="csl-right-inline">Cairo, A. Download the Datasaurus: Never trust summary statistics alone; always visualize your data. <a href="http://www.thefunctionalart.com/2016/08/download-datasaurus-never-trust-summary.html">http://www.thefunctionalart.com/2016/08/download-datasaurus-never-trust-summary.html</a>.</div>
</div>
<div id="ref-dvyMwrz9" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">35. </div><div class="csl-right-inline">Matejka, J., and Fitzmaurice, G. (2017). Same Stats, Different Graphs. In Proceedings of the 2017 CHI Conference on Human Factors in Computing Systems (ACM). <a href="https://doi.org/10.1145/3025453.3025912">https://doi.org/10.1145/3025453.3025912</a>.</div>
</div>
<div id="ref-FhufXP6P" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">36. </div><div class="csl-right-inline">Murray, L.L., and Wilson, J.G. (2021). Generating data sets for teaching the importance of regression analysis. Decision Sci J Innov Edu <em>19</em>, 157–166. <a href="https://doi.org/10.1111/dsji.12233">https://doi.org/10.1111/dsji.12233</a>.</div>
</div>
<div id="ref-yCHLC0oS" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">37. </div><div class="csl-right-inline">Reshef, D.N., Reshef, Y.A., Finucane, H.K., Grossman, S.R., McVean, G., Turnbaugh, P.J., Lander, E.S., Mitzenmacher, M., and Sabeti, P.C. (2011). Detecting Novel Associations in Large Data Sets. Science <em>334</em>, 1518–1524. <a href="https://doi.org/10.1126/science.1205438">https://doi.org/10.1126/science.1205438</a>.</div>
</div>
<div id="ref-1AMcxTw9R" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">38. </div><div class="csl-right-inline">Wang, Q., Zhang, H., Liang, Y., Jiang, H., Tan, S., Luo, F., Yuan, Z., and Chen, Y. (2020). A Novel Method to Efficiently Highlight Nonlinearly Expressed Genes. Front. Genet. <em>10</em>. <a href="https://doi.org/10.3389/fgene.2019.01410">https://doi.org/10.3389/fgene.2019.01410</a>.</div>
</div>
<div id="ref-16ddFzCqe" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">39. </div><div class="csl-right-inline">Spellman, P.T., Sherlock, G., Zhang, M.Q., Iyer, V.R., Anders, K., Eisen, M.B., Brown, P.O., Botstein, D., and Futcher, B. (1998). Comprehensive Identification of Cell Cycle–regulated Genes of the Yeast&lt;i&gt;Saccharomyces cerevisiae&lt;/i&gt;by Microarray Hybridization. MBoC <em>9</em>, 3273–3297. <a href="https://doi.org/10.1091/mbc.9.12.3273">https://doi.org/10.1091/mbc.9.12.3273</a>.</div>
</div>
<div id="ref-ZhJ7VK5R" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">40. </div><div class="csl-right-inline">Lex, A., Gehlenborg, N., Strobelt, H., Vuillemot, R., and Pfister, H. (2014). UpSet: Visualization of Intersecting Sets. IEEE Trans. Visual. Comput. Graphics <em>20</em>, 1983–1992. <a href="https://doi.org/10.1109/tvcg.2014.2346248">https://doi.org/10.1109/tvcg.2014.2346248</a>.</div>
</div>
<div id="ref-UeFCwwR7" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">41. </div><div class="csl-right-inline">Benjamini, Y., and Hochberg, Y. (1995). Controlling the False Discovery Rate: A Practical and Powerful Approach to Multiple Testing. Journal of the Royal Statistical Society: Series B (Methodological) <em>57</em>, 289–300. <a href="https://doi.org/10.1111/j.2517-6161.1995.tb02031.x">https://doi.org/10.1111/j.2517-6161.1995.tb02031.x</a>.</div>
</div>
<div id="ref-9i58Tfs4" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">42. </div><div class="csl-right-inline">Shi, L., Zhang, W., Zou, F., Mei, L., Wu, G., and Teng, Y. (2016). KLHL21, a novel gene that contributes to the progression of hepatocellular carcinoma. BMC Cancer <em>16</em>, 815. <a href="https://doi.org/10.1186/s12885-016-2851-7">https://doi.org/10.1186/s12885-016-2851-7</a>.</div>
</div>
<div id="ref-9ZXy7Ut2" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">43. </div><div class="csl-right-inline">Greene, C.S., Krishnan, A., Wong, A.K., Ricciotti, E., Zelaya, R.A., Himmelstein, D.S., Zhang, R., Hartmann, B.M., Zaslavsky, E., Sealfon, S.C., et al. (2015). Understanding multicellular function and disease with human tissue-specific networks. Nat Genet <em>47</em>, 569–576. <a href="https://doi.org/10.1038/ng.3259">https://doi.org/10.1038/ng.3259</a>.</div>
</div>
<div id="ref-ZSXf7Pdq" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">44. </div><div class="csl-right-inline">HumanBase: data-driven predictions of gene function and interactions <a href="https://hb.flatironinstitute.org/">https://hb.flatironinstitute.org/</a>.</div>
</div>
<div id="ref-BgCQnLU6" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">45. </div><div class="csl-right-inline">Data sources <a href="https://hb.flatironinstitute.org/data">https://hb.flatironinstitute.org/data</a>.</div>
</div>
<div id="ref-tt0Vdf4C" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">46. </div><div class="csl-right-inline">Subramanian, A., Tamayo, P., Mootha, V.K., Mukherjee, S., Ebert, B.L., Gillette, M.A., Paulovich, A., Pomeroy, S.L., Golub, T.R., Lander, E.S., et al. (2005). Gene set enrichment analysis: a knowledge-based approach for interpreting genome-wide expression profiles. Proc Natl Acad Sci U S A <em>102</em>, 15545–15550. <a href="https://doi.org/10.1073/pnas.0506580102">https://doi.org/10.1073/pnas.0506580102</a>.</div>
</div>
<div id="ref-mk4SLpDo" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">47. </div><div class="csl-right-inline">Ju, W., Greene, C.S., Eichinger, F., Nair, V., Hodgin, J.B., Bitzer, M., Lee, Y., Zhu, Q., Kehata, M., Li, M., et al. (2013). Defining cell-type specificity at the transcriptional level in human disease. Genome Res. <em>23</em>, 1862–1873. <a href="https://doi.org/10.1101/gr.155697.113">https://doi.org/10.1101/gr.155697.113</a>.</div>
</div>
<div id="ref-FKQEHB1F" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">48. </div><div class="csl-right-inline">RASSF2, CYTIP - HumanBase <a href="https://hb.flatironinstitute.org/gene/9770+9595">https://hb.flatironinstitute.org/gene/9770+9595</a>.</div>
</div>
<div id="ref-JpfbR0un" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">49. </div><div class="csl-right-inline">MYOZ1, TNNI2 - HumanBase <a href="https://hb.flatironinstitute.org/gene/58529+7136">https://hb.flatironinstitute.org/gene/58529+7136</a>.</div>
</div>
<div id="ref-4QgDZ8bF" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">50. </div><div class="csl-right-inline">Clayton, J.A., and Collins, F.S. (2014). Policy: NIH to balance sex in cell and animal studies. Nature <em>509</em>, 282–283. <a href="https://doi.org/10.1038/509282a">https://doi.org/10.1038/509282a</a>.</div>
</div>
<div id="ref-iox25BGw" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">51. </div><div class="csl-right-inline">Bhargava, A., Arnold, A.P., Bangasser, D.A., Denton, K.M., Gupta, A., Hilliard Krause, L.M., Mayer, E.A., McCarthy, M., Miller, W.L., Raznahan, A., et al. (2021). Considering Sex as a Biological Variable in Basic and Clinical Studies: An Endocrine Society Scientific Statement. Endocrine Reviews <em>42</em>, 219–258. <a href="https://doi.org/10.1210/endrev/bnaa034">https://doi.org/10.1210/endrev/bnaa034</a>.</div>
</div>
<div id="ref-1GpiBQk8d" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">52. </div><div class="csl-right-inline">Shansky, R.M., and Murphy, A.Z. (2021). Considering sex as a biological variable will require a global shift in science culture. Nat Neurosci <em>24</em>, 457–464. <a href="https://doi.org/10.1038/s41593-021-00806-8">https://doi.org/10.1038/s41593-021-00806-8</a>.</div>
</div>
<div id="ref-1CZbj0k1F" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">53. </div><div class="csl-right-inline">Pfeiffer, T., and Hoffmann, R. (2007). Temporal patterns of genes in scientific publications. Proc Natl Acad Sci U S A <em>104</em>, 12052–12056. <a href="https://doi.org/10.1073/pnas.0701315104">https://doi.org/10.1073/pnas.0701315104</a>.</div>
</div>
<div id="ref-104MiBf8H" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">54. </div><div class="csl-right-inline">Su, A.I., and Hogenesch, J.B. (2007). Power-law-like distributions in biomedical publications and research funding. Genome Biol <em>8</em>, 404. <a href="https://doi.org/10.1186/gb-2007-8-4-404">https://doi.org/10.1186/gb-2007-8-4-404</a>.</div>
</div>
<div id="ref-IlPwHaTE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">55. </div><div class="csl-right-inline">Stoeger, T., Gerlach, M., Morimoto, R.I., and Nunes Amaral, L.A. (2018). Large-scale investigation of the reasons why potentially important genes are ignored. PLoS Biol <em>16</em>, e2006643. <a href="https://doi.org/10.1371/journal.pbio.2006643">https://doi.org/10.1371/journal.pbio.2006643</a>.</div>
</div>
<div id="ref-efGVmXWt" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">56. </div><div class="csl-right-inline">Chen, J., Song, W., Du, Y., Li, Z., Xuan, Z., Zhao, L., Chen, J., Zhao, Y., Tuo, B., Zheng, S., et al. (2018). Inhibition of KLHL21 prevents cholangiocarcinoma progression through regulating cell proliferation and motility, arresting cell cycle and reducing Erk activation. Biochem Biophys Res Commun <em>499</em>, 433–440. <a href="https://doi.org/10.1016/j.bbrc.2018.03.152">https://doi.org/10.1016/j.bbrc.2018.03.152</a>.</div>
</div>
<div id="ref-1FQgFkJoJ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">57. </div><div class="csl-right-inline">Li, C., Li, R., Hu, X., Zhou, G., and Jiang, G. (2022). Tumor-promoting mechanisms of macrophage-derived extracellular vesicles-enclosed microRNA-660 in breast cancer progression. Breast Cancer Res Treat <em>192</em>, 353–368. <a href="https://doi.org/10.1007/s10549-021-06433-y">https://doi.org/10.1007/s10549-021-06433-y</a>.</div>
</div>
<div id="ref-yqBcaXHD" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">58. </div><div class="csl-right-inline">Visscher, P.M., Wray, N.R., Zhang, Q., Sklar, P., McCarthy, M.I., Brown, M.A., and Yang, J. (2017). 10 Years of GWAS Discovery: Biology, Function, and Translation. The American Journal of Human Genetics <em>101</em>, 5–22. <a href="https://doi.org/10.1016/j.ajhg.2017.06.005">https://doi.org/10.1016/j.ajhg.2017.06.005</a>.</div>
</div>
<div id="ref-17B4AigZT" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">59. </div><div class="csl-right-inline">Tam, V., Patel, N., Turcotte, M., Bossé, Y., Paré, G., and Meyre, D. (2019). Benefits and limitations of genome-wide association studies. Nat Rev Genet <em>20</em>, 467–484. <a href="https://doi.org/10.1038/s41576-019-0127-1">https://doi.org/10.1038/s41576-019-0127-1</a>.</div>
</div>
<div id="ref-12OuQVjYy" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">60. </div><div class="csl-right-inline">Burns, J.J.R., Shealy, B.T., Greer, M.S., Hadish, J.A., McGowan, M.T., Biggs, T., Smith, M.C., Feltus, F.A., and Ficklin, S.P. (2021). Addressing noise in co-expression network construction. Briefings in Bioinformatics <em>23</em>. <a href="https://doi.org/10.1093/bib/bbab495">https://doi.org/10.1093/bib/bbab495</a>.</div>
</div>
<div id="ref-I46r0ipf" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">61. </div><div class="csl-right-inline">Knijnenburg, T.A., Wessels, L.F.A., Reinders, M.J.T., and Shmulevich, I. (2009). Fewer permutations, more accurate &lt;i&gt;P&lt;/i&gt;-values. Bioinformatics <em>25</em>, i161–i168. <a href="https://doi.org/10.1093/bioinformatics/btp211">https://doi.org/10.1093/bioinformatics/btp211</a>.</div>
</div>
<div id="ref-8Miti2Gz" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">62. </div><div class="csl-right-inline">Virtanen, P., Gommers, R., Oliphant, T.E., Haberland, M., Reddy, T., Cournapeau, D., Burovski, E., Peterson, P., Weckesser, W., Bright, J., et al. (2020). SciPy 1.0: fundamental algorithms for scientific computing in Python. Nat Methods <em>17</em>, 261–272. <a href="https://doi.org/10.1038/s41592-019-0686-2">https://doi.org/10.1038/s41592-019-0686-2</a>.</div>
</div>
<div id="ref-GZZZRhEY" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">63. </div><div class="csl-right-inline">Pedregosa, F., Varoquaux, G., Gramfort, A., Michel, V., Thirion, B., Grisel, O., Blondel, M., Prettenhofer, P., Weiss, R., Dubourg, V., et al. (2011). <a href="https://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html">Scikit-learn: Machine Learning in Python</a>. Journal of Machine Learning Research <em>12</em>, 2825–2830.</div>
</div>
<div id="ref-AcVZZdTz" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">64. </div><div class="csl-right-inline">Reshef, Y., Reshef, D., Finucane, H., Sabeti, P., and Mitzenmacher, M. (2016). <a href="https://jmlr.org/papers/v17/15-308.html">Measuring Dependence Powerfully and Equitably</a>. Journal of Machine Learning Research <em>17</em>, 1–63.</div>
</div>
<div id="ref-1HaffUr3r" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">65. </div><div class="csl-right-inline">Vinh, N.X., Epps, J., and Bailey, J. (2010). <a href="https://jmlr.org/papers/v11/vinh10a.html">Information Theoretic Measures for Clusterings Comparison: Variants, Properties, Normalization and Correction for Chance</a>. Journal of Machine Learning Research <em>11</em>, 2837–2854.</div>
</div>
<div id="ref-tGvcfcsl" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">66. </div><div class="csl-right-inline">Barrett, T., Wilhite, S.E., Ledoux, P., Evangelista, C., Kim, I.F., Tomashevsky, M., Marshall, K.A., Phillippy, K.H., Sherman, P.M., Holko, M., et al. (2012). NCBI GEO: archive for functional genomics data sets—update. Nucleic Acids Research <em>41</em>, D991–D995. <a href="https://doi.org/10.1093/nar/gks1193">https://doi.org/10.1093/nar/gks1193</a>.</div>
</div>
<div id="ref-JHRpiznV" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">67. </div><div class="csl-right-inline">Chatr-Aryamontri, A., Breitkreutz, B.-J., Heinicke, S., Boucher, L., Winter, A., Stark, C., Nixon, J., Ramage, L., Kolas, N., O'Donnell, L., et al. (2013). The BioGRID interaction database: 2013 update. Nucleic Acids Res <em>41</em>, D816–23. <a href="https://doi.org/10.1093/nar/gks1158">https://doi.org/10.1093/nar/gks1158</a>.</div>
</div>
<div id="ref-7fExfA0i" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">68. </div><div class="csl-right-inline">Kerrien, S., Aranda, B., Breuza, L., Bridge, A., Broackes-Carter, F., Chen, C., Duesbury, M., Dumousseau, M., Feuermann, M., Hinz, U., et al. (2011). The IntAct molecular interaction database in 2012. Nucleic Acids Research <em>40</em>, D841–D846. <a href="https://doi.org/10.1093/nar/gkr1088">https://doi.org/10.1093/nar/gkr1088</a>.</div>
</div>
<div id="ref-ODdRgjHY" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">69. </div><div class="csl-right-inline">Licata, L., Briganti, L., Peluso, D., Perfetto, L., Iannuccelli, M., Galeota, E., Sacco, F., Palma, A., Nardozza, A.P., Santonico, E., et al. (2011). MINT, the molecular interaction database: 2012 update. Nucleic Acids Research <em>40</em>, D857–D861. <a href="https://doi.org/10.1093/nar/gkr930">https://doi.org/10.1093/nar/gkr930</a>.</div>
</div>
<div id="ref-2AZQGpIZ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">70. </div><div class="csl-right-inline">Mewes, H.W., Heumann, K., Kaps, A., Mayer, K., Pfeiffer, F., Stocker, S., and Frishman, D. (1999). MIPS: a database for genomes and protein sequences. Nucleic Acids Res <em>27</em>, 44–48. <a href="https://doi.org/10.1093/nar/27.1.44">https://doi.org/10.1093/nar/27.1.44</a>.</div>
</div>
<div id="ref-h6BVQcKz" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">71. </div><div class="csl-right-inline">Portales-Casamar, E., Thongjuea, S., Kwon, A.T., Arenillas, D., Zhao, X., Valen, E., Yusuf, D., Lenhard, B., Wasserman, W.W., and Sandelin, A. (2009). JASPAR 2010: the greatly expanded open-access database of transcription factor binding profiles. Nucleic Acids Research <em>38</em>, D105–D110. <a href="https://doi.org/10.1093/nar/gkp950">https://doi.org/10.1093/nar/gkp950</a>.</div>
</div>
<div id="ref-15p5LWIVP" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">72. </div><div class="csl-right-inline">Subramanian, A., Tamayo, P., Mootha, V.K., Mukherjee, S., Ebert, B.L., Gillette, M.A., Paulovich, A., Pomeroy, S.L., Golub, T.R., Lander, E.S., et al. (2005). Gene set enrichment analysis: A knowledge-based approach for interpreting genome-wide expression profiles. Proc. Natl. Acad. Sci. U.S.A. <em>102</em>, 15545–15550. <a href="https://doi.org/10.1073/pnas.0506580102">https://doi.org/10.1073/pnas.0506580102</a>.</div>
</div>
<div id="ref-Diiyst6u" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">73. </div><div class="csl-right-inline">Huttenhower, C., Schroeder, M., Chikina, M.D., and Troyanskaya, O.G. (2008). The Sleipnir library for computational functional genomics. Bioinformatics <em>24</em>, 1559–1561. <a href="https://doi.org/10.1093/bioinformatics/btn237">https://doi.org/10.1093/bioinformatics/btn237</a>.</div>
</div>
<div id="ref-wY5WU58s" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">74. </div><div class="csl-right-inline">Albanese, D., Filosi, M., Visintainer, R., Riccadonna, S., Jurman, G., and Furlanello, C. (2012). minerva and minepy: a C engine for the MINE suite and its R, Python and MATLAB wrappers. Bioinformatics <em>29</em>, 407–408. <a href="https://doi.org/10.1093/bioinformatics/bts707">https://doi.org/10.1093/bioinformatics/bts707</a>.</div>
</div>
<div id="ref-ah0tGOQ0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">75. </div><div class="csl-right-inline"><a href="https://github.com/minepy/minepy">minepy - Maximal Information-based Nonparametric Exploration</a> (2024). (minepy - Maximal Information-based Nonparametric Exploration (MINE) in C and Python).</div>
</div>
<div id="ref-CHkDwNWM" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">76. </div><div class="csl-right-inline">Cao, D., Chen, Y., Chen, J., Zhang, H., and Yuan, Z. (2021). An improved algorithm for the maximal information coefficient and its application. R. Soc. open sci. <em>8</em>. <a href="https://doi.org/10.1098/rsos.201424">https://doi.org/10.1098/rsos.201424</a>.</div>
</div>
<div id="ref-NRWTH7z2" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">77. </div><div class="csl-right-inline">Chen, Y., Zeng, Y., Luo, F., and Yuan, Z. (2016). A New Algorithm to Optimize Maximal Information Coefficient. PLoS ONE <em>11</em>, e0157567. <a href="https://doi.org/10.1371/journal.pone.0157567">https://doi.org/10.1371/journal.pone.0157567</a>.</div>
</div>
<div id="ref-hjqdXcdK" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">78. </div><div class="csl-right-inline">Tang, D., Wang, M., Zheng, W., and Wang, H. (2014). RapidMic: Rapid Computation of the Maximal Information Coefficient. Evol Bioinform Online <em>10</em>, EBO.S13121. <a href="https://doi.org/10.4137/ebo.s13121">https://doi.org/10.4137/ebo.s13121</a>.</div>
</div>
<div id="ref-HLCXXMYj" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">79. </div><div class="csl-right-inline">Zhang, Y., Jia, S., Huang, H., Qiu, J., and Zhou, C. (2014). A Novel Algorithm for the Precise Calculation of the Maximal Information Coefficient. Sci Rep <em>4</em>. <a href="https://doi.org/10.1038/srep06662">https://doi.org/10.1038/srep06662</a>.</div>
</div>
</div>
<h2 class="page_break_before" id="supplemental-information">Supplemental information</h2>
<h3 id="figures">Figures</h3>
<div id="fig:gtex_tissues:kdm6a_ddx3y" class="fignos">
<figure>
<img src="images/coefs_comp/kdm6a_vs_ddx3y/gtex-KDM6A_vs_DDX3Y-main.svg" style="width:95.0%" data-tag="S1" alt="Figure S1: The expression levels of KDM6A and DDX3Y display sex-specific associations across GTEx tissues. CCC captures this nonlinear relationship in all GTEx tissues (nine examples are shown in the first three rows), except in female-specific organs (last row)." />
<figcaption aria-hidden="true"><span>Figure S1:</span> <strong>The expression levels of <em>KDM6A</em> and <em>DDX3Y</em> display sex-specific associations across GTEx tissues.</strong>
CCC captures this nonlinear relationship in all GTEx tissues (nine examples are shown in the first three rows), except in female-specific organs (last row).</figcaption>
</figure>
</div>
<div id="fig:constant_baseline:k_max" class="fignos">
<figure>
<img src="images/misc/constant_baseline-k_max.svg" style="width:65.0%" data-tag="S2" alt="Figure S2: Constant baseline property: CCC values are close to zero for random/independent variables. The plot shows CCC values for normally distributed and independent variables with different sizes n and using different values for parameter k_{\mathrm{max}} (maximum number of clusters)." />
<figcaption aria-hidden="true"><span>Figure S2:</span> <strong>Constant baseline property: CCC values are close to zero for random/independent variables.</strong>
The plot shows CCC values for normally distributed and independent variables with different sizes <span class="math inline">\(n\)</span> and using different values for parameter <span class="math inline">\(k_{\mathrm{max}}\)</span> (maximum number of clusters).</figcaption>
</figure>
</div>
<div id="fig:linear_in_nonlinear" class="fignos">
<figure>
<img src="images/coefs_comp/kdm6a_vs_uty/gtex-KDM6A_vs_UTY-nonlinear_and_linear.svg" style="width:70.0%" data-tag="S3" alt="Figure S3: Linear and nonlinear patterns between UTY and KDM6A in brain cerebellum and small intestine (terminal ileum) in GTEx. a) Correlation values for Pearson, Spearman and CCC when only male samples are considered in brain cerebellum and small intestine. b) Correlation value of CCC when all samples (males and females) are considered. Vertical and horizontal red lines show how CCC clustered data points using each gene separately." />
<figcaption aria-hidden="true"><span>Figure S3:</span> <strong>Linear and nonlinear patterns between <em>UTY</em> and <em>KDM6A</em> in brain cerebellum and small intestine (terminal ileum) in GTEx.</strong>
<strong>a)</strong> Correlation values for Pearson, Spearman and CCC when only male samples are considered in brain cerebellum and small intestine.
<strong>b)</strong> Correlation value of CCC when all samples (males and females) are considered.
Vertical and horizontal red lines show how CCC clustered data points using each gene separately.</figcaption>
</figure>
</div>
<div id="fig:substructure" class="fignos">
<figure>
<img src="images/misc/data_with_substructures/clusters.svg" style="width:100.0%" data-tag="S4" alt="Figure S4: Behavior of CCC and MIC when substructure is present in the data. Two simulated, normally distributed clusters across two variables (x and y) are placed diagonally (left), horizontally (middle) and vertically (right), and the CCC and MIC values are calculated. Vertical and horizontal red lines show how CCC clustered data points using each variable separately." />
<figcaption aria-hidden="true"><span>Figure S4:</span> <strong>Behavior of CCC and MIC when substructure is present in the data.</strong>
Two simulated, normally distributed clusters across two variables (<span class="math inline">\(x\)</span> and <span class="math inline">\(y\)</span>) are placed diagonally (left), horizontally (middle) and vertically (right), and the CCC and MIC values are calculated.
Vertical and horizontal red lines show how CCC clustered data points using each variable separately.</figcaption>
</figure>
</div>
<h3 class="page_break_before" id="tables">Tables</h3>
<div id="tbl:giant:weights" class="tablenos">
<table id="tbl:giant:weights" data-tag="S1">
<caption><span>Table S1:</span> <strong>Network statistics of seven gene pairs shown in Figure <a href="#fig:upsetplot_coefs">3</a>b for blood and predicted cell types.</strong>
Only gene pairs present in GIANT models are listed.
For each gene in the pair (first column), the minimum, average and maximum interaction coefficients with the other genes in the network are shown.
</caption>
<colgroup>
<col style="width: 13%" />
<col style="width: 11%" />
<col style="width: 11%" />
<col style="width: 11%" />
<col style="width: 16%" />
<col style="width: 11%" />
<col style="width: 11%" />
<col style="width: 11%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: center;"></th>
<th style="text-align: center;"><strong>Interaction confidence</strong> <!-- $colspan="7" --></th>
<th style="text-align: center;"></th>
<th style="text-align: center;"></th>
<th style="text-align: center;"></th>
<th style="text-align: center;"></th>
<th style="text-align: center;"></th>
<th style="text-align: center;"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;"></td>
<td style="text-align: center;"><strong>Blood</strong> <!-- $colspan="3" --></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"><strong>Predicted cell type</strong> <!-- $colspan="4" --></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong>Gene</strong></td>
<td style="text-align: center;"><strong>Min.</strong></td>
<td style="text-align: center;"><strong>Avg.</strong></td>
<td style="text-align: center;"><strong>Max.</strong></td>
<td style="text-align: center;"><strong>Cell type</strong></td>
<td style="text-align: center;"><strong>Min.</strong></td>
<td style="text-align: center;"><strong>Avg.</strong></td>
<td style="text-align: center;"><strong>Max.</strong></td>
</tr>
<tr class="odd">
<td style="text-align: center;"><em>IFNG</em></td>
<td style="text-align: center;">0.19</td>
<td style="text-align: center;">0.42</td>
<td style="text-align: center;">0.54</td>
<td style="text-align: center;">Natural killer cell<!-- $rowspan="2" --></td>
<td style="text-align: center;">0.74</td>
<td style="text-align: center;">0.90</td>
<td style="text-align: center;">0.99</td>
</tr>
<tr class="even">
<td style="text-align: center;"><em>SDS</em></td>
<td style="text-align: center;">0.18</td>
<td style="text-align: center;">0.29</td>
<td style="text-align: center;">0.41</td>
<td style="text-align: center;">0.65</td>
<td style="text-align: center;">0.81</td>
<td style="text-align: center;">0.94<!-- $removenext="2" --></td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: center;"><!-- $colspan="8" --></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: center;"><em>PRSS36</em></td>
<td style="text-align: center;">0.07</td>
<td style="text-align: center;">0.10</td>
<td style="text-align: center;">0.14</td>
<td style="text-align: center;">Macrophage<!-- $rowspan="2" --></td>
<td style="text-align: center;">0.04</td>
<td style="text-align: center;">0.05</td>
<td style="text-align: center;">0.08</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><em>CCL18</em></td>
<td style="text-align: center;">0.07</td>
<td style="text-align: center;">0.74</td>
<td style="text-align: center;">0.86</td>
<td style="text-align: center;">0.05</td>
<td style="text-align: center;">0.69</td>
<td style="text-align: center;">0.90<!-- $removenext="2" --></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: center;"><!-- $colspan="8" --></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: center;"><em>UTY</em></td>
<td style="text-align: center;">0.03</td>
<td style="text-align: center;">0.36</td>
<td style="text-align: center;">0.84</td>
<td style="text-align: center;">Placenta<!-- $rowspan="2" --></td>
<td style="text-align: center;">0.01</td>
<td style="text-align: center;">0.03</td>
<td style="text-align: center;">0.04</td>
</tr>
<tr class="even">
<td style="text-align: center;"><em>KDM6A</em></td>
<td style="text-align: center;">0.03</td>
<td style="text-align: center;">0.42</td>
<td style="text-align: center;">0.58</td>
<td style="text-align: center;">0.04</td>
<td style="text-align: center;">0.38</td>
<td style="text-align: center;">0.61<!-- $removenext="2" --></td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: center;"><!-- $colspan="8" --></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: center;"><em>DDX3Y</em></td>
<td style="text-align: center;">0.05</td>
<td style="text-align: center;">0.33</td>
<td style="text-align: center;">0.78</td>
<td style="text-align: center;">Testis<!-- $rowspan="2" --></td>
<td style="text-align: center;">0.07</td>
<td style="text-align: center;">0.11</td>
<td style="text-align: center;">0.18</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><em>KDM6A</em></td>
<td style="text-align: center;">0.43</td>
<td style="text-align: center;">0.51</td>
<td style="text-align: center;">0.58</td>
<td style="text-align: center;">0.27</td>
<td style="text-align: center;">0.34</td>
<td style="text-align: center;">0.48<!-- $removenext="2" --></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: center;"><!-- $colspan="8" --></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: center;"><em>RASSF2</em></td>
<td style="text-align: center;">0.69</td>
<td style="text-align: center;">0.77</td>
<td style="text-align: center;">0.90</td>
<td style="text-align: center;">Leukocyte<!-- $rowspan="2" --></td>
<td style="text-align: center;">0.66</td>
<td style="text-align: center;">0.74</td>
<td style="text-align: center;">0.88</td>
</tr>
<tr class="even">
<td style="text-align: center;"><em>CYTIP</em></td>
<td style="text-align: center;">0.74</td>
<td style="text-align: center;">0.85</td>
<td style="text-align: center;">0.91</td>
<td style="text-align: center;">0.76</td>
<td style="text-align: center;">0.84</td>
<td style="text-align: center;">0.91<!-- $removenext="2" --></td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: center;"><!-- $colspan="8" --></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: center;"><em>MYOZ1</em></td>
<td style="text-align: center;">0.09</td>
<td style="text-align: center;">0.17</td>
<td style="text-align: center;">0.37</td>
<td style="text-align: center;">Skeletal muscle<!-- $rowspan="2" --></td>
<td style="text-align: center;">0.11</td>
<td style="text-align: center;">0.11</td>
<td style="text-align: center;">0.12</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><em>TNNI2</em></td>
<td style="text-align: center;">0.10</td>
<td style="text-align: center;">0.22</td>
<td style="text-align: center;">0.44</td>
<td style="text-align: center;">0.10</td>
<td style="text-align: center;">0.11</td>
<td style="text-align: center;">0.12<!-- $removenext="2" --></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: center;"><!-- $colspan="8" --></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: center;"><em>SCGB3A1</em></td>
<td style="text-align: center;">0.16</td>
<td style="text-align: center;">0.19</td>
<td style="text-align: center;">0.23</td>
<td style="text-align: center;">Placenta<!-- $rowspan="2" --></td>
<td style="text-align: center;">0.11</td>
<td style="text-align: center;">0.11</td>
<td style="text-align: center;">0.12</td>
</tr>
<tr class="even">
<td style="text-align: center;"><em>C19orf33</em></td>
<td style="text-align: center;">0.15</td>
<td style="text-align: center;">0.19</td>
<td style="text-align: center;">0.28</td>
<td style="text-align: center;">0.11</td>
<td style="text-align: center;">0.12</td>
<td style="text-align: center;">0.17<!-- $removenext="2" --></td>
<td style="text-align: center;"></td>
</tr>
</tbody>
</table>
</div>
<!-- default theme -->

<style>
  /* import google fonts */
  @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
  @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

  /* -------------------------------------------------- */
  /* global */
  /* -------------------------------------------------- */

  /* all elements */
  * {
    /* force sans-serif font unless specified otherwise */
    font-family: "Open Sans", "Helvetica", sans-serif;

    /* prevent text inflation on some mobile browsers */
    -webkit-text-size-adjust: none !important;
    -moz-text-size-adjust: none !important;
    -o-text-size-adjust: none !important;
    text-size-adjust: none !important;
  }

  @media only screen {
    /* "page" element */
    body {
      position: relative;
      box-sizing: border-box;
      font-size: 12pt;
      line-height: 1.5;
      max-width: 8.5in;
      margin: 20px auto;
      padding: 40px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* "page" element */
    body {
      padding: 20px;
      margin: 0;
      border-radius: 0;
      border: none;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
      background: none;
    }
  }

  /* -------------------------------------------------- */
  /* headings */
  /* -------------------------------------------------- */

  /* all headings */
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin: 20px 0;
    padding: 0;
    font-weight: bold;
  }

  /* biggest heading */
  h1 {
    margin: 40px 0;
    text-align: center;
  }

  /* second biggest heading */
  h2 {
    margin-top: 30px;
    padding-bottom: 5px;
    border-bottom: solid 1px #bdbdbd;
  }

  /* heading font sizes */
  h1 {
    font-size: 2em;
  }
  h2 {
    font-size: 1.5em;
  }
  h3 {
    font-size: 1.35em;
  }
  h4 {
    font-size: 1.25em;
  }
  h5 {
    font-size: 1.15em;
  }
  h6 {
    font-size: 1em;
  }

  /* -------------------------------------------------- */
  /* manuscript header */
  /* -------------------------------------------------- */

  /* manuscript title */
  header > h1 {
    margin: 0;
  }

  /* manuscript title caption text (ie "automatically generated on") */
  header + p {
    text-align: center;
    margin-top: 10px;
  }

  /* -------------------------------------------------- */
  /* text elements */
  /* -------------------------------------------------- */

  /* links */
  a {
    color: #2196f3;
    overflow-wrap: break-word;
  }

  /* superscripts and subscripts */
  sub,
  sup {
    /* prevent from affecting line height */
    line-height: 0;
  }

  /* unordered and ordered lists*/
  ul,
  ol {
    padding-left: 20px;
  }

  /* class for styling text semibold */
  .semibold {
    font-weight: 600;
  }

  /* class for styling elements horizontally left aligned */
  .left {
    display: block;
    text-align: left;
    margin-left: auto;
    margin-right: 0;
    justify-content: left;
  }

  /* class for styling elements horizontally centered */
  .center {
    display: block;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    justify-content: center;
  }

  /* class for styling elements horizontally right aligned */
  .right {
    display: block;
    text-align: right;
    margin-left: 0;
    margin-right: auto;
    justify-content: right;
  }

  /* -------------------------------------------------- */
  /* section elements */
  /* -------------------------------------------------- */

  /* horizontal divider line */
  hr {
    border: none;
    height: 1px;
    background: #bdbdbd;
  }

  /* paragraphs, horizontal dividers, figures, tables, code */
  p,
  hr,
  figure,
  table,
  pre {
    /* treat all as "paragraphs", with consistent vertical margins */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* figures */
  /* -------------------------------------------------- */

  /* figure */
  figure {
    max-width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure caption */
  figcaption {
    padding: 0;
    padding-top: 10px;
  }

  /* figure image element */
  figure > img,
  figure > svg {
    max-width: 100%;
    display: block;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure auto-number */
  img + figcaption > span:first-of-type,
  svg + figcaption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* tables */
  /* -------------------------------------------------- */

  /* table */
  table {
    border-collapse: collapse;
    border-spacing: 0;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* table cells */
  th,
  td {
    border: solid 1px #bdbdbd;
    padding: 10px;
    /* squash table if too wide for page by forcing line breaks */
    overflow-wrap: break-word;
    word-break: break-word;
  }

  /* header row and even rows */
  th,
  tr:nth-child(2n) {
    background-color: #fafafa;
  }

  /* odd rows */
  tr:nth-child(2n + 1) {
    background-color: #ffffff;
  }

  /* table caption */
  caption {
    text-align: left;
    padding: 0;
    padding-bottom: 10px;
  }

  /* table auto-number */
  table > caption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* code */
  /* -------------------------------------------------- */

  /* multi-line code block */
  pre {
    padding: 10px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
    break-inside: avoid;
    text-align: left;
  }

  /* inline code, ie code within normal text */
  :not(pre) > code {
    padding: 0 4px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
  }

  /* code text */
  /* apply all children, to reach syntax highlighting sub-elements */
  code,
  code * {
    /* force monospace font */
    font-family: "Source Code Pro", "Courier New", monospace;
  }

  /* -------------------------------------------------- */
  /* quotes */
  /* -------------------------------------------------- */

  /* quoted text */
  blockquote {
    margin: 0;
    padding: 0;
    border-left: 4px solid #bdbdbd;
    padding-left: 16px;
    break-inside: avoid;
  }

  /* -------------------------------------------------- */
  /* banners */
  /* -------------------------------------------------- */

  /* info banners */
  .banner {
    box-sizing: border-box;
    display: block;
    position: relative;
    width: 100%;
    margin-top: 20px;
    margin-bottom: 20px;
    padding: 20px;
    text-align: center;
  }

  /* paragraph in banner */
  .banner > p {
    margin: 0;
  }

  /* -------------------------------------------------- */
  /* highlight colors */
  /* -------------------------------------------------- */

  .white {
    background: #ffffff;
  }
  .lightgrey {
    background: #eeeeee;
  }
  .grey {
    background: #757575;
  }
  .darkgrey {
    background: #424242;
  }
  .black {
    background: #000000;
  }
  .lightred {
    background: #ffcdd2;
  }
  .lightyellow {
    background: #ffecb3;
  }
  .lightgreen {
    background: #dcedc8;
  }
  .lightblue {
    background: #e3f2fd;
  }
  .lightpurple {
    background: #f3e5f5;
  }
  .red {
    background: #f44336;
  }
  .orange {
    background: #ff9800;
  }
  .yellow {
    background: #ffeb3b;
  }
  .green {
    background: #4caf50;
  }
  .blue {
    background: #2196f3;
  }
  .purple {
    background: #9c27b0;
  }
  .white,
  .lightgrey,
  .lightred,
  .lightyellow,
  .lightgreen,
  .lightblue,
  .lightpurple,
  .orange,
  .yellow,
  .white a,
  .lightgrey a,
  .lightred a,
  .lightyellow a,
  .lightgreen a,
  .lightblue a,
  .lightpurple a,
  .orange a,
  .yellow a {
    color: #000000;
  }
  .grey,
  .darkgrey,
  .black,
  .red,
  .green,
  .blue,
  .purple,
  .grey a,
  .darkgrey a,
  .black a,
  .red a,
  .green a,
  .blue a,
  .purple a {
    color: #ffffff;
  }

  /* -------------------------------------------------- */
  /* buttons */
  /* -------------------------------------------------- */

  /* class for styling links like buttons */
  .button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    margin: 5px;
    padding: 10px 20px;
    font-size: 0.75em;
    font-weight: 600;
    text-transform: uppercase;
    text-decoration: none;
    letter-spacing: 1px;
    background: none;
    color: #2196f3;
    border: solid 1px #bdbdbd;
    border-radius: 5px;
  }

  /* buttons when hovered */
  .button:hover:not([disabled]),
  .icon_button:hover:not([disabled]) {
    cursor: pointer;
    background: #f5f5f5;
  }

  /* buttons when disabled */
  .button[disabled],
  .icon_button[disabled] {
    opacity: 0.35;
    pointer-events: none;
  }

  /* class for styling buttons containg only single icon */
  .icon_button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    text-decoration: none;
    margin: 0;
    padding: 0;
    background: none;
    border-radius: 5px;
    border: none;
    width: 20px;
    height: 20px;
    min-width: 20px;
    min-height: 20px;
  }

  /* icon button inner svg image */
  .icon_button > svg {
    height: 16px;
  }

  /* -------------------------------------------------- */
  /* icons */
  /* -------------------------------------------------- */

  /* class for styling icons inline with text */
  .inline_icon {
    height: 1em;
    position: relative;
    top: 0.125em;
  }

  /* -------------------------------------------------- */
  /* references */
  /* -------------------------------------------------- */

  .csl-entry {
    margin-top: 15px;
    margin-bottom: 15px;
  }

  /* -------------------------------------------------- */
  /* print control */
  /* -------------------------------------------------- */

  @media print {
    @page {
      /* suggested printing margin */
      margin: 0.5in;
    }

    /* document and "page" elements */
    html,
    body {
      margin: 0;
      padding: 0;
      width: 100%;
      height: 100%;
    }

    /* "page" element */
    body {
      font-size: 11pt !important;
      line-height: 1.35;
    }

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      margin: 15px 0;
    }

    /* figures and tables */
    figure,
    table {
      font-size: 0.85em;
    }

    /* table cells */
    th,
    td {
      padding: 5px;
    }

    /* shrink font awesome icons */
    i.fas,
    i.fab,
    i.far,
    i.fal {
      transform: scale(0.85);
    }

    /* decrease banner margins */
    .banner {
      margin-top: 15px;
      margin-bottom: 15px;
      padding: 15px;
    }

    /* class for centering an element vertically on its own page */
    .page_center {
      margin: auto;
      width: 100%;
      height: 100%;
      display: flex;
      align-items: center;
      vertical-align: middle;
      break-before: page;
      break-after: page;
    }

    /* always insert a page break before the element */
    .page_break_before {
      break-before: page;
    }

    /* always insert a page break after the element */
    .page_break_after {
      break-after: page;
    }

    /* avoid page break before the element */
    .page_break_before_avoid {
      break-before: avoid;
    }

    /* avoid page break after the element */
    .page_break_after_avoid {
      break-after: avoid;
    }

    /* avoid page break inside the element */
    .page_break_inside_avoid {
      break-inside: avoid;
    }
  }

  /* -------------------------------------------------- */
  /* override pandoc css quirks */
  /* -------------------------------------------------- */

  .sourceCode {
    /* prevent unsightly overflow in wide code blocks */
    overflow: auto !important;
  }

  div.sourceCode {
    /* prevent background fill on top-most code block  container */
    background: none !important;
  }

  .sourceCode * {
    /* force consistent line spacing */
    line-height: 1.5 !important;
  }

  div.sourceCode {
    /* style code block margins same as <pre> element */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* tablenos */
  /* -------------------------------------------------- */

  /* tablenos wrapper */
  .tablenos {
    width: 100%;
    margin: 20px 0;
  }

  .tablenos > table {
    /* move margins from table to table_wrapper to allow margin collapsing */
    margin: 0;
  }

  @media only screen {
    /* tablenos wrapper */
    .tablenos {
      /* show scrollbar on tables if necessary to prevent overflow */
      overflow-x: auto !important;
    }

    .tablenos th,
    .tablenos td {
      overflow-wrap: unset !important;
      word-break: unset !important;
    }

    /* table in wrapper */
    .tablenos table,
    .tablenos table * {
      /* don't break table words */
      overflow-wrap: normal !important;
    }
  }
</style>
<!-- 
    Plugin Core

    Functions needed for and shared across all first-party plugins.
-->

<script>
  // get element that is target of hash (from link element or url)
  function getHashTarget(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector(`[id="${id}"]`);
    if (!target) return;

    // if figure or table, modify target to get expected element
    if (id.indexOf("fig:") === 0) target = target.querySelector("figure");
    if (id.indexOf("tbl:") === 0) target = target.querySelector("table");

    return target;
  }

  // get position/dimensions of element or viewport
  function getRectInView(element) {
    let rect = {};
    rect.left = 0;
    rect.top = 0;
    rect.right = document.documentElement.clientWidth;
    rect.bottom = document.documentElement.clientHeight;
    let style = {};

    if (element instanceof HTMLElement) {
      rect = element.getBoundingClientRect();
      style = window.getComputedStyle(element);
    }

    const margin = {};
    margin.left = parseFloat(style.marginLeftWidth) || 0;
    margin.top = parseFloat(style.marginTopWidth) || 0;
    margin.right = parseFloat(style.marginRightWidth) || 0;
    margin.bottom = parseFloat(style.marginBottomWidth) || 0;

    const border = {};
    border.left = parseFloat(style.borderLeftWidth) || 0;
    border.top = parseFloat(style.borderTopWidth) || 0;
    border.right = parseFloat(style.borderRightWidth) || 0;
    border.bottom = parseFloat(style.borderBottomWidth) || 0;

    const newRect = {};
    newRect.left = rect.left + margin.left + border.left;
    newRect.top = rect.top + margin.top + border.top;
    newRect.right = rect.right + margin.right + border.right;
    newRect.bottom = rect.bottom + margin.bottom + border.bottom;
    newRect.width = newRect.right - newRect.left;
    newRect.height = newRect.bottom - newRect.top;

    return newRect;
  }

  // get position of element relative to page
  function getRectInPage(element) {
    const rect = getRectInView(element);
    const body = getRectInView(document.body);

    const newRect = {};
    newRect.left = rect.left - body.left;
    newRect.top = rect.top - body.top;
    newRect.right = rect.right - body.left;
    newRect.bottom = rect.bottom - body.top;
    newRect.width = rect.width;
    newRect.height = rect.height;

    return newRect;
  }

  // get closest element before specified element that matches query
  function firstBefore(element, query) {
    while (element && element !== document.body && !element.matches(query))
      element = element.previousElementSibling || element.parentNode;

    return element;
  }

  // check if element is part of collapsed heading
  function isCollapsed(element) {
    while (element && element !== document.body) {
      if (element.dataset.collapsed === "true") return true;
      element = element.parentNode;
    }
    return false;
  }

  // expand any collapsed parent containers of element if necessary
  function expandElement(element) {
    if (isCollapsed(element)) {
      // accordion plugin
      const heading = firstBefore(element, "h2");
      if (heading) heading.click();
      // details/summary HTML element
      const summary = firstBefore(element, "summary");
      if (summary) summary.click();
    }
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);

    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // get list of elements after a start element up to element matching query
  function nextUntil(element, query, exclude) {
    const elements = [];
    while (((element = element.nextElementSibling), element)) {
      if (element.matches(query)) break;
      if (!element.matches(exclude)) elements.push(element);
    }
    return elements;
  }
</script>
<!--
  Accordion Plugin

  Allows sections of content under h2 headings to be collapsible.
-->

<script type="module">
  // whether to always start expanded ('false'), always start collapsed
  // ('true'), or start collapsed when screen small ('auto')
  const startCollapsed = "auto";

  // start script
  function start() {
    // run through each <h2> heading
    const headings = document.querySelectorAll("h2");
    for (const heading of headings) {
      addArrow(heading);

      // start expanded/collapsed based on option
      if (
        startCollapsed === "true" ||
        (startCollapsed === "auto" && isSmallScreen()) ||
        heading.dataset.collapsed === "true"
      )
        collapseHeading(heading);
      else expandElement(heading);
    }

    // attach hash change listener to window
    window.addEventListener("hashchange", onHashChange);
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) goToElement(target);
  }

  // add arrow to heading
  function addArrow(heading) {
    // add arrow button
    const arrow = document.createElement("button");
    arrow.innerHTML = document.querySelector(".icon_angle_down").innerHTML;
    arrow.classList.add("icon_button", "accordion_arrow");
    heading.insertBefore(arrow, heading.firstChild);

    // attach click listener to heading and button
    heading.addEventListener("click", onHeadingClick);
    arrow.addEventListener("click", onArrowClick);
  }

  // determine if on mobile-like device with small screen
  function isSmallScreen() {
    return Math.min(window.innerWidth, window.innerHeight) < 480;
  }

  // when <h2> heading is clicked
  function onHeadingClick(event) {
    // only collapse if <h2> itself is target of click (eg, user did
    // not click on anchor within <h2>)
    if (event.target === this) toggleCollapse(this);
  }

  // when arrow button is clicked
  function onArrowClick() {
    toggleCollapse(this.parentNode);
  }

  // collapse section if expanded, expand if collapsed
  function toggleCollapse(heading) {
    if (heading.dataset.collapsed === "false") collapseHeading(heading);
    else expandElement(heading);
  }

  // elements to exclude from collapse, such as table of contents panel,
  // hypothesis panel, etc
  const exclude = "#toc_panel, div.annotator-frame, #lightbox_overlay";

  // collapse section
  function collapseHeading(heading) {
    heading.setAttribute("data-collapsed", "true");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "true");
  }

  // expand section
  function expandElement(heading) {
    heading.setAttribute("data-collapsed", "false");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "false");
  }

  // get list of elements between this <h2> and next <h2> or <h1>
  // ("children" of the <h2> section)
  function getChildren(heading) {
    return nextUntil(heading, "h2, h1", exclude);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
  <!-- modified from: https://fontawesome.com/icons/angle-down -->
  <svg width="16" height="16" viewBox="0 0 448 512">
    <path
      fill="currentColor"
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* accordion arrow button */
    .accordion_arrow {
      margin-right: 10px;
    }

    /* arrow icon when <h2> data-collapsed attribute true */
    h2[data-collapsed="true"] > .accordion_arrow > svg {
      transform: rotate(-90deg);
    }

    /* all elements (except <h2>'s) when data-collapsed attribute true */
    *:not(h2)[data-collapsed="true"] {
      display: none;
    }

    /* accordion arrow button when hovered and <h2>'s when hovered */
    .accordion_arrow:hover,
    h2[data-collapsed="true"]:hover,
    h2[data-collapsed="false"]:hover {
      cursor: pointer;
    }
  }

  /* always hide accordion arrow button on print */
  @media only print {
    .accordion_arrow {
      display: none;
    }
  }
</style>
<!--
  Anchors Plugin

  Adds an anchor next to each of a certain type of element that provides a
  human-readable url to that specific item/position in the document (e.g.
  "manuscript.html#abstract"). It also makes it such that scrolling out of view
  of a target removes its identifier from the url.
-->

<script type="module">
  // which types of elements to add anchors next to, in "document.querySelector"
  // format
  const typesQuery =
    'h1, h2, h3, div[id^="fig:"], div[id^="tbl:"], span[id^="eq:"]';

  // start script
  function start() {
    // add anchor to each element of specified types
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) addAnchor(element);

    // attach scroll listener to window
    window.addEventListener("scroll", onScroll);
  }

  // when window is scrolled
  function onScroll() {
    // if url has hash and user has scrolled out of view of hash
    // target, remove hash from url
    const tolerance = 100;
    const target = getHashTarget();
    if (target) {
      if (
        target.getBoundingClientRect().top > window.innerHeight + tolerance ||
        target.getBoundingClientRect().bottom < 0 - tolerance
      )
        history.pushState(null, null, " ");
    }
  }

  // add anchor to element
  function addAnchor(element) {
    let addTo; // element to add anchor button to

    // if figure or table, modify withId and addTo to get expected
    // elements
    if (element.id.indexOf("fig:") === 0) {
      addTo = element.querySelector("figcaption");
    } else if (element.id.indexOf("tbl:") === 0) {
      addTo = element.querySelector("caption");
    } else if (element.id.indexOf("eq:") === 0) {
      addTo = element.querySelector(".eqnos-number");
    }

    addTo = addTo || element;
    const id = element.id || null;

    // do not add anchor if element doesn't have assigned id.
    // id is generated by pandoc and is assumed to be unique and
    // human-readable
    if (!id) return;

    // create anchor button
    const anchor = document.createElement("a");
    anchor.innerHTML = document.querySelector(".icon_link").innerHTML;
    anchor.title = "Link to this part of the document";
    anchor.classList.add("icon_button", "anchor");
    anchor.dataset.ignore = "true";
    anchor.href = "#" + id;
    addTo.appendChild(anchor);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- link icon -->

<template class="icon_link">
  <!-- modified from: https://fontawesome.com/icons/link -->
  <svg width="16" height="16" viewBox="0 0 512 512">
    <path
      fill="currentColor"
      d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* anchor button */
    .anchor {
      opacity: 0;
      margin-left: 5px;
    }

    /* anchor buttons within <h2>'s */
    h2 .anchor {
      margin-left: 10px;
    }

    /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
    *:hover > .anchor,
    .anchor:hover,
    .anchor:focus {
      opacity: 1;
    }

    /* anchor button when hovered */
    .anchor:hover {
      cursor: pointer;
    }
  }

  /* always show anchor button on devices with no mouse/hover ability */
  @media (hover: none) {
    .anchor {
      opacity: 1;
    }
  }

  /* always hide anchor button on print */
  @media only print {
    .anchor {
      display: none;
    }
  }
</style>
<!-- 
    Attributes Plugin

    Allows arbitrary HTML attributes to be attached to (almost) any element.
    Place an HTML comment inside or next to the desired element with the content:
    $attribute="value"
-->

<script type="module">
  // start script
  function start() {
    // get list of comments in document
    const comments = findComments();

    for (const comment of comments)
      if (comment.parentElement)
        addAttributes(comment.parentElement, comment.nodeValue.trim());
  }

  // add html attributes to specified element based on string of
  // html attributes and values
  function addAttributes(element, text) {
    // regex's for finding attribute/value pairs in the format of
    // attribute="value" or attribute='value
    const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
    const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

    // loop through attribute/value pairs
    let match;
    while ((match = text.match(regex2) || text.match(regex1))) {
      // get attribute and value from regex capture groups
      let attribute = match[1];
      let value = match[2];

      // remove from string
      text = text.substring(match.index + match[0].length);

      if (!attribute || !value) break;

      // set attribute of parent element
      try {
        element.setAttribute(attribute, value);
      } catch (error) {
        console.log(error);
      }

      // special case for colspan
      if (attribute === "colspan") removeTableCells(element, value);
      if (attribute === "removenext") removeTableCells(element, value);
    }
  }

  // get list of comment elements in document
  function findComments() {
    const comments = [];

    // iterate over comment nodes in document
    function acceptNode(node) {
      return NodeFilter.FILTER_ACCEPT;
    }
    const iterator = document.createNodeIterator(
      document.body,
      NodeFilter.SHOW_COMMENT,
      acceptNode
    );
    let node;
    while ((node = iterator.nextNode())) comments.push(node);

    return comments;
  }

  // remove certain number of cells after specified cell
  function removeTableCells(cell, number) {
    number = parseInt(number);
    if (!number) return;

    // remove elements
    for (; number > 1; number--) {
      if (cell.nextElementSibling) cell.nextElementSibling.remove();
    }
  }

  // start script on DOMContentLoaded instead of load to ensure this plugins
  // runs before other plugins
  window.addEventListener("DOMContentLoaded", start);
</script>
<!--
  Jump to First Plugin

  Adds a button next to each reference entry, figure, and table that jumps the
  page to the first occurrence of a link to that item in the manuscript.
-->

<script type="module">
  // whether to add buttons next to reference entries
  const references = "true";
  // whether to add buttons next to figures
  const figures = "true";
  // whether to add buttons next to tables
  const tables = "true";

  // start script
  function start() {
    if (references !== "false")
      makeButtons(`div[id^="ref-"]`, ".csl-left-margin", "reference");
    if (figures !== "false")
      makeButtons(`div[id^="fig:"]`, "figcaption", "figure");
    if (tables !== "false") makeButtons(`div[id^="tbl:"]`, "caption", "table");
  }

  // when jump button clicked
  function onButtonClick() {
    const first = getFirstOccurrence(this.dataset.id);
    if (!first) return;

    // update url hash so navigating "back" in history will return user to button
    window.location.hash = this.dataset.id;
    // scroll to link
    const timeout = function () {
      goToElement(first, window.innerHeight * 0.5);
    };
    window.setTimeout(timeout, 0);
  }

  // get first occurrence of link to item in document
  function getFirstOccurrence(id) {
    let query = "a";
    query += '[href="#' + id + '"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelector(query);
  }

  // add button next to each reference entry, figure, or table
  function makeButtons(query, containerQuery, subject) {
    const elements = document.querySelectorAll(query);
    for (const element of elements) {
      const id = element.id;
      const buttonContainer = element.querySelector(containerQuery);
      const first = getFirstOccurrence(id);

      // if can't find link to reference or place to put button, ignore
      if (!first || !buttonContainer) continue;

      // make jump button
      let button = document.createElement("button");
      button.classList.add("icon_button", "jump_arrow");
      button.title = `Jump to the first occurrence of this ${subject} in the document`;
      const icon = document.querySelector(".icon_angle_double_up");
      button.innerHTML = icon.innerHTML;
      button.dataset.id = id;
      button.dataset.ignore = "true";
      button.addEventListener("click", onButtonClick);
      buttonContainer.prepend(button);
    }
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
  <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
  <svg width="16" height="16" viewBox="0 0 320 512">
    <path
      fill="currentColor"
      d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* jump button */
    .jump_arrow {
      position: relative;
      top: 0.125em;
      margin-right: 5px;
    }
  }

  /* always hide jump button on print */
  @media only print {
    .jump_arrow {
      display: none;
    }
  }
</style>
<!-- 
    Lightbox Plugin

    Makes it such that when a user clicks on an image, the image fills the
    screen and the user can pan/drag/zoom the image and navigate between other
    images in the document.
-->

<script type="module">
  // list of possible zoom/scale factors
  const zooms =
    "0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1, 1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8";
  // whether to fit image to view ('fit'), display at 100% and shrink if
  // necessary ('shrink'), or always display at 100% ('100')
  const defaultZoom = "fit";
  // whether to zoom in/out toward center of view ('true') or mouse ('false')
  const centerZoom = "false";

  // start script
  function start() {
    // run through each <img> element
    const imgs = document.querySelectorAll("figure > img");
    let count = 1;
    for (const img of imgs) {
      img.classList.add("lightbox_document_img");
      img.dataset.number = count;
      img.dataset.total = imgs.length;
      img.addEventListener("click", openLightbox);
      count++;
    }

    // attach mouse and key listeners to window
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("keyup", onKeyUp);
  }

  // when mouse is moved anywhere in window
  function onWindowMouseMove(event) {
    window.mouseX = event.clientX;
    window.mouseY = event.clientY;
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("lightbox_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("lightbox_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeLightbox();
        break;
    }
  }

  // open lightbox
  function openLightbox() {
    const lightbox = makeLightbox(this);
    if (!lightbox) return;

    blurBody(lightbox);
    document.body.appendChild(lightbox);
  }

  // make lightbox
  function makeLightbox(img) {
    // delete lightbox if it exists, start fresh
    closeLightbox();

    // create screen overlay containing lightbox
    const overlay = document.createElement("div");
    overlay.id = "lightbox_overlay";

    // create image info boxes
    const numberInfo = document.createElement("div");
    const zoomInfo = document.createElement("div");
    numberInfo.id = "lightbox_number_info";
    zoomInfo.id = "lightbox_zoom_info";

    // create container for image
    const imageContainer = document.createElement("div");
    imageContainer.id = "lightbox_image_container";
    const lightboxImg = makeLightboxImg(
      img,
      imageContainer,
      numberInfo,
      zoomInfo
    );
    imageContainer.appendChild(lightboxImg);

    // create bottom container for caption and navigation buttons
    const bottomContainer = document.createElement("div");
    bottomContainer.id = "lightbox_bottom_container";
    const caption = makeCaption(img);
    const prevButton = makePrevButton(img);
    const nextButton = makeNextButton(img);
    bottomContainer.appendChild(prevButton);
    bottomContainer.appendChild(caption);
    bottomContainer.appendChild(nextButton);

    // attach top middle and bottom to overlay
    overlay.appendChild(numberInfo);
    overlay.appendChild(zoomInfo);
    overlay.appendChild(imageContainer);
    overlay.appendChild(bottomContainer);

    return overlay;
  }

  // make <img> object that is intuitively draggable and zoomable
  function makeLightboxImg(sourceImg, container, numberInfoBox, zoomInfoBox) {
    // create copy of source <img>
    const img = sourceImg.cloneNode(true);
    img.classList.remove("lightbox_document_img");
    img.removeAttribute("id");
    img.removeAttribute("width");
    img.removeAttribute("height");
    img.style.position = "unset";
    img.style.margin = "0";
    img.style.padding = "0";
    img.style.width = "";
    img.style.height = "";
    img.style.minWidth = "";
    img.style.minHeight = "";
    img.style.maxWidth = "";
    img.style.maxHeight = "";
    img.id = "lightbox_img";

    // build sorted list of zoomSteps
    const zoomSteps = zooms.split(/[^0-9.]/).map((step) => parseFloat(step));
    zoomSteps.sort((a, b) => a - b);

    // <img> object property variables
    let zoom = 1;
    let translateX = 0;
    let translateY = 0;
    let clickMouseX = undefined;
    let clickMouseY = undefined;
    let clickTranslateX = undefined;
    let clickTranslateY = undefined;

    updateNumberInfo();

    // update image numbers displayed in info box
    function updateNumberInfo() {
      numberInfoBox.innerHTML =
        sourceImg.dataset.number + " of " + sourceImg.dataset.total;
    }

    // update zoom displayed in info box
    function updateZoomInfo() {
      let zoomInfo = zoom * 100;
      if (!Number.isInteger(zoomInfo)) zoomInfo = zoomInfo.toFixed(2);
      zoomInfoBox.innerHTML = zoomInfo + "%";
    }

    // move to closest zoom step above current zoom
    const zoomIn = function () {
      for (const zoomStep of zoomSteps) {
        if (zoomStep > zoom) {
          zoom = zoomStep;
          break;
        }
      }
      updateTransform();
    };

    // move to closest zoom step above current zoom
    const zoomOut = function () {
      zoomSteps.reverse();
      for (const zoomStep of zoomSteps) {
        if (zoomStep < zoom) {
          zoom = zoomStep;
          break;
        }
      }
      zoomSteps.reverse();

      updateTransform();
    };

    // update display of <img> based on scale/translate properties
    const updateTransform = function () {
      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      // get new width/height after scale
      const rect = img.getBoundingClientRect();
      // limit translate
      translateX = Math.max(translateX, -rect.width / 2);
      translateX = Math.min(translateX, rect.width / 2);
      translateY = Math.max(translateY, -rect.height / 2);
      translateY = Math.min(translateY, rect.height / 2);

      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      updateZoomInfo();
    };

    // fit <img> to container
    const fit = function () {
      // no x/y offset, 100% zoom by default
      translateX = 0;
      translateY = 0;
      zoom = 1;

      // widths of <img> and container
      const imgWidth = img.naturalWidth;
      const imgHeight = img.naturalHeight;
      const containerWidth = parseFloat(
        window.getComputedStyle(container).width
      );
      const containerHeight = parseFloat(
        window.getComputedStyle(container).height
      );

      // how much zooming is needed to fit <img> to container
      const xRatio = imgWidth / containerWidth;
      const yRatio = imgHeight / containerHeight;
      const maxRatio = Math.max(xRatio, yRatio);
      const newZoom = 1 / maxRatio;

      // fit <img> to container according to option
      if (defaultZoom === "shrink") {
        if (maxRatio > 1) zoom = newZoom;
      } else if (defaultZoom === "fit") zoom = newZoom;

      updateTransform();
    };

    // when mouse wheel is rolled anywhere in container
    const onContainerWheel = function (event) {
      if (!event) return;

      // let ctrl + mouse wheel to zoom behave as normal
      if (event.ctrlKey) return;

      // prevent normal scroll behavior
      event.preventDefault();
      event.stopPropagation();

      // point around which to scale img
      const viewRect = container.getBoundingClientRect();
      const viewX = (viewRect.left + viewRect.right) / 2;
      const viewY = (viewRect.top + viewRect.bottom) / 2;
      const originX = centerZoom === "true" ? viewX : mouseX;
      const originY = centerZoom === "true" ? viewY : mouseY;

      // get point on image under origin
      const oldRect = img.getBoundingClientRect();
      const oldPercentX = (originX - oldRect.left) / oldRect.width;
      const oldPercentY = (originY - oldRect.top) / oldRect.height;

      // increment/decrement zoom
      if (event.deltaY < 0) zoomIn();
      if (event.deltaY > 0) zoomOut();

      // get offset between previous image point and origin
      const newRect = img.getBoundingClientRect();
      const offsetX = originX - (newRect.left + newRect.width * oldPercentX);
      const offsetY = originY - (newRect.top + newRect.height * oldPercentY);

      // translate image to keep image point under origin
      translateX += offsetX;
      translateY += offsetY;

      // perform translate
      updateTransform();
    };

    // when container is clicked
    function onContainerClick(event) {
      // if container itself is target of click, and not other
      // element above it
      if (event.target === this) closeLightbox();
    }

    // when mouse button is pressed on image
    const onImageMouseDown = function (event) {
      // store original mouse position relative to image
      clickMouseX = window.mouseX;
      clickMouseY = window.mouseY;
      clickTranslateX = translateX;
      clickTranslateY = translateY;
      event.stopPropagation();
      event.preventDefault();
    };

    // when mouse button is released anywhere in window
    const onWindowMouseUp = function (event) {
      // reset original mouse position
      clickMouseX = undefined;
      clickMouseY = undefined;
      clickTranslateX = undefined;
      clickTranslateY = undefined;

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mouseup", onWindowMouseUp);
    };

    // when mouse is moved anywhere in window
    const onWindowMouseMove = function (event) {
      if (
        clickMouseX === undefined ||
        clickMouseY === undefined ||
        clickTranslateX === undefined ||
        clickTranslateY === undefined
      )
        return;

      // offset image based on original and current mouse position
      translateX = clickTranslateX + window.mouseX - clickMouseX;
      translateY = clickTranslateY + window.mouseY - clickMouseY;
      updateTransform();
      event.preventDefault();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mousemove", onWindowMouseMove);
    };

    // when window is resized
    const onWindowResize = function (event) {
      fit();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("resize", onWindowResize);
    };

    // attach the necessary event listeners
    img.addEventListener("dblclick", fit);
    img.addEventListener("mousedown", onImageMouseDown);
    container.addEventListener("wheel", onContainerWheel);
    container.addEventListener("mousedown", onContainerClick);
    container.addEventListener("touchstart", onContainerClick);
    window.addEventListener("mouseup", onWindowMouseUp);
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("resize", onWindowResize);

    // run fit() after lightbox atttached to document and <img> Loaded
    // so needed container and img dimensions available
    img.addEventListener("load", fit);

    return img;
  }

  // make caption
  function makeCaption(img) {
    const caption = document.createElement("div");
    caption.id = "lightbox_caption";
    const captionSource = img.nextElementSibling;
    if (captionSource.tagName.toLowerCase() === "figcaption") {
      const captionCopy = makeCopy(captionSource);
      caption.innerHTML = captionCopy.innerHTML;
    }

    caption.addEventListener("touchstart", function (event) {
      event.stopPropagation();
    });

    return caption;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // make button to jump to previous image in document
  function makePrevButton(img) {
    const prevButton = document.createElement("button");
    prevButton.id = "lightbox_prev_button";
    prevButton.title = "Jump to the previous image in the document [←]";
    prevButton.classList.add("icon_button", "lightbox_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;

    // attach click listeners to button
    prevButton.addEventListener("click", function () {
      getPrevImg(img).click();
    });

    return prevButton;
  }

  // make button to jump to next image in document
  function makeNextButton(img) {
    const nextButton = document.createElement("button");
    nextButton.id = "lightbox_next_button";
    nextButton.title = "Jump to the next image in the document [→]";
    nextButton.classList.add("icon_button", "lightbox_button");
    nextButton.innerHTML = document.querySelector(
      ".icon_caret_right"
    ).innerHTML;

    // attach click listeners to button
    nextButton.addEventListener("click", function () {
      getNextImg(img).click();
    });

    return nextButton;
  }

  // get previous image in document
  function getPrevImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if < 1
    if (index - 1 >= 0) index--;
    else index = imgs.length - 1;
    return imgs[index];
  }

  // get next image in document
  function getNextImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if > total
    if (index + 1 <= imgs.length - 1) index++;
    else index = 0;
    return imgs[index];
  }

  // close lightbox
  function closeLightbox() {
    focusBody();

    const lightbox = document.getElementById("lightbox_overlay");
    if (lightbox) lightbox.remove();
  }

  // make all elements behind lightbox non-focusable
  function blurBody(overlay) {
    const all = document.querySelectorAll("*");
    for (const element of all) element.tabIndex = -1;
    document.body.classList.add("body_no_scroll");
  }

  // make all elements focusable again
  function focusBody() {
    const all = document.querySelectorAll("*");
    for (const element of all) element.removeAttribute("tabIndex");
    document.body.classList.remove("body_no_scroll");
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* regular <img> in document when hovered */
    img.lightbox_document_img:hover {
      cursor: pointer;
    }

    .body_no_scroll {
      overflow: hidden !important;
    }

    /* screen overlay */
    #lightbox_overlay {
      display: flex;
      flex-direction: column;
      position: fixed;
      left: 0;
      top: 0;
      right: 0;
      bottom: 0;
      background: rgba(0, 0, 0, 0.75);
      z-index: 3;
    }

    /* middle area containing lightbox image */
    #lightbox_image_container {
      flex-grow: 1;
      display: flex;
      justify-content: center;
      align-items: center;
      overflow: hidden;
      position: relative;
      padding: 20px;
    }

    /* bottom area containing caption */
    #lightbox_bottom_container {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 100px;
      min-height: 100px;
      max-height: 100px;
      background: rgba(0, 0, 0, 0.5);
    }

    /* image number info text box */
    #lightbox_number_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      left: 2px;
      top: 0;
      z-index: 4;
    }

    /* zoom info text box */
    #lightbox_zoom_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      right: 2px;
      top: 0;
      z-index: 4;
    }

    /* copy of image caption */
    #lightbox_caption {
      box-sizing: border-box;
      display: inline-block;
      width: 100%;
      max-height: 100%;
      padding: 10px 0;
      text-align: center;
      overflow-y: auto;
      color: #ffffff;
    }

    /* navigation previous/next button */
    .lightbox_button {
      width: 100px;
      height: 100%;
      min-width: 100px;
      min-height: 100%;
      color: #ffffff;
    }

    /* navigation previous/next button when hovered */
    .lightbox_button:hover {
      background: none !important;
    }

    /* navigation button icon */
    .lightbox_button > svg {
      height: 25px;
    }

    /* figure auto-number */
    #lightbox_caption > span:first-of-type {
      font-weight: bold;
      margin-right: 5px;
    }

    /* lightbox image when hovered */
    #lightbox_img:hover {
      cursor: grab;
    }

    /* lightbox image when grabbed */
    #lightbox_img:active {
      cursor: grabbing;
    }
  }

  /* when on screen < 480px wide */
  @media only screen and (max-width: 480px) {
    /* make navigation buttons skinnier on small screens to make more room for caption text */
    .lightbox_button {
      width: 50px;
      min-width: 50px;
    }
  }

  /* always hide lightbox on print */
  @media only print {
    #lightbox_overlay {
      display: none;
    }
  }
</style>
<!-- 
  Link Highlight Plugin

  Makes it such that when a user hovers or focuses a link, other links that have
  the same target will be highlighted. It also makes it such that when clicking
  a link, the target of the link (eg reference, figure, table) is briefly
  highlighted.
-->

<script type="module">
  // whether to also highlight links that go to external urls
  const externalLinks = "false";
  // whether user must click off to unhighlight instead of just
  // un-hovering
  const clickUnhighlight = "false";
  // whether to also highlight links that are unique
  const highlightUnique = "true";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach mouse and focus listeners to link
      link.addEventListener("mouseenter", onLinkFocus);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("mouseleave", onLinkUnhover);
    }

    // attach click and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("hashchange", onHashChange);

    // run hash change on window load in case user has navigated
    // directly to hash
    onHashChange();
  }

  // when link is focused (tabbed to) or hovered
  function onLinkFocus() {
    highlight(this);
  }

  // when link is unhovered
  function onLinkUnhover() {
    if (clickUnhighlight !== "true") unhighlightAll();
  }

  // when the mouse is clicked anywhere in window
  function onClick(event) {
    unhighlightAll();
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) glowElement(target);
  }

  // start glow sequence on an element
  function glowElement(element) {
    const startGlow = function () {
      onGlowEnd();
      element.dataset.glow = "true";
      element.addEventListener("animationend", onGlowEnd);
    };
    const onGlowEnd = function () {
      element.removeAttribute("data-glow");
      element.removeEventListener("animationend", onGlowEnd);
    };
    startGlow();
  }

  // highlight link and all others with same target
  function highlight(link) {
    // force unhighlight all to start fresh
    unhighlightAll();

    // get links with same target
    if (!link) return;
    const sameLinks = getSameLinks(link);

    // if link unique and option is off, exit and don't highlight
    if (sameLinks.length <= 1 && highlightUnique !== "true") return;

    // highlight all same links, and "select" (special highlight) this
    // one
    for (const sameLink of sameLinks) {
      if (sameLink === link) sameLink.setAttribute("data-selected", "true");
      else sameLink.setAttribute("data-highlighted", "true");
    }
  }

  // unhighlight all links
  function unhighlightAll() {
    const links = getLinks();
    for (const link of links) {
      link.setAttribute("data-selected", "false");
      link.setAttribute("data-highlighted", "false");
    }
  }

  // get links with same target
  function getSameLinks(link) {
    const results = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        results.push(otherLink);
    }
    return results;
  }

  // get all links of types we wish to handle
  function getLinks() {
    let query = "a";
    if (externalLinks !== "true") query += '[href^="#"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelectorAll(query);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<style>
  @media only screen {
    /* anything with data-highlighted attribute true */
    [data-highlighted="true"] {
      background: #ffeb3b;
    }

    /* anything with data-selected attribute true */
    [data-selected="true"] {
      background: #ff8a65 !important;
    }

    /* animation definition for glow */
    @keyframes highlight_glow {
      0% {
        background: none;
      }
      10% {
        background: #bbdefb;
      }
      100% {
        background: none;
      }
    }

    /* anything with data-glow attribute true */
    [data-glow="true"] {
      animation: highlight_glow 2s;
    }
  }
</style>
<!--
  Table of Contents Plugin

  Provides a "table of contents" (toc) panel on the side of the document that
  allows the user to conveniently navigate between sections of the document.
-->

<script type="module">
  // which types of elements to add links for, in "document.querySelector" format
  const typesQuery = "h1, h2, h3";
  // whether toc starts open. use 'true' or 'false', or 'auto' to
  // use 'true' behavior when screen wide enough and 'false' when not
  const startOpen = "false";
  // whether toc closes when clicking on toc link. use 'true' or
  // 'false', or 'auto' to use 'false' behavior when screen wide
  // enough and 'true' when not
  const clickClose = "auto";
  // if list item is more than this many characters, text will be
  // truncated
  const charLimit = "50";
  // whether or not to show bullets next to each toc item
  const bullets = "false";

  // start script
  function start() {
    // make toc panel and populate with entries (links to document
    // sections)
    const panel = makePanel();
    if (!panel) return;
    makeEntries(panel);
    // attach panel to document after making entries, so 'toc' heading
    // in panel isn't included in toc
    document.body.insertBefore(panel, document.body.firstChild);

    // initial panel state
    if (startOpen === "true" || (startOpen === "auto" && !isSmallScreen()))
      openPanel();
    else closePanel();

    // attach click, scroll, and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("scroll", onScroll);
    window.addEventListener("hashchange", onScroll);
    window.addEventListener("keyup", onKeyUp);
    onScroll();

    // add class to push document body down out of way of toc button
    document.body.classList.add("toc_body_nudge");
  }

  // determine if screen wide enough to fit toc panel
  function isSmallScreen() {
    // in default theme:
    // 816px = 8.5in = width of "page" (<body>) element
    // 260px = min width of toc panel (*2 for both sides of <body>)
    return window.innerWidth < 816 + 260 * 2;
  }

  // when mouse is clicked anywhere in window
  function onClick() {
    if (isSmallScreen()) closePanel();
  }

  // when window is scrolled or hash changed
  function onScroll() {
    highlightViewed();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    // close on esc
    if (event.key === "Escape") closePanel();
  }

  // find entry of currently viewed document section in toc and highlight
  function highlightViewed() {
    const firstId = getFirstInView(typesQuery);

    // get toc entries (links), unhighlight all, then highlight viewed
    const list = document.getElementById("toc_list");
    if (!firstId || !list) return;
    const links = list.querySelectorAll("a");
    for (const link of links) link.dataset.viewing = "false";
    const link = list.querySelector('a[href="#' + firstId + '"]');
    if (!link) return;
    link.dataset.viewing = "true";
  }

  // get first or previous toc listed element in top half of view
  function getFirstInView(query) {
    // get all elements matching query and with id
    const elements = document.querySelectorAll(query);
    const elementsWithIds = [];
    for (const element of elements) {
      if (element.id) elementsWithIds.push(element);
    }

    // get first or previous element in top half of view
    for (let i = 0; i < elementsWithIds.length; i++) {
      const element = elementsWithIds[i];
      const prevElement = elementsWithIds[Math.max(0, i - 1)];
      if (element.getBoundingClientRect().top >= 0) {
        if (element.getBoundingClientRect().top < window.innerHeight / 2)
          return element.id;
        else return prevElement.id;
      }
    }
  }

  // make panel
  function makePanel() {
    // create panel
    const panel = document.createElement("div");
    panel.id = "toc_panel";
    if (bullets === "true") panel.dataset.bullets = "true";

    // create header
    const header = document.createElement("div");
    header.id = "toc_header";

    // create toc button
    const button = document.createElement("button");
    button.id = "toc_button";
    button.innerHTML = document.querySelector(".icon_th_list").innerHTML;
    button.title = "Table of Contents";
    button.classList.add("icon_button");

    // create header text
    const text = document.createElement("h4");
    text.innerHTML = "Table of Contents";

    // create container for toc list
    const list = document.createElement("div");
    list.id = "toc_list";

    // attach click listeners
    panel.addEventListener("click", onPanelClick);
    header.addEventListener("click", onHeaderClick);
    button.addEventListener("click", onButtonClick);

    // attach elements
    header.appendChild(button);
    header.appendChild(text);
    panel.appendChild(header);
    panel.appendChild(list);

    return panel;
  }

  // create toc entries (links) to each element of the specified types
  function makeEntries(panel) {
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) {
      // do not add link if element doesn't have assigned id
      if (!element.id) continue;

      // create link/list item
      const link = document.createElement("a");
      link.classList.add("toc_link");
      switch (element.tagName.toLowerCase()) {
        case "h1":
          link.dataset.level = "1";
          break;
        case "h2":
          link.dataset.level = "2";
          break;
        case "h3":
          link.dataset.level = "3";
          break;
        case "h4":
          link.dataset.level = "4";
          break;
      }
      link.title = element.innerText;
      let text = element.innerText;
      if (text.length > charLimit) text = text.slice(0, charLimit) + "...";
      link.innerHTML = text;
      link.href = "#" + element.id;
      link.addEventListener("click", onLinkClick);

      // attach link
      panel.querySelector("#toc_list").appendChild(link);
    }
  }

  // when panel is clicked
  function onPanelClick(event) {
    // stop click from propagating to window/document and closing panel
    event.stopPropagation();
  }

  // when header itself is clicked
  function onHeaderClick(event) {
    togglePanel();
  }

  // when button is clicked
  function onButtonClick(event) {
    togglePanel();
    // stop header underneath button from also being clicked
    event.stopPropagation();
  }

  // when link is clicked
  function onLinkClick(event) {
    if (clickClose === "true" || (clickClose === "auto" && isSmallScreen()))
      closePanel();
    else openPanel();
  }

  // open panel if closed, close if opened
  function togglePanel() {
    const panel = document.getElementById("toc_panel");
    if (!panel) return;

    if (panel.dataset.open === "true") closePanel();
    else openPanel();
  }

  // open panel
  function openPanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "true";
  }

  // close panel
  function closePanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "false";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- th list icon -->

<template class="icon_th_list">
  <!-- modified from: https://fontawesome.com/icons/th-list -->
  <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
    <path
      fill="currentColor"
      d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* toc panel */
    #toc_panel {
      box-sizing: border-box;
      position: fixed;
      top: 0;
      left: 0;
      background: #ffffff;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      z-index: 2;
    }

    /* toc panel when closed */
    #toc_panel[data-open="false"] {
      min-width: 60px;
      width: 60px;
      height: 60px;
      border-right: solid 1px #bdbdbd;
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc panel when open */
    #toc_panel[data-open="true"] {
      min-width: 260px;
      max-width: 480px;
      /* keep panel edge consistent distance away from "page" edge */
      width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
      bottom: 0;
      border-right: solid 1px #bdbdbd;
    }

    /* toc panel header */
    #toc_header {
      box-sizing: border-box;
      display: flex;
      flex-direction: row;
      align-items: center;
      height: 60px;
      margin: 0;
      padding: 20px;
    }

    /* toc panel header when hovered */
    #toc_header:hover {
      cursor: pointer;
    }

    /* toc panel header when panel open */
    #toc_panel[data-open="true"] > #toc_header {
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc open/close header button */
    #toc_button {
      margin-right: 20px;
    }

    /* hide toc list and header text when closed */
    #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
    #toc_panel[data-open="false"] > #toc_list {
      display: none;
    }

    /* toc list of entries */
    #toc_list {
      box-sizing: border-box;
      width: 100%;
      padding: 20px;
      position: absolute;
      top: calc(60px + 1px);
      bottom: 0;
      overflow: auto;
    }

    /* toc entry, link to section in document */
    .toc_link {
      display: block;
      padding: 5px;
      position: relative;
      font-weight: 600;
      text-decoration: none;
    }

    /* toc entry when hovered or when "viewed" */
    .toc_link:hover,
    .toc_link[data-viewing="true"] {
      background: #f5f5f5;
    }

    /* toc entry, level 1 indentation */
    .toc_link[data-level="1"] {
      margin-left: 0;
    }

    /* toc entry, level 2 indentation */
    .toc_link[data-level="2"] {
      margin-left: 20px;
    }

    /* toc entry, level 3 indentation */
    .toc_link[data-level="3"] {
      margin-left: 40px;
    }

    /* toc entry, level 4 indentation */
    .toc_link[data-level="4"] {
      margin-left: 60px;
    }

    /* toc entry bullets */
    #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
      position: absolute;
      left: -15px;
      top: -1px;
      font-size: 1.5em;
    }

    /* toc entry, level 2 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
      content: "\2022";
    }

    /* toc entry, level 3 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
      content: "\25AB";
    }

    /* toc entry, level 4 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
      content: "-";
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* push <body> ("page") element down to make room for toc icon */
    .toc_body_nudge {
      padding-top: 60px;
    }

    /* toc icon when panel closed and not hovered */
    #toc_panel[data-open="false"]:not(:hover) {
      background: rgba(255, 255, 255, 0.75);
    }
  }

  /* always hide toc panel on print */
  @media only print {
    #toc_panel {
      display: none;
    }
  }
</style>
<!-- 
  Tooltips Plugin

  Makes it such that when the user hovers or focuses a link to a citation or
  figure, a tooltip appears with a preview of the reference content, along with
  arrows to navigate between instances of the same reference in the document.
-->

<script type="module">
  // whether user must click off to close tooltip instead of just un-hovering
  const clickClose = "false";
  // delay (in ms) between opening and closing tooltip
  const delay = "100";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach hover and focus listeners to link
      link.addEventListener("mouseover", onLinkHover);
      link.addEventListener("mouseleave", onLinkUnhover);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("touchend", onLinkTouch);
    }

    // attach mouse, key, and resize listeners to window
    window.addEventListener("mousedown", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("keyup", onKeyUp);
    window.addEventListener("resize", onResize);
  }

  // when link is hovered
  function onLinkHover() {
    // function to open tooltip
    const delayOpenTooltip = function () {
      openTooltip(this);
    }.bind(this);

    // run open function after delay
    this.openTooltipTimer = window.setTimeout(delayOpenTooltip, delay);
  }

  // when mouse leaves link
  function onLinkUnhover() {
    // cancel opening tooltip
    window.clearTimeout(this.openTooltipTimer);

    // don't close on unhover if option specifies
    if (clickClose === "true") return;

    // function to close tooltip
    const delayCloseTooltip = function () {
      // if tooltip open and if mouse isn't over tooltip, close
      const tooltip = document.getElementById("tooltip");
      if (tooltip && !tooltip.matches(":hover")) closeTooltip();
    };

    // run close function after delay
    this.closeTooltipTimer = window.setTimeout(delayCloseTooltip, delay);
  }

  // when link is focused (tabbed to)
  function onLinkFocus(event) {
    openTooltip(this);
  }

  // when link is touched on touch screen
  function onLinkTouch(event) {
    // attempt to force hover state on first tap always, and trigger
    // regular link click (and navigation) on second tap
    if (event.target === document.activeElement) event.target.click();
    else {
      document.activeElement.blur();
      event.target.focus();
    }
    if (event.cancelable) event.preventDefault();
    event.stopPropagation();
    return false;
  }

  // when mouse is clicked anywhere in window
  function onClick(event) {
    closeTooltip();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("tooltip_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("tooltip_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeTooltip();
        break;
    }
  }

  // when window is resized or zoomed
  function onResize() {
    closeTooltip();
  }

  // get all links of types we wish to handle
  function getLinks() {
    const queries = [];
    // exclude buttons, anchor links, toc links, etc
    const exclude =
      ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    queries.push('a[href^="#ref-"]' + exclude); // citation links
    queries.push('a[href^="#fig:"]' + exclude); // figure links
    const query = queries.join(", ");
    return document.querySelectorAll(query);
  }

  // get links with same target, get index of link in set, get total
  // same links
  function getSameLinks(link) {
    const sameLinks = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        sameLinks.push(otherLink);
    }

    return {
      elements: sameLinks,
      index: sameLinks.indexOf(link),
      total: sameLinks.length,
    };
  }

  // open tooltip
  function openTooltip(link) {
    // delete tooltip if it exists, start fresh
    closeTooltip();

    // make tooltip element
    const tooltip = makeTooltip(link);

    // if source couldn't be found and tooltip not made, exit
    if (!tooltip) return;

    // make navbar elements
    const navBar = makeNavBar(link);
    if (navBar) tooltip.firstElementChild.appendChild(navBar);

    // attach tooltip to page
    document.body.appendChild(tooltip);

    // position tooltip
    const position = function () {
      positionTooltip(link);
    };
    position();

    // if tooltip contains images, position again after they've loaded
    const imgs = tooltip.querySelectorAll("img");
    for (const img of imgs) img.addEventListener("load", position);
  }

  // close (delete) tooltip
  function closeTooltip() {
    const tooltip = document.getElementById("tooltip");
    if (tooltip) tooltip.remove();
  }

  // make tooltip
  function makeTooltip(link) {
    // get target element that link points to
    const source = getSource(link);

    // if source can't be found, exit
    if (!source) return;

    // create new tooltip
    const tooltip = document.createElement("div");
    tooltip.id = "tooltip";
    const tooltipContent = document.createElement("div");
    tooltipContent.id = "tooltip_content";
    tooltip.appendChild(tooltipContent);

    // make copy of source node and put in tooltip
    const sourceCopy = makeCopy(source);
    tooltipContent.appendChild(sourceCopy);

    // attach mouse event listeners
    tooltip.addEventListener("click", onTooltipClick);
    tooltip.addEventListener("mousedown", onTooltipClick);
    tooltip.addEventListener("touchstart", onTooltipClick);
    tooltip.addEventListener("mouseleave", onTooltipUnhover);

    // (for interaction with lightbox plugin)
    // transfer click on tooltip copied img to original img
    const sourceImg = source.querySelector("img");
    const sourceCopyImg = sourceCopy.querySelector("img");
    if (sourceImg && sourceCopyImg) {
      const clickImg = function () {
        sourceImg.click();
        closeTooltip();
      };
      sourceCopyImg.addEventListener("click", clickImg);
    }

    return tooltip;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
      "class",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // when tooltip is clicked
  function onTooltipClick(event) {
    // when user clicks on tooltip, stop click from transferring
    // outside of tooltip (eg, click off to close tooltip, or eg click
    // off to unhighlight same refs)
    event.stopPropagation();
  }

  // when tooltip is unhovered
  function onTooltipUnhover(event) {
    if (clickClose === "true") return;

    // make sure new mouse/touch/focus no longer over tooltip or any
    // element within it
    const tooltip = document.getElementById("tooltip");
    if (!tooltip) return;
    if (this.contains(event.relatedTarget)) return;

    closeTooltip();
  }

  // make nav bar to go betwen prev/next instances of same reference
  function makeNavBar(link) {
    // find other links to the same source
    const sameLinks = getSameLinks(link);

    // don't show nav bar when singular reference
    if (sameLinks.total <= 1) return;

    // find prev/next links with same target
    const prevLink = getPrevLink(link, sameLinks);
    const nextLink = getNextLink(link, sameLinks);

    // create nav bar
    const navBar = document.createElement("div");
    navBar.id = "tooltip_nav_bar";
    const text = sameLinks.index + 1 + " of " + sameLinks.total;

    // create nav bar prev/next buttons
    const prevButton = document.createElement("button");
    const nextButton = document.createElement("button");
    prevButton.id = "tooltip_prev_button";
    nextButton.id = "tooltip_next_button";
    prevButton.title =
      "Jump to the previous occurence of this item in the document [←]";
    nextButton.title =
      "Jump to the next occurence of this item in the document [→]";
    prevButton.classList.add("icon_button");
    nextButton.classList.add("icon_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;
    nextButton.innerHTML =
      document.querySelector(".icon_caret_right").innerHTML;
    navBar.appendChild(prevButton);
    navBar.appendChild(document.createTextNode(text));
    navBar.appendChild(nextButton);

    // attach click listeners to buttons
    prevButton.addEventListener("click", function () {
      onPrevNextClick(link, prevLink);
    });
    nextButton.addEventListener("click", function () {
      onPrevNextClick(link, nextLink);
    });

    return navBar;
  }

  // get previous link with same target
  function getPrevLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if < 1
    let index;
    if (sameLinks.index - 1 >= 0) index = sameLinks.index - 1;
    else index = sameLinks.total - 1;
    return sameLinks.elements[index];
  }

  // get next link with same target
  function getNextLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if > total
    let index;
    if (sameLinks.index + 1 <= sameLinks.total - 1) index = sameLinks.index + 1;
    else index = 0;
    return sameLinks.elements[index];
  }

  // get element that is target of link or url hash
  function getSource(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector('[id="' + id + '"]');
    if (!target) return;

    // if ref or figure, modify target to get expected element
    if (id.indexOf("ref-") === 0) target = target.querySelector(":nth-child(2)");
    else if (id.indexOf("fig:") === 0) target = target.querySelector("figure");

    return target;
  }

  // when prev/next arrow button is clicked
  function onPrevNextClick(link, prevNextLink) {
    if (link && prevNextLink)
      goToElement(prevNextLink, window.innerHeight * 0.5);
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);
    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // determine position to place tooltip based on link position in
  // viewport and tooltip size
  function positionTooltip(link, left, top) {
    const tooltipElement = document.getElementById("tooltip");
    if (!tooltipElement) return;

    // get convenient vars for position/dimensions of
    // link/tooltip/page/view
    link = getRectInPage(link);
    const tooltip = getRectInPage(tooltipElement);
    const view = getRectInPage();

    // horizontal positioning
    if (left)
      // use explicit value
      left = left;
    else if (link.left + tooltip.width < view.right)
      // fit tooltip to right of link
      left = link.left;
    else if (link.right - tooltip.width > view.left)
      // fit tooltip to left of link
      left = link.right - tooltip.width;
    // center tooltip in view
    else left = (view.right - view.left) / 2 - tooltip.width / 2;

    // vertical positioning
    if (top)
      // use explicit value
      top = top;
    else if (link.top - tooltip.height > view.top)
      // fit tooltip above link
      top = link.top - tooltip.height;
    else if (link.bottom + tooltip.height < view.bottom)
      // fit tooltip below link
      top = link.bottom;
    else {
      // center tooltip in view
      top = view.top + view.height / 2 - tooltip.height / 2;
      // nudge off of link to left/right if possible
      if (link.right + tooltip.width < view.right) left = link.right;
      else if (link.left - tooltip.width > view.left)
        left = link.left - tooltip.width;
    }

    tooltipElement.style.left = left + "px";
    tooltipElement.style.top = top + "px";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* tooltip container */
    #tooltip {
      position: absolute;
      width: 50%;
      min-width: 240px;
      max-width: 75%;
      z-index: 1;
    }

    /* tooltip content */
    #tooltip_content {
      margin-bottom: 5px;
      padding: 20px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
      overflow-wrap: break-word;
    }

    /* tooltip copy of paragraphs and figures */
    #tooltip_content > p,
    #tooltip_content > figure {
      margin: 0;
      max-height: 320px;
      overflow-y: auto;
    }

    /* tooltip copy of <img> */
    #tooltip_content > figure > img,
    #tooltip_content > figure > svg {
      max-height: 260px;
    }

    /* navigation bar */
    #tooltip_nav_bar {
      margin-top: 10px;
      text-align: center;
    }

    /* navigation bar previous/next buton */
    #tooltip_nav_bar > .icon_button {
      position: relative;
      top: 3px;
    }

    /* navigation bar previous button */
    #tooltip_nav_bar > .icon_button:first-of-type {
      margin-right: 5px;
    }

    /* navigation bar next button */
    #tooltip_nav_bar > .icon_button:last-of-type {
      margin-left: 5px;
    }
  }

  /* always hide tooltip on print */
  @media only print {
    #tooltip {
      display: none;
    }
  }
</style>
<!--
  Analytics Plugin (third-party) 
  
  Copy and paste code from Google Analytics or similar service here.
-->
<!-- 
  Annotations Plugin

  Allows public annotation of the  manuscript. See https://web.hypothes.is/.
-->

<script type="module">
  // configuration
  window.hypothesisConfig = function () {
    return {
      branding: {
        accentColor: "#2196f3",
        appBackgroundColor: "#f8f8f8",
        ctaBackgroundColor: "#f8f8f8",
        ctaTextColor: "#000000",
        selectionFontFamily: "Open Sans, Helvetica, sans serif",
        annotationFontFamily: "Open Sans, Helvetica, sans serif",
      },
    };
  };

  // hypothesis client script
  const embed = "https://hypothes.is/embed.js";
  // hypothesis annotation count query url
  const query = "https://api.hypothes.is/api/search?limit=0&url=";

  // start script
  function start() {
    const button = makeButton();
    document.body.insertBefore(button, document.body.firstChild);
    insertCount(button);
  }

  // make button
  function makeButton() {
    // create button
    const button = document.createElement("button");
    button.id = "hypothesis_button";
    button.innerHTML = document.querySelector(".icon_hypothesis").innerHTML;
    button.title = "Hypothesis annotations";
    button.classList.add("icon_button");

    function onClick(event) {
      onButtonClick(event, button);
    }

    // attach click listeners
    button.addEventListener("click", onClick);

    return button;
  }

  // insert annotations count
  async function insertCount(button) {
    // get annotation count from Hypothesis based on url
    let count = "-";
    try {
      const canonical = document.querySelector('link[rel="canonical"]');
      const location = window.location;
      const url = encodeURIComponent((canonical || location).href);
      const response = await fetch(query + url);
      const json = await response.json();
      count = json.total || "-";
    } catch (error) {
      console.log(error);
    }

    // put count into button
    const counter = document.createElement("span");
    counter.id = "hypothesis_count";
    counter.innerHTML = count;
    button.title = "View " + count + " Hypothesis annotations";
    button.append(counter);
  }

  // when button is clicked
  function onButtonClick(event, button) {
    const script = document.createElement("script");
    script.src = embed;
    document.body.append(script);
    button.remove();
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- hypothesis icon -->

<template class="icon_hypothesis">
  <!-- modified from: https://simpleicons.org/icons/hypothesis.svg / https://git.io/Jf1VB -->
  <svg width="16" height="16" viewBox="0 0 24 24" tabindex="-1">
    <path
      fill="currentColor"
      d="M3.43 0C2.5 0 1.72 .768 1.72 1.72V18.86C1.72 19.8 2.5 20.57 3.43 20.57H9.38L12 24L14.62 20.57H20.57C21.5 20.57 22.29 19.8 22.29 18.86V1.72C22.29 .77 21.5 0 20.57 0H3.43M5.14 3.43H7.72V9.43S8.58 7.72 10.28 7.72C12 7.72 13.74 8.57 13.74 11.24V17.14H11.16V12C11.16 10.61 10.28 10.07 9.43 10.29C8.57 10.5 7.72 11.41 7.72 13.29V17.14H5.14V3.43M18 13.72C18.95 13.72 19.72 14.5 19.72 15.42A1.71 1.71 0 0 1 18 17.13A1.71 1.71 0 0 1 16.29 15.42C16.29 14.5 17.05 13.71 18 13.71Z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  /* hypothesis activation button */
  #hypothesis_button {
    box-sizing: border-box;
    position: fixed;
    top: 0;
    right: 0;
    width: 60px;
    height: 60px;
    background: #ffffff;
    border-radius: 0;
    border-left: solid 1px #bdbdbd;
    border-bottom: solid 1px #bdbdbd;
    box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
    z-index: 2;
  }

  /* hypothesis button svg */
  #hypothesis_button > svg {
    position: relative;
    top: -4px;
  }

  /* hypothesis annotation count */
  #hypothesis_count {
    position: absolute;
    left: 0;
    right: 0;
    bottom: 5px;
  }

  /* side panel */
  .annotator-frame {
    width: 280px !important;
  }

  /* match highlight color to rest of theme */
  .annotator-highlights-always-on .annotator-hl {
    background-color: #ffeb3b !important;
  }

  /* match focused color to rest of theme */
  .annotator-hl.annotator-hl-focused {
    background-color: #ff8a65 !important;
  }

  /* match bucket bar color to rest of theme */
  .annotator-bucket-bar {
    background: #f5f5f5 !important;
  }

  /* always hide button, toolbar, and tooltip on print */
  @media only print {
    #hypothesis_button {
      display: none;
    }

    .annotator-frame {
      display: none !important;
    }

    hypothesis-adder {
      display: none !important;
    }
  }
</style>
<!-- 
  Mathjax Plugin (third-party) 

  Allows the proper rendering of math/equations written in LaTeX.
  See https://www.mathjax.org/.
-->

<script type="text/x-mathjax-config">
  // configuration
  MathJax.Hub.Config({
    "CommonHTML": { linebreaks: { automatic: true } },
    "HTML-CSS": { linebreaks: { automatic: true } },
    "SVG": { linebreaks: { automatic: true } },
    "fast-preview": { disabled: true }
  });
</script>

<script
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"
  integrity="sha512-0doc9hKxR3PYwso42RD1p5ySZpzzuDiOwMrdCEh2WdJZCjcmFKc/wEnL+z8fBQrnHoiNWbo+3fiGkOYXBdQp4A=="
  crossorigin="anonymous"
></script>

<style>
  /* mathjax containers */
  .math.display > span:not(.MathJax_Preview) {
    /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
    display: flex !important;
    overflow-x: auto !important;
    overflow-y: hidden !important;
    justify-content: center;
    align-items: center;
    margin: 0 !important;
  }

  /* right click menu */
  .MathJax_Menu {
    border-radius: 5px !important;
    border: solid 1px #bdbdbd !important;
    box-shadow: none !important;
  }

  /* equation auto-number */
  span[id^="eq:"] > span.math.display + span {
    font-weight: 600;
  }

  /* equation */
  span[id^="eq:"] > span.math.display > span {
    /* nudge to make room for equation auto-number and anchor */
    margin-right: 60px !important;
  }
</style>
</body>
</html>
